<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreecccf312d4bec46ee08bb315ec42ac751092a165dtargetblankcccf312a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/cccf312d4bec46ee08bb315ec42ac751092a165d" target="_blank">cccf312</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreecccf312d4bec46ee08bb315ec42ac751092a165dtargetblankcccf312a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/cccf312d4bec46ee08bb315ec42ac751092a165d" target="_blank">cccf312</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/rds/
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruv-all-gsea-c2.csv
    Ignored:    output/salmon-ruv-all-gsea-c5.csv
    Ignored:    output/salmon-ruv-all.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  analysis/old_analysis/
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/salmon-SatuRn.Rmd
    Deleted:    analysis/salmon-bandits.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/cccf312d4bec46ee08bb315ec42ac751092a165d/analysis/index.Rmd" target="_blank">cccf312</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-26
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/salmon-RUV-all.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/index.html" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a9955fecedf1cbec5b44a7f142d12a9482b0163c/docs/index.html" target="_blank">a9955fe</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/0703d92188a25663a39ce00dae1c01079efb70e5/analysis/index.Rmd" target="_blank">0703d92</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-09
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/5a4b9cbcd86e3715149351888cb0fe5a5e9d9808/docs/index.html" target="_blank">5a4b9cb</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/7c667ac36ae19589df66aaff65cefadabd648f78/analysis/index.Rmd" target="_blank">7c667ac</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-03
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4af30fbdc861e2d9d8b7665cd9f77306a9310a2e/docs/index.html" target="_blank">4af30fb</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/3ba9d96b4fc74711107b340259dfb4a113f68a65/analysis/index.Rmd" target="_blank">3ba9d96</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-12-21
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/STAR-FC-RUV-exclude-US-ab.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/83478b2b3db568b59261be63ba8244501571cc29/docs/index.html" target="_blank">83478b2</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a5f6e5ec9ad2bdae2a6d2d784b7a7f2f3bb18cbf/analysis/index.Rmd" target="_blank">a5f6e5e</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/d96660bec01347bcec29e84c1c4313e95abc8d04/docs/index.html" target="_blank">d96660b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/d32e63b606d16b69ae5c35eae7e7615e92ac90c3/analysis/index.Rmd" target="_blank">d32e63b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
wflow_publish(c(“analysis/STAR-FC-all.Rmd”, “analysis/STAR-FC-exclude-US-ab.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/index.html" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/787e0a195e4863f0eb47d672e5689a3683f6ab43/analysis/index.Rmd" target="_blank">787e0a1</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
<td>
wflow_publish(c(“analysis/STAR-FC-all.Rmd”, “analysis/STAR-FC-exclude-US-ab.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/index.html" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/2ee5593cf8f25f38c0fae29032e77240be940634/analysis/index.Rmd" target="_blank">2ee5593</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/STAR-FC-all.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/29f3949d70cf4ad2bcd5813ca86ec919beb93e31/docs/index.html" target="_blank">29f3949</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/5dc9543377653d0165c905efe20e775e123315aa/analysis/index.Rmd" target="_blank">5dc9543</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-02-24
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/index.Rmd”, “analysis/license.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/89b29ff62ca8c0c8c6792552d14a3637cd0cca84/docs/index.html" target="_blank">89b29ff</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-02-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/0d8231a42c85240c47bb11ee84dc314f7e0bd5bc/analysis/index.Rmd" target="_blank">0d8231a</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-02-24
</td>
<td>
wflow_publish(c(“analysis/about.Rmd”, “analysis/index.Rmd”, “analysis/license.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/ab6655b5d7988cd3a714d26be79c5737d9c2f5f3/analysis/index.Rmd" target="_blank">ab6655b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-02-24
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="amniotic-fluid-cfrna" class="section level1">
<h1>Amniotic fluid cfRNA</h1>
<p>This site contains the analysis of RNAseq data generated from amniotic fluid cell-free RNA (cfRNA).</p>
<p>The data is from Lisa Hui in the Embryonic Epidemiology group at the Murdoch Children’s Research Institute (MCRI). They have collected amniotic fluid from pregnancies with suspected (and subsequently confirmed) cytomegalovirus (CMV) infection. They also have amniotic fluid from “normal” pregnancies i.e. pregnancies that were confirmed to be negative for CMV infection. They are interested in differences in the cfRNA between the CMV negative and positive pregnancies. Nineteen samples (10 CMV negative and 9 CMV positive) were originally sequenced (150bp paired-end) at the MCRI TGU. A further batch of 9 samples was then sequenced.</p>
<p>Follow the links below to explore the different parts of the analysis.</p>
<div id="differential-gene-expression-dge-analysis" class="section level2">
<h2>Differential gene expression (DGE) analysis</h2>
<ul>
<li><a href="salmon-ruv-all.html">DGE analysis</a> - Following quality control and normalisation, we preformed a differential expression analysis between CMV positive and CMV negative samples. We also performed gene set testing to identify any gene sets of interest.</li>
</ul>
</div>
<div id="differential-transcript-usage-dtu-analysis" class="section level2">
<h2>Differential transcript usage (DTU) Analysis</h2>
<ul>
<li>Salmon-BANDITS
<ul>
<li><a href="salmon-bandits.html">Salmon-BANDITS</a></li>
</ul></li>
<li>Salmon-DRIMseq
<ul>
<li><a href="salmon-DRIMseq.html">Salmon-DRIMseq</a></li>
</ul></li>
<li>Salmon-SatuRn
<ul>
<li><a href="salmon-SatuRn.html">Salmon-SatuRn</a></li>
</ul></li>
</ul>
<!-- New Analysis -->
<!-- ------------ -->
<!-- * Read mapping, counting and quality control. -->
<!--     * [MultiQC analysis](multiqc_report.html) -->
<!-- * Differential expression analysis -->
<!--     * [Analysis results](STAR-FC-all.html) - This is the new analysis comparing CMV positive and CMV negative samples of the full cohort.    -->
<!-- * Differential expression analysis (exclude samples with ultrasound abnormality) -->
<!--     * [Anlaysis results](STAR-FC-exclude-US-ab.html) - This is the same analysis as above but excluding all samples that had an ultrasound abnormality identified. -->
<!-- * Differential expression analysis (RUVseq) -->
<!--     * [Analysis results](STAR-FC-RUV-all.html) - This is the new analysis comparing CMV positive and CMV negative samples of the full cohort, using [RUVseq](https://bioconductor.org/packages/release/bioc/html/RUVSeq.html) to remove additional unwanted variation.     -->
<!-- * Differential expression analysis (RUVseq) -->
<!--     * [Analysis results](STAR-FC-RUV-exclude-US-ab.html) - RUVseq analysis comparing CMV positive and CMV negative samples, excluding samples with ultrasound abnormalities.  -->
<!-- Pilot Analysis -->
<!-- -------------- -->
<!-- * Differential expression analysis -->
<!--     * [Limma-voom analysis](salmon-limma-voom.html) - This is the initial analysis comparing CMV positive and CMV negative samples. Following quality control and normalisation, we preformed a differential expression analysis between CMV positive and CMV negative samples. We also performed gene set testing to identify any gene sets that may differentially expressed.    -->
<!-- * Differential expression analysis (exclude CMV11 and pairs) -->
<!--     * [Limma-voom analysis excluding CMV11](salmon-limma-voom-exclude-CMV11.html) - This is the same analysis as above but excluding CMV11 (which had an ultrasound abnormality identified) and its pairs.  -->
</div>
<div id="licenses" class="section level2">
<h2>Licenses</h2>
<p>The code in this analysis is covered by the <a href="https://choosealicense.com/licenses/mit/" title="MIT License">MIT license</a> and the written content on this website is covered by a <a href="https://creativecommons.org/licenses/by/4.0/" title="CC-BY License">Creative Commons CC-BY</a> license.</p>
</div>
<div id="citations" class="section level2">
<h2>Citations</h2>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
