<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Transcript Usage (DTU) Analysis: SatuRn</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential Transcript Usage (DTU) Analysis: SatuRn</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree63282589d25b165c63f636b0405ded24e5a200bbtargetblank6328258a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/63282589d25b165c63f636b0405ded24e5a200bb" target="_blank">6328258</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree63282589d25b165c63f636b0405ded24e5a200bbtargetblank6328258a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/63282589d25b165c63f636b0405ded24e5a200bb" target="_blank">6328258</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    analysis/old_analysis/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/rds/
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruv-all-gsea-c2.csv
    Ignored:    output/salmon-ruv-all-gsea-c5.csv
    Ignored:    output/salmon-ruv-all.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-SatuRn.Rmd</code>) and HTML (<code>docs/salmon-SatuRn.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/63282589d25b165c63f636b0405ded24e5a200bb/analysis/salmon-SatuRn.Rmd" target="_blank">6328258</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(c(“analysis/salmon-BANDITS.Rmd”, “analysis/salmon-DRIMseq.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/8e31f4365bf53b088b6467efd1d23e6e3f1594d5/docs/salmon-SatuRn.html" target="_blank">8e31f43</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/382671cd4607b48ba3f75da5f7c50b043e5fdcf5/analysis/salmon-SatuRn.Rmd" target="_blank">382671c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-09
</td>
<td>
wflow_publish(“analysis/salmon-SatuRn.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
</tbody>
</table>
<p>Only retain paired samples with clinical information for donstream analysis.</p>
<pre class="r"><code>int &lt;- intersect(names(files), targets$id)
targets &lt;- targets[match(int, targets$id),]
files &lt;- files[match(int, names(files))]

files</code></pre>
<pre><code>                                                                                                                                                                                           CMV30 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV31 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV8 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV9 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV26 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV14 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV15 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV20 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV21 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV1 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV2 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV3 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV4 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV10 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV11 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV19 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV35 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV51 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV52 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV53 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV54 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV56 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV57 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV58 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV60 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV61 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; </code></pre>
<p>Read in transcript count data and remove transcripts with no counts.</p>
<pre class="r"><code>txi &lt;- tximport(files, type = &quot;salmon&quot;, txOut = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 </code></pre>
<pre class="r"><code>cts &lt;- txi$counts
cts &lt;- cts[rowSums(cts) &gt; 0,]

dim(cts)</code></pre>
<pre><code>[1] 155749     26</code></pre>
<p>Plot effective library size per sample.</p>
<pre class="r"><code>colSums(cts) %&gt;%
  data.frame %&gt;%
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  dplyr::rename(&quot;libsize&quot; = &quot;.&quot;) %&gt;%
ggplot(aes(x = sample, y = libsize)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ```{r} -->
<!-- edb <- EnsDb.Hsapiens.v86 -->
<!-- txs <- transcripts(edb)  -->
<!-- head(txs) -->
<!-- ``` -->
<p>Associate transcripts with gene IDs for gene-level summarization. Filter out genes without Entrez IDs, genes with only one transcript isoform and transcripts without any counts across all samples.</p>
<pre class="r"><code>txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene
txdf &lt;- select(txdb, keys(txdb, keytype = &quot;TXNAME&quot;), 
               &quot;GENEID&quot;, &quot;TXNAME&quot;, &quot;SYMBOL&quot;) %&gt;%
  drop_na(GENEID)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>txdf %&gt;% dplyr::inner_join(select(org.Hs.eg.db, 
                                  keys = unique(txdf$GENEID), 
                           columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
                           keytype=&quot;ENTREZID&quot;), 
                           by = c(&quot;GENEID&quot; = &quot;ENTREZID&quot;)) %&gt;%
  dplyr::distinct() %&gt;%
  dplyr::filter((duplicated(GENEID) | 
                  duplicated(GENEID, fromLast = TRUE)) &amp;
                  TXNAME %in% rownames(cts)) -&gt; txdf</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>head(txdf, n = 10) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">TXNAME</th>
<th align="left">GENEID</th>
<th align="left">SYMBOL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000623808.3</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000445118.7</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000669922.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000670780.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000667414.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000666741.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000623070.4</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000659124.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000658846.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000608189.5</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
</tbody>
</table>
<p>Match transcript count data to transcript annotation table.</p>
<pre class="r"><code>cts &lt;- cts[match(txdf$TXNAME, rownames(cts)),]

head(cts) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CMV30</th>
<th align="right">CMV31</th>
<th align="right">CMV8</th>
<th align="right">CMV9</th>
<th align="right">CMV26</th>
<th align="right">CMV14</th>
<th align="right">CMV15</th>
<th align="right">CMV20</th>
<th align="right">CMV21</th>
<th align="right">CMV1</th>
<th align="right">CMV2</th>
<th align="right">CMV3</th>
<th align="right">CMV4</th>
<th align="right">CMV10</th>
<th align="right">CMV11</th>
<th align="right">CMV19</th>
<th align="right">CMV35</th>
<th align="right">CMV51</th>
<th align="right">CMV52</th>
<th align="right">CMV53</th>
<th align="right">CMV54</th>
<th align="right">CMV56</th>
<th align="right">CMV57</th>
<th align="right">CMV58</th>
<th align="right">CMV60</th>
<th align="right">CMV61</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000623808.3</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">2.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">2.840</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.113</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000445118.7</td>
<td align="right">81.080</td>
<td align="right">49.139</td>
<td align="right">51.049</td>
<td align="right">35.215</td>
<td align="right">40.649</td>
<td align="right">71.627</td>
<td align="right">112.693</td>
<td align="right">72.270</td>
<td align="right">88.424</td>
<td align="right">23.261</td>
<td align="right">36.969</td>
<td align="right">3.356</td>
<td align="right">25.536</td>
<td align="right">88.967</td>
<td align="right">15.132</td>
<td align="right">3.924</td>
<td align="right">51.927</td>
<td align="right">39.892</td>
<td align="right">80.193</td>
<td align="right">45.610</td>
<td align="right">48.656</td>
<td align="right">85.781</td>
<td align="right">95.585</td>
<td align="right">155.849</td>
<td align="right">89.77</td>
<td align="right">52.703</td>
</tr>
<tr class="odd">
<td align="left">ENST00000669922.1</td>
<td align="right">7.952</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">37.732</td>
<td align="right">7.738</td>
<td align="right">0.000</td>
<td align="right">6.574</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.538</td>
<td align="right">0.000</td>
<td align="right">18.000</td>
<td align="right">0.000</td>
<td align="right">11.000</td>
<td align="right">3.808</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000670780.1</td>
<td align="right">15.481</td>
<td align="right">22.471</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">24.094</td>
<td align="right">14.058</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">4.001</td>
<td align="right">0.000</td>
<td align="right">15.947</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">5.814</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">16.986</td>
<td align="right">11.345</td>
<td align="right">0.000</td>
<td align="right">1.836</td>
<td align="right">0.000</td>
<td align="right">6.21</td>
<td align="right">20.954</td>
</tr>
<tr class="odd">
<td align="left">ENST00000667414.1</td>
<td align="right">0.000</td>
<td align="right">1.155</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.196</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.049</td>
<td align="right">1.436</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000666741.1</td>
<td align="right">5.858</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">37.606</td>
<td align="right">7.820</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">21.911</td>
<td align="right">16.865</td>
<td align="right">0.000</td>
<td align="right">49.674</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">39.220</td>
<td align="right">25.097</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
</tbody>
</table>
</div>
<div id="data-wrangling" class="section level1">
<h1>Data wrangling</h1>
<p>Next, we perform some data wrangling steps to get the data in a format that is suited for <em>satuRn</em>.</p>
<p>! Important: <em>satuRn</em> is implemented such that the columns with transcript identifiers is names isoform_id, while the column containing gene identifiers should be named gene_id. In addition, following chunk removes transcripts that are the only isoform expressed of a certain gene, as they cannot be used in a DTU analysis.</p>
<!-- ```{r} -->
<!-- # Get the transcript information in correct format -->
<!-- txInfo <- as.data.frame(matrix(data = NA, nrow = length(txs), ncol = 2)) -->
<!-- colnames(txInfo) <- c("isoform_id", "gene_id") -->
<!-- txInfo$isoform_id <- txs$tx_id -->
<!-- txInfo$gene_id <- txs$gene_id -->
<!-- rownames(txInfo) <- txInfo$isoform_id -->
<!-- head(txInfo) -->
<!-- ``` -->
<pre class="r"><code>txdf %&gt;% dplyr::rename(isoform_id = TXNAME, 
                       gene_id = GENEID) -&gt; txdf 
dim(txdf)</code></pre>
<pre><code>[1] 130034      3</code></pre>
<!-- Import transcript counts, match with gene IDs and remove transcripts that are the only isoform of a gene. -->
<!-- ```{r} -->
<!-- txi <- tximport(files, type = "salmon", txOut = TRUE) -->
<!-- cts <- txi$counts -->
<!-- cts <- cts[rowSums(cts) > 0,] -->
<!-- #rownames(cts) <- sub("\\..*", "", rownames(cts))  -->
<!-- txInfo <- txInfo[txInfo$isoform_id %in% rownames(cts), ] -->
<!-- txInfo <- subset(txInfo,  -->
<!--                  duplicated(gene_id) | duplicated(gene_id, fromLast = TRUE)) -->
<!-- cts <- cts[which(rownames(cts) %in% txInfo$isoform_id), ] -->
<!-- dim(cts) -->
<!-- ``` -->
</div>
<div id="filtering" class="section level1">
<h1>Filtering</h1>
<p>Here we perform some feature-level filtering. For this task, we adopt the filtering criterion that is implemented in the R package <em>BANDITS</em>. <!-- ```{r} --> <!-- filter_edgeR <- filterByExpr(cts, --> <!--     design = NULL, --> <!--     group = targets$CMV_status, --> <!-- )  --></p>
<!-- table(filter_edgeR) -->
<!-- ``` -->
<pre class="r"><code>transcripts_to_keep &lt;- filter_transcripts(gene_to_transcript = txdf[, c(2,1)],
                                          transcript_counts = cts, 
                                          min_transcript_proportion = 0.01,
                                          min_transcript_counts = min(table(targets$CMV_status)), 
                                          min_gene_counts = ncol(cts))</code></pre>
<pre><code>After filtering, 50.63% of transcripts are kept</code></pre>
<pre class="r"><code>length(transcripts_to_keep)</code></pre>
<pre><code>[1] 65831</code></pre>
<p>Keep only transcripts that pass filtering criteria.</p>
<pre class="r"><code>keep &lt;- match(transcripts_to_keep, rownames(cts))

cts &lt;- cts[keep,]
txdf &lt;- txdf[keep,]</code></pre>
</div>
<div id="generate-summarizedexperiment" class="section level1">
<h1>Generate SummarizedExperiment</h1>
<p>All three main functions of <em>satuRn</em> require a <code>SummarizedExperiment</code> object as an input class.</p>
<p>Do not forget to include the design matrix formula to the <code>SummarizedExperiment</code> as indicated below. As such, the object contains all the information required for the downstream DTU analysis.</p>
<pre class="r"><code>sumExp &lt;- SummarizedExperiment::SummarizedExperiment(
    assays = list(counts = cts),
    colData = targets,
    rowData = txdf
)

# specify design formula from colData
metadata(sumExp)$formula &lt;- ~ 0 + as.factor(colData(sumExp)$CMV_status)
sumExp</code></pre>
<pre><code>class: SummarizedExperiment 
dim: 65831 26 
metadata(1): formula
assays(1): counts
rownames(65831): ENST00000598345.1 ENST00000599109.5 ...
  ENST00000543927.6 ENST00000638598.2
rowData names(3): isoform_id gene_id SYMBOL
colnames(26): CMV30 CMV31 ... CMV60 CMV61
colData names(6): id CMV_status ... GA_at_amnio indication</code></pre>
</div>
<div id="fit-quasi-binomial-generalized-linear-models-models" class="section level1">
<h1>Fit quasi-binomial generalized linear models models</h1>
<p>The <code>fitDTU</code> function of <em>satuRn</em> is used to model transcript usage in different groups of samples or cells. Here we adopt the default settings of the function.</p>
<pre class="r"><code>system.time({
sumExp &lt;- satuRn::fitDTU(
    object = sumExp,
    formula = ~ 0 + CMV_status,
    parallel = FALSE,
    BPPARAM = BiocParallel::bpparam(),
    verbose = TRUE
)
})</code></pre>
<pre><code>   user  system elapsed 
106.943   0.030 106.982 </code></pre>
</div>
<div id="set-up-contrast-matrix" class="section level1">
<h1>Set up contrast matrix</h1>
<p>First, we set up a contrast matrix. This allows us to test for differential transcript usage between groups of interest.</p>
<pre class="r"><code>group &lt;- as.factor(targets$CMV_status)
#pair &lt;- as.factor(targets$pair)
design &lt;- model.matrix(~ 0 + group)# + pair) # construct design matrix
colnames(design)[1:2] &lt;- levels(group)

L &lt;- limma::makeContrasts(
    Contrast1 = neg - pos,
    levels = design
)
L # contrast matrix</code></pre>
<pre><code>      Contrasts
Levels Contrast1
   neg         1
   pos        -1</code></pre>
</div>
<div id="perform-the-test" class="section level1">
<h1>Perform the test</h1>
<p>Next we can perform differential usage testing using <code>testDTU</code>. We again adopt default settings.</p>
<pre class="r"><code>sumExp &lt;- satuRn::testDTU(
    object = sumExp,
    contrasts = L,
    plot = FALSE,
    sort = TRUE
)</code></pre>
<p>The test results are now saved into the <code>rowData</code> of our <code>SummarizedExperiment</code> object under the name <code>fitDTUResult_</code> followed by the name of the contrast of interest (i.e. the column names of the contrast matrix). The results can be accessed as follows:</p>
<pre class="r"><code>head(rowData(sumExp)[[&quot;fitDTUResult_Contrast1&quot;]]) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">estimates</th>
<th align="right">se</th>
<th align="right">df</th>
<th align="right">t</th>
<th align="right">pval</th>
<th align="right">regular_FDR</th>
<th align="right">empirical_pval</th>
<th align="right">empirical_FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000265417.7</td>
<td align="right">3.958770e+15</td>
<td align="right">2.132338e+14</td>
<td align="right">25.77265</td>
<td align="right">18.56540</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.0e+00</td>
</tr>
<tr class="even">
<td align="left">ENST00000283296.12</td>
<td align="right">-3.958770e+15</td>
<td align="right">2.132338e+14</td>
<td align="right">25.77265</td>
<td align="right">-18.56540</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENST00000435136.8</td>
<td align="right">-3.926028e+15</td>
<td align="right">2.189772e+14</td>
<td align="right">25.77265</td>
<td align="right">-17.92894</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.0e+00</td>
</tr>
<tr class="even">
<td align="left">ENST00000224807.9</td>
<td align="right">3.832546e+15</td>
<td align="right">2.397173e+14</td>
<td align="right">25.77265</td>
<td align="right">15.98777</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1.0e-07</td>
</tr>
<tr class="odd">
<td align="left">ENST00000393459.5</td>
<td align="right">-3.832546e+15</td>
<td align="right">2.397173e+14</td>
<td align="right">25.77265</td>
<td align="right">-15.98777</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1.0e-07</td>
</tr>
<tr class="even">
<td align="left">ENST00000671800.1</td>
<td align="right">3.339259e+15</td>
<td align="right">2.563662e+14</td>
<td align="right">25.77265</td>
<td align="right">13.02535</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1.5e-06</td>
</tr>
</tbody>
</table>
</div>
<div id="visualize-dtu" class="section level1">
<h1>Visualize DTU</h1>
<p>Finally, we may visualize the usage of select transcripts in select groups of interest.</p>
<pre class="r"><code>group1 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;neg&quot;]
group2 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;pos&quot;]

plots &lt;- satuRn::plotDTU(
    object = sumExp,
    contrast = &quot;Contrast1&quot;,
    groups = list(group1, group2),
    coefficients = list(c(1, 0), 
                        c(0, 1)),
    summaryStat = &quot;model&quot;,
    top.n = 5
)

wrap_plots(plots, ncol = 1) + plot_layout(guides = &quot;collect&quot;)</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="two-stage-testing-procedure" class="section level1">
<h1>Two-stage testing procedure</h1>
<p><em>satuRn</em> returns transcript-level p-values for each of the specified contrasts. While we have shown that satuRn is able to adequately control the false discovery rate (FDR) at the transcript level (Gilis Jeroen 2021), (Van den Berge Koen 2017) argued that it is often desirable to control the FDR at the gene level. This boosts statistical power and eases downstream biological interpretation and validation, which typically occur at the gene level.</p>
<p>To this end, (Van den Berge Koen 2017) developed a testing procedure that is implemented in the BioConductor R package <em>stageR</em>. The procedure consists of two stages; a screening stage and a confirmation stage.</p>
<p>In the screening stage, gene-level FDR-adjusted p-values are computed, which aggregate the evidence for differential transcript usage over all transcripts within the gene. Only genes with an FDR below the desired nominal level are further considered in the second stage. In the confirmation stage, transcript-level p-values are adjusted for those genes, using a FWER-controlling method on the FDR-adjusted significance level.</p>
<p>In its current implementation, <em>stageR</em> can only perform stage-wise testing if only one contrast is of interest in a DTU setting. An analogous correction for the assessment of multiple contrasts for multiple transcripts per gene has not yet been implemented.</p>
<pre class="r"><code># transcript level p-values from satuRn
pvals &lt;- rowData(sumExp)[[&quot;fitDTUResult_Contrast1&quot;]]$empirical_pval

# compute gene level q-values
geneID &lt;- factor(rowData(sumExp)$gene_id)
geneSplit &lt;- split(seq(along = geneID), geneID)
pGene &lt;- sapply(geneSplit, function(i) min(pvals[i]))
pGene[is.na(pGene)] &lt;- 1
theta &lt;- unique(sort(pGene))

# gene-level significance testing
q &lt;- DEXSeq:::perGeneQValueExact(pGene, theta, geneSplit) 
qScreen &lt;- rep(NA_real_, length(pGene))
qScreen &lt;- q[match(pGene, theta)]
qScreen &lt;- pmin(1, qScreen)
names(qScreen) &lt;- names(geneSplit)

# prepare stageR input
tx2gene &lt;- as.data.frame(rowData(sumExp)[c(&quot;isoform_id&quot;, &quot;gene_id&quot;)])
colnames(tx2gene) &lt;- c(&quot;transcript&quot;, &quot;gene&quot;)

pConfirmation &lt;- matrix(matrix(pvals),
    ncol = 1,
    dimnames = list(rownames(tx2gene), &quot;transcript&quot;)
)

# create a stageRTx object
stageRObj &lt;- stageR::stageRTx(
    pScreen = qScreen,
    pConfirmation = pConfirmation,
    pScreenAdjusted = TRUE,
    tx2gene = tx2gene
)

# perform the two-stage testing procedure
stageRObj &lt;- stageR::stageWiseAdjustment(
    object = stageRObj,
    method = &quot;dtu&quot;,
    alpha = 0.05,
    allowNA = TRUE
)

# retrieves the adjusted p-values from the stageRTx object
padj &lt;- stageR::getAdjustedPValues(stageRObj,
    order = FALSE,
    onlySignificantGenes = TRUE
)</code></pre>
<pre><code>The returned adjusted p-values are based on a stage-wise testing approach and are only valid for the provided target OFDR level of 5%. If a different target OFDR level is of interest,the entire adjustment should be re-run. </code></pre>
<pre class="r"><code>padj %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">geneID</th>
<th align="left">txID</th>
<th align="right">gene</th>
<th align="right">transcript</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">ENST00000598345.1</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">ENST00000599109.5</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">ENST00000286479.4</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">ENST00000520116.1</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="left">ENST00000372874.9</td>
<td align="right">0.0000001</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">100</td>
<td align="left">ENST00000537820.1</td>
<td align="right">0.0000001</td>
<td align="right">0.0000006</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="left">ENST00000539235.5</td>
<td align="right">0.0000001</td>
<td align="right">0.0000015</td>
</tr>
<tr class="even">
<td align="left">1000</td>
<td align="left">ENST00000269141.8</td>
<td align="right">0.0346337</td>
<td align="right">0.0171931</td>
</tr>
<tr class="odd">
<td align="left">1000</td>
<td align="left">ENST00000399380.7</td>
<td align="right">0.0346337</td>
<td align="right">0.2700311</td>
</tr>
<tr class="even">
<td align="left">1000</td>
<td align="left">ENST00000418492.5</td>
<td align="right">0.0346337</td>
<td align="right">0.6179705</td>
</tr>
<tr class="odd">
<td align="left">1000</td>
<td align="left">ENST00000413878.1</td>
<td align="right">0.0346337</td>
<td align="right">0.6179705</td>
</tr>
</tbody>
</table>
<pre class="r"><code>select(org.Hs.eg.db, 
       keys = padj$geneID, 
       columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
       keytype = &quot;ENTREZID&quot;) %&gt;%
  dplyr::distinct() %&gt;% 
  dplyr::inner_join(padj, by = c(&quot;ENTREZID&quot; = &quot;geneID&quot;)) %&gt;%
  dplyr::arrange(gene, transcript) -&gt; top_gene</code></pre>
<pre><code>&#39;select()&#39; returned many:1 mapping between keys and columns</code></pre>
<pre class="r"><code>top_gene %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ENTREZID</th>
<th align="left">SYMBOL</th>
<th align="left">txID</th>
<th align="right">gene</th>
<th align="right">transcript</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">A1BG</td>
<td align="left">ENST00000598345.1</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">A1BG</td>
<td align="left">ENST00000599109.5</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">NAT2</td>
<td align="left">ENST00000286479.4</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">NAT2</td>
<td align="left">ENST00000520116.1</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="left">ADA</td>
<td align="left">ENST00000372874.9</td>
<td align="right">0.0000001</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">100</td>
<td align="left">ADA</td>
<td align="left">ENST00000537820.1</td>
<td align="right">0.0000001</td>
<td align="right">0.0000006</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="left">ADA</td>
<td align="left">ENST00000539235.5</td>
<td align="right">0.0000001</td>
<td align="right">0.0000015</td>
</tr>
<tr class="even">
<td align="left">1000</td>
<td align="left">CDH2</td>
<td align="left">ENST00000269141.8</td>
<td align="right">0.0346337</td>
<td align="right">0.0171931</td>
</tr>
<tr class="odd">
<td align="left">1000</td>
<td align="left">CDH2</td>
<td align="left">ENST00000399380.7</td>
<td align="right">0.0346337</td>
<td align="right">0.2700311</td>
</tr>
<tr class="even">
<td align="left">1000</td>
<td align="left">CDH2</td>
<td align="left">ENST00000418492.5</td>
<td align="right">0.0346337</td>
<td align="right">0.6179705</td>
</tr>
<tr class="odd">
<td align="left">1000</td>
<td align="left">CDH2</td>
<td align="left">ENST00000413878.1</td>
<td align="right">0.0346337</td>
<td align="right">0.6179705</td>
</tr>
</tbody>
</table>
<pre class="r"><code>group1 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;neg&quot;]
group2 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;pos&quot;]

plots &lt;- satuRn::plotDTU(
    object = sumExp,
    contrast = &quot;Contrast1&quot;,
    groups = list(group1, group2),
    coefficients = list(c(1, 0), 
                        c(0, 1)),
    summaryStat = &quot;model&quot;,
    genes = unique(padj$geneID)
)

wrap_plots(plots, ncol = 1) + plot_layout(guides = &quot;collect&quot;)</code></pre>
<pre><code>Warning: Removed 11 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 11 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 11 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 11 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 7 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 7 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 7 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BANDITS_1.6.0                           
 [2] stageR_1.12.0                           
 [3] DEXSeq_1.36.0                           
 [4] RColorBrewer_1.1-2                      
 [5] DESeq2_1.30.1                           
 [6] BiocParallel_1.24.1                     
 [7] SummarizedExperiment_1.20.0             
 [8] MatrixGenerics_1.2.1                    
 [9] matrixStats_0.59.0                      
[10] edgeR_3.32.1                            
[11] limma_3.46.0                            
[12] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
[13] GenomicFeatures_1.42.1                  
[14] GenomicRanges_1.42.0                    
[15] GenomeInfoDb_1.26.7                     
[16] org.Hs.eg.db_3.12.0                     
[17] AnnotationDbi_1.52.0                    
[18] IRanges_2.24.1                          
[19] S4Vectors_0.28.1                        
[20] satuRn_0.99.7                           
[21] tximport_1.18.0                         
[22] patchwork_1.1.1                         
[23] NMF_0.23.0                              
[24] Biobase_2.50.0                          
[25] BiocGenerics_0.36.1                     
[26] cluster_2.1.0                           
[27] rngtools_1.5                            
[28] pkgmaker_0.32.2                         
[29] registry_0.5-1                          
[30] forcats_0.5.1                           
[31] stringr_1.4.0                           
[32] dplyr_1.0.4                             
[33] purrr_0.3.4                             
[34] readr_1.4.0                             
[35] tidyr_1.1.2                             
[36] tibble_3.1.2                            
[37] ggplot2_3.3.5                           
[38] tidyverse_1.3.0                         
[39] here_1.0.1                              
[40] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.2.1          DRIMSeq_1.18.0          
  [4] BiocFileCache_1.14.0     plyr_1.8.6               splines_4.0.2           
  [7] gridBase_0.4-7           digest_0.6.27            foreach_1.5.1           
 [10] htmltools_0.5.1.1        fansi_0.5.0              magrittr_2.0.1          
 [13] memoise_2.0.0            doParallel_1.0.16        annotate_1.68.0         
 [16] Biostrings_2.58.0        modelr_0.1.8             askpass_1.1             
 [19] prettyunits_1.1.1        colorspace_2.0-2         blob_1.2.1              
 [22] rvest_0.3.6              rappdirs_0.3.3           haven_2.3.1             
 [25] xfun_0.23                crayon_1.4.1             RCurl_1.98-1.3          
 [28] jsonlite_1.7.2           genefilter_1.72.1        survival_3.2-7          
 [31] iterators_1.0.13         glue_1.4.2               gtable_0.3.0            
 [34] zlibbioc_1.36.0          XVector_0.30.0           DelayedArray_0.16.3     
 [37] scales_1.1.1             DBI_1.1.1                Rcpp_1.0.6              
 [40] xtable_1.8-4             progress_1.2.2           bit_4.0.4               
 [43] httr_1.4.2               ellipsis_0.3.2           farver_2.1.0            
 [46] pkgconfig_2.0.3          XML_3.99-0.5             dbplyr_2.1.0            
 [49] locfit_1.5-9.4           utf8_1.2.1               labeling_0.4.2          
 [52] tidyselect_1.1.0         rlang_0.4.11             reshape2_1.4.4          
 [55] later_1.1.0.1            munsell_0.5.0            cellranger_1.1.0        
 [58] tools_4.0.2              cachem_1.0.4             cli_3.0.0               
 [61] generics_0.1.0           RSQLite_2.2.5            broom_0.7.4             
 [64] evaluate_0.14            fastmap_1.1.0            yaml_2.2.1              
 [67] knitr_1.31               bit64_4.0.5              fs_1.5.0                
 [70] doRNG_1.8.2              pbapply_1.4-3            whisker_0.4             
 [73] xml2_1.3.2               biomaRt_2.46.3           compiler_4.0.2          
 [76] rstudioapi_0.13          curl_4.3                 reprex_1.0.0            
 [79] statmod_1.4.35           geneplotter_1.68.0       stringi_1.5.3           
 [82] highr_0.8                lattice_0.20-41          Matrix_1.3-2            
 [85] vctrs_0.3.8              pillar_1.6.1             lifecycle_1.0.0         
 [88] data.table_1.13.6        bitops_1.0-7             httpuv_1.5.5            
 [91] rtracklayer_1.50.0       hwriter_1.3.2            R6_2.5.0                
 [94] promises_1.2.0.1         codetools_0.2-18         MASS_7.3-53.1           
 [97] boot_1.3-26              assertthat_0.2.1         openssl_1.4.3           
[100] rprojroot_2.0.2          withr_2.4.2              GenomicAlignments_1.26.0
[103] Rsamtools_2.6.0          GenomeInfoDbData_1.2.4   locfdr_1.1-8            
[106] hms_1.0.0                grid_4.0.2               rmarkdown_2.6           
[109] git2r_0.28.0             lubridate_1.7.9.2       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
