<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Alternative splicing analysis</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Alternative splicing analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreeed601692a43bf148e82949815a8d24a6e438eab1targetblanked60169a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/ed601692a43bf148e82949815a8d24a6e438eab1" target="_blank">ed60169</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreeed601692a43bf148e82949815a8d24a6e438eab1targetblanked60169a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/ed601692a43bf148e82949815a8d24a6e438eab1" target="_blank">ed60169</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  .bpipe/
    Untracked:  analysis/STAR-diffSplice.Rmd
    Untracked:  analysis/saturn-dtu.Rmd
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-DRIMseq.Rmd</code>) and HTML (<code>docs/salmon-DRIMseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/ed601692a43bf148e82949815a8d24a6e438eab1/analysis/salmon-DRIMseq.Rmd" target="_blank">ed60169</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-14
</td>
<td>
wflow_publish(“analysis/salmon-DRIMseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/salmon-DRIMseq.html" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
</tbody>
</table>
<pre class="r"><code>int &lt;- intersect(names(files), targets$id)
targets &lt;- targets[match(int, targets$id),]
files &lt;- files[match(int, names(files))]</code></pre>
<pre class="r"><code>txi &lt;- tximport(files, type = &quot;salmon&quot;, txOut = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 </code></pre>
<pre class="r"><code>cts &lt;- txi$counts
cts &lt;- cts[rowSums(cts) &gt; 0,]
dim(cts)</code></pre>
<pre><code>[1] 155749     26</code></pre>
<pre class="r"><code>colSums(cts) %&gt;%
  data.frame %&gt;%
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  dplyr::rename(&quot;libsize&quot; = &quot;.&quot;) %&gt;%
ggplot(aes(x = sample, y = libsize)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-6-1.png" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Associate transcripts with gene IDs for gene-level summarization.</p>
<pre class="r"><code>library(TxDb.Hsapiens.UCSC.hg38.knownGene)
txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene
k &lt;- keys(txdb, keytype = &quot;TXNAME&quot;)
txdf &lt;- select(txdb, k, &quot;GENEID&quot;, &quot;TXNAME&quot;, &quot;SYMBOL&quot;)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>head(txdf) </code></pre>
<pre><code>             TXNAME    GENEID
1 ENST00000456328.2 100287102
2 ENST00000450305.2 100287102
3 ENST00000473358.1      &lt;NA&gt;
4 ENST00000469289.1      &lt;NA&gt;
5 ENST00000607096.1 100302278
6 ENST00000606857.1      &lt;NA&gt;</code></pre>
<pre class="r"><code>txdf %&gt;% inner_join(rownames_to_column(cts %&gt;% data.frame, var = &quot;TXNAME&quot;)) %&gt;%
  drop_na(GENEID) -&gt; cts</code></pre>
<pre><code>Joining, by = &quot;TXNAME&quot;</code></pre>
<pre class="r"><code>head(cts) </code></pre>
<pre><code>             TXNAME GENEID  CMV30  CMV31   CMV8   CMV9  CMV26  CMV14   CMV15
1 ENST00000326734.2 400728  0.000  0.000  0.000  0.000  0.000  0.000   0.000
2 ENST00000623808.3 643837  0.000  0.000  0.000  0.000  0.000  0.000   0.000
3 ENST00000445118.7 643837 81.080 49.139 51.049 35.215 40.649 71.627 112.693
4 ENST00000669922.1 643837  7.952  0.000  0.000  0.000  0.000  0.000   0.000
5 ENST00000670780.1 643837 15.481 22.471  0.000  0.000 24.094 14.058   0.000
6 ENST00000667414.1 643837  0.000  1.155  0.000  0.000  0.000  0.000   0.000
   CMV20  CMV21   CMV1   CMV2   CMV3   CMV4  CMV10  CMV11  CMV19  CMV35  CMV51
1  0.000  0.000  0.000  0.000  1.000  0.000  0.000  0.000  0.000  0.000  2.000
2  0.000  0.000  0.000  2.000  0.000  0.000  0.000  0.000  2.840  0.000  0.000
3 72.270 88.424 23.261 36.969  3.356 25.536 88.967 15.132  3.924 51.927 39.892
4 37.732  7.738  0.000  6.574  0.000  0.000  1.538  0.000 18.000  0.000 11.000
5  0.000  0.000  4.001  0.000 15.947  0.000  0.000  0.000  0.000  5.814  0.000
6  1.196  0.000  0.000  0.000  1.049  1.436  0.000  0.000  0.000  0.000  0.000
   CMV52  CMV53  CMV54  CMV56  CMV57   CMV58 CMV60  CMV61
1  0.000  0.000  0.000  0.000  0.000   0.000  0.00  0.000
2  0.000  0.000  1.113  0.000  0.000   0.000  0.00  0.000
3 80.193 45.610 48.656 85.781 95.585 155.849 89.77 52.703
4  3.808  0.000  0.000  0.000  0.000   0.000  0.00  0.000
5  0.000 16.986 11.345  0.000  1.836   0.000  6.21 20.954
6  0.000  0.000  0.000  0.000  0.000   0.000  0.00  0.000</code></pre>
<pre class="r"><code>cts %&gt;% dplyr::rename(gene_id = GENEID,
                      feature_id = TXNAME) -&gt; cts

targets %&gt;% dplyr::select(id, CMV_status, pair) %&gt;%
  dplyr::rename(sample_id = id,
                condition = CMV_status) %&gt;%
  data.frame -&gt; smps

library(DRIMSeq)</code></pre>
<pre><code>
Attaching package: &#39;DRIMSeq&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    samples</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    proportions</code></pre>
<pre class="r"><code>d &lt;- dmDSdata(counts = cts, samples = smps)
d</code></pre>
<pre><code>An object of class dmDSdata 
with 22802 genes and 26 samples
* data accessors: counts(), samples()</code></pre>
<pre class="r"><code>n &lt;- nrow(smps)
n.small &lt;- min(table(smps$condition))
d &lt;- dmFilter(d,
                min_samps_feature_expr=n.small, min_feature_expr=10,
                min_samps_feature_prop=n.small, min_feature_prop=0.1,
                min_samps_gene_expr=n, min_gene_expr=10)
d</code></pre>
<pre><code>An object of class dmDSdata 
with 5293 genes and 26 samples
* data accessors: counts(), samples()</code></pre>
<pre class="r"><code>table(table(counts(d)$gene_id))</code></pre>
<pre><code>
   2    3    4    5    6 
3694 1292  274   30    3 </code></pre>
<pre class="r"><code>design_full &lt;- model.matrix(~condition, data=DRIMSeq::samples(d))
colnames(design_full)</code></pre>
<pre><code>[1] &quot;(Intercept)&quot;  &quot;conditionneg&quot;</code></pre>
<pre class="r"><code>set.seed(1)
  d &lt;- dmPrecision(d, design=design_full)</code></pre>
<pre><code>! Using a subset of 0.1 genes to estimate common precision !</code></pre>
<pre><code>! Using common_precision = 4.7973 as prec_init !</code></pre>
<pre><code>! Using 0 as a shrinkage factor !</code></pre>
<pre class="r"><code>  d &lt;- dmFit(d, design=design_full)
  d &lt;- dmTest(d, coef=&quot;conditionneg&quot;)</code></pre>
<pre class="r"><code>res &lt;- DRIMSeq::results(d)
head(res)</code></pre>
<pre><code>  gene_id        lr df     pvalue adj_pvalue
1  148413 1.7745371  2 0.41177898  0.8602793
2   64856 0.2581608  1 0.61138651  0.9168039
3   83858 2.9058509  1 0.08825867  0.6416938
4  142678 3.0597604  2 0.21656161  0.7611292
5    8510 0.1119351  1 0.73795172  0.9510539
6    5590 0.9903457  1 0.31965789  0.8122656</code></pre>
<pre class="r"><code>res.txp &lt;- DRIMSeq::results(d, level=&quot;feature&quot;)
head(res.txp)</code></pre>
<pre><code>  gene_id        feature_id        lr df     pvalue adj_pvalue
1  148413 ENST00000448629.6 0.1166372  1 0.73271078  1.0000000
2  148413 ENST00000444362.2 1.5535841  1 0.21260711  0.8504284
3  148413 ENST00000418833.2 1.4309329  1 0.23161224  0.8582004
4   64856 ENST00000338660.5 0.2581608  1 0.61138651  0.9735692
5   64856 ENST00000476993.2 0.2581608  1 0.61138651  0.9735692
6   83858 ENST00000308647.8 2.9058509  1 0.08825867  0.7348017</code></pre>
<pre class="r"><code>no.na &lt;- function(x) ifelse(is.na(x), 1, x)
res$pvalue &lt;- no.na(res$pvalue)
res.txp$pvalue &lt;- no.na(res.txp$pvalue)</code></pre>
<pre class="r"><code>idx &lt;- which(res$adj_pvalue &lt; 0.05)
res[idx,]</code></pre>
<pre><code>     gene_id       lr df       pvalue adj_pvalue
702    84289 16.90512  1 3.929545e-05 0.02599885
1210    7037 15.41462  1 8.631791e-05 0.04568807
1656  153768 17.64185  1 2.666553e-05 0.02377087
2185  168850 19.15512  1 1.205136e-05 0.01599922
2569   54932 19.14888  1 1.209085e-05 0.01599922
2599    7587 15.84144  1 6.887743e-05 0.04050758
2618   23522 20.79346  1 5.115739e-06 0.01599922
3233  160518 23.34615  2 8.520151e-06 0.01599922
3523   80127 17.32890  1 3.143701e-05 0.02377087
3886    9961 20.95983  2 2.809517e-05 0.02377087
4381   10982 18.26108  2 1.083069e-04 0.04582531
4583    9138 14.77756  1 1.209666e-04 0.04582531
4722   27151 14.77380  1 1.212081e-04 0.04582531
5232   54880 18.26082  2 1.083212e-04 0.04582531</code></pre>
<pre class="r"><code>plotProportions(d, res$gene_id[idx][1], &quot;condition&quot;)</code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-18-1.png" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="stager-following-drimseq" class="section level1">
<h1>stageR following DRIMSeq</h1>
<p>We first construct a vector of p-values for the screening stage. Because of how the stageR package will combine transcript and gene names, we need to strip the gene and transcript version numbers from their Ensembl IDs (this is done by keeping only the first 15 characters of the gene and transcript IDs).</p>
<pre class="r"><code>pScreen &lt;- res$pvalue
strp &lt;- function(x) substr(x,1,15)
names(pScreen) &lt;- strp(res$gene_id)</code></pre>
<p>We construct a one column matrix of the confirmation p-values:</p>
<pre class="r"><code>pConfirmation &lt;- matrix(res.txp$pvalue, ncol=1)
rownames(pConfirmation) &lt;- strp(res.txp$feature_id)</code></pre>
<p>We arrange a two column data.frame with the transcript and gene identifiers.</p>
<pre class="r"><code>tx2gene &lt;- res.txp[,c(&quot;feature_id&quot;, &quot;gene_id&quot;)]
for (i in 1:2) tx2gene[,i] &lt;- strp(tx2gene[,i])</code></pre>
<p>The following functions then perform the stageR analysis. We must specify an alpha, which will be the overall false discovery rate target for the analysis, defined below. Unlike typical adjusted p-values or q-values, we cannot choose an arbitrary threshold later: after specifying alpha=0.05, we need to use 5% as the target in downstream steps. There are also convenience functions getSignificantGenes and getSignificantTx, which are demonstrated in the stageR vignette.</p>
<pre class="r"><code>library(stageR)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    count</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    rowMedians</code></pre>
<pre><code>
Attaching package: &#39;stageR&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:methods&#39;:

    getMethod</code></pre>
<pre class="r"><code>stageRObj &lt;- stageRTx(pScreen=pScreen, pConfirmation=pConfirmation,
                         pScreenAdjusted=FALSE, tx2gene=tx2gene)
stageRObj &lt;- stageWiseAdjustment(stageRObj, method=&quot;dtu&quot;, alpha=0.05) 
suppressWarnings({
  drim.padj &lt;- getAdjustedPValues(stageRObj, order=FALSE,
                                      onlySignificantGenes=TRUE)
})</code></pre>
<pre><code>The returned adjusted p-values are based on a stage-wise testing approach and are only valid for the provided target OFDR level of 5%. If a different target OFDR level is of interest,the entire adjustment should be re-run. </code></pre>
<pre class="r"><code>head(drim.padj)</code></pre>
<pre><code>  geneID            txID       gene transcript
1  84289 ENST00000636051 0.02599885          0
2  84289 ENST00000492488 0.02599885          0
3   7037 ENST00000360110 0.04568807          0
4   7037 ENST00000392396 0.04568807          0
5 153768 ENST00000334744 0.02377087          0
6 153768 ENST00000394450 0.02377087          0</code></pre>
<pre class="r"><code>select(org.Hs.eg.db, keys = drim.padj$geneID, columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
       keytype = &quot;ENTREZID&quot;) %&gt;%
  dplyr::distinct() %&gt;% 
  dplyr::inner_join(drim.padj, by = c(&quot;ENTREZID&quot; = &quot;geneID&quot;)) %&gt;%
  dplyr::arrange(gene, transcript) -&gt; top_gene</code></pre>
<pre><code>&#39;select()&#39; returned many:1 mapping between keys and columns</code></pre>
<pre class="r"><code>top_gene %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ENTREZID</th>
<th align="left">SYMBOL</th>
<th align="left">txID</th>
<th align="right">gene</th>
<th align="right">transcript</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">168850</td>
<td align="left">ZNF800</td>
<td align="left">ENST00000265827</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">168850</td>
<td align="left">ZNF800</td>
<td align="left">ENST00000393312</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">54932</td>
<td align="left">EXD3</td>
<td align="left">ENST00000340951</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">54932</td>
<td align="left">EXD3</td>
<td align="left">ENST00000465160</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">23522</td>
<td align="left">KAT6B</td>
<td align="left">ENST00000287239</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">23522</td>
<td align="left">KAT6B</td>
<td align="left">ENST00000372711</td>
<td align="right">0.0159992</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">160518</td>
<td align="left">DENND5B</td>
<td align="left">ENST00000536562</td>
<td align="right">0.0159992</td>
<td align="right">0.0004634</td>
</tr>
<tr class="even">
<td align="left">160518</td>
<td align="left">DENND5B</td>
<td align="left">ENST00000389082</td>
<td align="right">0.0159992</td>
<td align="right">0.6142600</td>
</tr>
<tr class="odd">
<td align="left">160518</td>
<td align="left">DENND5B</td>
<td align="left">ENST00000546299</td>
<td align="right">0.0159992</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">153768</td>
<td align="left">PRELID2</td>
<td align="left">ENST00000334744</td>
<td align="right">0.0237709</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">153768</td>
<td align="left">PRELID2</td>
<td align="left">ENST00000394450</td>
<td align="right">0.0237709</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000394009</td>
<td align="right">0.0237709</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000464394</td>
<td align="right">0.0237709</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000357402</td>
<td align="right">0.0237709</td>
<td align="right">0.1055928</td>
</tr>
<tr class="odd">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000395353</td>
<td align="right">0.0237709</td>
<td align="right">0.1241421</td>
</tr>
<tr class="even">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000563915</td>
<td align="right">0.0237709</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">84289</td>
<td align="left">ING5</td>
<td align="left">ENST00000636051</td>
<td align="right">0.0259989</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">84289</td>
<td align="left">ING5</td>
<td align="left">ENST00000492488</td>
<td align="right">0.0259989</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000351773</td>
<td align="right">0.0405076</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000361085</td>
<td align="right">0.0405076</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">7037</td>
<td align="left">TFRC</td>
<td align="left">ENST00000360110</td>
<td align="right">0.0456881</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">7037</td>
<td align="left">TFRC</td>
<td align="left">ENST00000392396</td>
<td align="right">0.0456881</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">9138</td>
<td align="left">ARHGEF1</td>
<td align="left">ENST00000378152</td>
<td align="right">0.0458253</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">9138</td>
<td align="left">ARHGEF1</td>
<td align="left">ENST00000337665</td>
<td align="right">0.0458253</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000443236</td>
<td align="right">0.0458253</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000594249</td>
<td align="right">0.0458253</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">54880</td>
<td align="left">BCOR</td>
<td align="left">ENST00000397354</td>
<td align="right">0.0458253</td>
<td align="right">0.0210870</td>
</tr>
<tr class="even">
<td align="left">10982</td>
<td align="left">MAPRE2</td>
<td align="left">ENST00000538170</td>
<td align="right">0.0458253</td>
<td align="right">0.0276653</td>
</tr>
<tr class="odd">
<td align="left">10982</td>
<td align="left">MAPRE2</td>
<td align="left">ENST00000436190</td>
<td align="right">0.0458253</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">10982</td>
<td align="left">MAPRE2</td>
<td align="left">ENST00000300249</td>
<td align="right">0.0458253</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">54880</td>
<td align="left">BCOR</td>
<td align="left">ENST00000378455</td>
<td align="right">0.0458253</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">54880</td>
<td align="left">BCOR</td>
<td align="left">ENST00000378444</td>
<td align="right">0.0458253</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>dtu_genes &lt;- unique(top_gene$ENTREZID) 
p &lt;- vector(&quot;list&quot;, length(dtu_genes))

for(i in 1:length(p)){
  p[[i]] &lt;- plotProportions(d, unique(top_gene$ENTREZID)[1], &quot;condition&quot;) +
    ggtitle(top_gene$SYMBOL[top_gene == dtu_genes[i]])
}

wrap_plots(p, ncol = 1)</code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-24-1.png" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gst &lt;- goana(dtu_genes, 
             universe = d@results_gene$gene_id)
topGO(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="11%" />
<col width="56%" />
<col width="4%" />
<col width="4%" />
<col width="3%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Term</th>
<th align="left">Ont</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0070776" class="uri">GO:0070776</a></td>
<td align="left">MOZ/MORF histone acetyltransferase complex</td>
<td align="left">CC</td>
<td align="right">8</td>
<td align="right">2</td>
<td align="right">0.0001804</td>
<td align="right">0.0036071</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0070775" class="uri">GO:0070775</a></td>
<td align="left">H3 histone acetyltransferase complex</td>
<td align="left">CC</td>
<td align="right">9</td>
<td align="right">2</td>
<td align="right">0.0002315</td>
<td align="right">0.0043992</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0140261" class="uri">GO:0140261</a></td>
<td align="left">BCOR complex</td>
<td align="left">CC</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000415" class="uri">GO:0000415</a></td>
<td align="left">negative regulation of histone H3-K36 methylation</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042483" class="uri">GO:0042483</a></td>
<td align="left">negative regulation of odontogenesis</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0070171" class="uri">GO:0070171</a></td>
<td align="left">negative regulation of tooth mineralization</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1990050" class="uri">GO:1990050</a></td>
<td align="left">phosphatidic acid transfer activity</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1905725" class="uri">GO:1905725</a></td>
<td align="left">protein localization to microtubule end</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1904825" class="uri">GO:1904825</a></td>
<td align="left">protein localization to microtubule plus-end</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0004998" class="uri">GO:0004998</a></td>
<td align="left">transferrin receptor activity</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0026455</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043966" class="uri">GO:0043966</a></td>
<td align="left">histone H3 acetylation</td>
<td align="left">BP</td>
<td align="right">33</td>
<td align="right">2</td>
<td align="right">0.0032748</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0018193" class="uri">GO:0018193</a></td>
<td align="left">peptidyl-amino acid modification</td>
<td align="left">BP</td>
<td align="right">427</td>
<td align="right">5</td>
<td align="right">0.0036245</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043967" class="uri">GO:0043967</a></td>
<td align="left">histone H4 acetylation</td>
<td align="left">BP</td>
<td align="right">38</td>
<td align="right">2</td>
<td align="right">0.0043274</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1990830" class="uri">GO:1990830</a></td>
<td align="left">cellular response to leukemia inhibitory factor</td>
<td align="left">BP</td>
<td align="right">39</td>
<td align="right">2</td>
<td align="right">0.0045544</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1990823" class="uri">GO:1990823</a></td>
<td align="left">response to leukemia inhibitory factor</td>
<td align="left">BP</td>
<td align="right">39</td>
<td align="right">2</td>
<td align="right">0.0045544</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051572" class="uri">GO:0051572</a></td>
<td align="left">negative regulation of histone H3-K4 methylation</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0052845</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0032014" class="uri">GO:0032014</a></td>
<td align="left">positive regulation of ARF protein signal transduction</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0052845</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0070170" class="uri">GO:0070170</a></td>
<td align="left">regulation of tooth mineralization</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0052845</td>
<td align="right">0.0476190</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0065001" class="uri">GO:0065001</a></td>
<td align="left">specification of axis polarity</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0052845</td>
<td align="right">0.0476190</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000123" class="uri">GO:0000123</a></td>
<td align="left">histone acetyltransferase complex</td>
<td align="left">CC</td>
<td align="right">48</td>
<td align="right">2</td>
<td align="right">0.0068395</td>
<td align="right">0.0476190</td>
</tr>
</tbody>
</table>
<pre class="r"><code>gst &lt;- kegga(dtu_genes, 
             universe = d@results_gene$gene_id)
topKEGG(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="15%" />
<col width="55%" />
<col width="4%" />
<col width="3%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Pathway</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">path:hsa04640</td>
<td align="left">Hematopoietic cell lineage</td>
<td align="right">9</td>
<td align="right">1</td>
<td align="right">0.0235723</td>
<td align="right">0.4714470</td>
</tr>
<tr class="even">
<td align="left">path:hsa04216</td>
<td align="left">Ferroptosis</td>
<td align="right">17</td>
<td align="right">1</td>
<td align="right">0.0440913</td>
<td align="right">0.8377353</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04145</td>
<td align="left">Phagosome</td>
<td align="right">28</td>
<td align="right">1</td>
<td align="right">0.0716517</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa04270</td>
<td align="left">Vascular smooth muscle contraction</td>
<td align="right">31</td>
<td align="right">1</td>
<td align="right">0.0790390</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04611</td>
<td align="left">Platelet activation</td>
<td align="right">32</td>
<td align="right">1</td>
<td align="right">0.0814893</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa04928</td>
<td align="left">Parathyroid hormone synthesis, secretion and action</td>
<td align="right">37</td>
<td align="right">1</td>
<td align="right">0.0936502</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04066</td>
<td align="left">HIF-1 signaling pathway</td>
<td align="right">45</td>
<td align="right">1</td>
<td align="right">0.1127972</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05135</td>
<td align="left">Yersinia infection</td>
<td align="right">52</td>
<td align="right">1</td>
<td align="right">0.1292419</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05417</td>
<td align="left">Lipid and atherosclerosis</td>
<td align="right">65</td>
<td align="right">1</td>
<td align="right">0.1590337</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05205</td>
<td align="left">Proteoglycans in cancer</td>
<td align="right">69</td>
<td align="right">1</td>
<td align="right">0.1680082</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04810</td>
<td align="left">Regulation of actin cytoskeleton</td>
<td align="right">71</td>
<td align="right">1</td>
<td align="right">0.1724620</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05130</td>
<td align="left">Pathogenic Escherichia coli infection</td>
<td align="right">72</td>
<td align="right">1</td>
<td align="right">0.1746806</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05163</td>
<td align="left">Human cytomegalovirus infection</td>
<td align="right">73</td>
<td align="right">1</td>
<td align="right">0.1768937</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa04144</td>
<td align="left">Endocytosis</td>
<td align="right">109</td>
<td align="right">1</td>
<td align="right">0.2529905</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05200</td>
<td align="left">Pathways in cancer</td>
<td align="right">148</td>
<td align="right">1</td>
<td align="right">0.3280242</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05168</td>
<td align="left">Herpes simplex virus 1 infection</td>
<td align="right">186</td>
<td align="right">1</td>
<td align="right">0.3943481</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00780</td>
<td align="left">Biotin metabolism</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa00471</td>
<td align="left">D-Glutamine and D-glutamate metabolism</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04950</td>
<td align="left">Maturity onset diabetes of the young</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05033</td>
<td align="left">Nicotine addiction</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] stageR_1.12.0                           
 [2] SummarizedExperiment_1.20.0             
 [3] MatrixGenerics_1.2.1                    
 [4] matrixStats_0.59.0                      
 [5] DRIMSeq_1.18.0                          
 [6] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
 [7] org.Hs.eg.db_3.12.0                     
 [8] tximport_1.18.0                         
 [9] patchwork_1.1.1                         
[10] NMF_0.23.0                              
[11] cluster_2.1.0                           
[12] rngtools_1.5                            
[13] pkgmaker_0.32.2                         
[14] registry_0.5-1                          
[15] edgeR_3.32.1                            
[16] limma_3.46.0                            
[17] EnsDb.Hsapiens.v86_2.99.0               
[18] ensembldb_2.14.0                        
[19] AnnotationFilter_1.14.0                 
[20] GenomicFeatures_1.42.1                  
[21] AnnotationDbi_1.52.0                    
[22] Biobase_2.50.0                          
[23] GenomicRanges_1.42.0                    
[24] GenomeInfoDb_1.26.7                     
[25] IRanges_2.24.1                          
[26] S4Vectors_0.28.1                        
[27] BiocGenerics_0.36.1                     
[28] forcats_0.5.1                           
[29] stringr_1.4.0                           
[30] dplyr_1.0.4                             
[31] purrr_0.3.4                             
[32] readr_1.4.0                             
[33] tidyr_1.1.2                             
[34] tibble_3.1.2                            
[35] ggplot2_3.3.5                           
[36] tidyverse_1.3.0                         
[37] here_1.0.1                              
[38] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.2.1          BiocFileCache_1.14.0    
  [4] plyr_1.8.6               lazyeval_0.2.2           BiocParallel_1.24.1     
  [7] gridBase_0.4-7           digest_0.6.27            foreach_1.5.1           
 [10] htmltools_0.5.1.1        GO.db_3.12.1             fansi_0.5.0             
 [13] magrittr_2.0.1           memoise_2.0.0            doParallel_1.0.16       
 [16] Biostrings_2.58.0        modelr_0.1.8             askpass_1.1             
 [19] prettyunits_1.1.1        colorspace_2.0-2         blob_1.2.1              
 [22] rvest_0.3.6              rappdirs_0.3.3           haven_2.3.1             
 [25] xfun_0.23                crayon_1.4.1             RCurl_1.98-1.3          
 [28] jsonlite_1.7.2           iterators_1.0.13         glue_1.4.2              
 [31] gtable_0.3.0             zlibbioc_1.36.0          XVector_0.30.0          
 [34] DelayedArray_0.16.3      scales_1.1.1             DBI_1.1.1               
 [37] Rcpp_1.0.6               xtable_1.8-4             progress_1.2.2          
 [40] bit_4.0.4                httr_1.4.2               RColorBrewer_1.1-2      
 [43] ellipsis_0.3.2           farver_2.1.0             pkgconfig_2.0.3         
 [46] XML_3.99-0.5             dbplyr_2.1.0             locfit_1.5-9.4          
 [49] utf8_1.2.1               tidyselect_1.1.0         labeling_0.4.2          
 [52] rlang_0.4.11             reshape2_1.4.4           later_1.1.0.1           
 [55] munsell_0.5.0            cellranger_1.1.0         tools_4.0.2             
 [58] cachem_1.0.4             cli_3.0.0                generics_0.1.0          
 [61] RSQLite_2.2.5            broom_0.7.4              evaluate_0.14           
 [64] fastmap_1.1.0            yaml_2.2.1               knitr_1.31              
 [67] bit64_4.0.5              fs_1.5.0                 whisker_0.4             
 [70] xml2_1.3.2               biomaRt_2.46.3           compiler_4.0.2          
 [73] rstudioapi_0.13          curl_4.3                 reprex_1.0.0            
 [76] stringi_1.5.3            highr_0.8                lattice_0.20-41         
 [79] ProtGenerics_1.22.0      Matrix_1.3-2             vctrs_0.3.8             
 [82] pillar_1.6.1             lifecycle_1.0.0          bitops_1.0-7            
 [85] httpuv_1.5.5             rtracklayer_1.50.0       R6_2.5.0                
 [88] promises_1.2.0.1         codetools_0.2-18         assertthat_0.2.1        
 [91] openssl_1.4.3            rprojroot_2.0.2          withr_2.4.2             
 [94] GenomicAlignments_1.26.0 Rsamtools_2.6.0          GenomeInfoDbData_1.2.4  
 [97] hms_1.0.0                grid_4.0.2               rmarkdown_2.6           
[100] git2r_0.28.0             lubridate_1.7.9.2       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
