<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Salmon quasi-mapping with RUVseq-edgeR analysis</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Salmon quasi-mapping with RUVseq-edgeR analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreee20c62d21447b7a1e31bb69f57e0e415e80bcf8atargetblanke20c62da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/e20c62d21447b7a1e31bb69f57e0e415e80bcf8a" target="_blank">e20c62d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreee20c62d21447b7a1e31bb69f57e0e415e80bcf8atargetblanke20c62da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/e20c62d21447b7a1e31bb69f57e0e415e80bcf8a" target="_blank">e20c62d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    analysis/obsolete_analysis/
    Ignored:    analysis/salmon-ruvseq-edger.nb.html
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    code/obsolete_analysis/
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_exons.csv
    Ignored:    data/HK_exons.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/rds/
    Ignored:    data/salmon-pilot-analysis/
    Ignored:    data/samples.csv
    Ignored:    data/star-genome-analysis/
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData

Untracked files:
    Untracked:  analysis/STAR-DEXSeq-exclude-US-ab.Rmd
    Untracked:  code/output.R
    Untracked:  code/rnaseq.salmon-quant.groovy
    Untracked:  code/rnaseq.star.count-exons.groovy
    Untracked:  code/rnaseq.star.groovy
    Untracked:  output/salmon-limma-voom-GO-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c2Cam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c5Cam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c5Cam.csv
    Untracked:  output/salmon-limma-voom-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-hCam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-keggCam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom.csv
    Untracked:  output/salmon-ruvseq-edger.csv
    Untracked:  output/star-fc-ruv-all-gsea-c2.csv
    Untracked:  output/star-fc-ruv-all-gsea-c5.csv
    Untracked:  output/star-fc-ruv-all.csv
    Untracked:  output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Untracked:  output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Untracked:  output/star-fc-ruv-no_us_ab.csv
    Untracked:  renv.lock

Unstaged changes:
    Modified:   .gitignore
    Deleted:    analysis/STAR-DEXSeq-Summed.Rmd
    Deleted:    analysis/salmon-BANDITS.Rmd
    Deleted:    analysis/salmon-DRIMseq.Rmd
    Deleted:    analysis/salmon-RUV-all.Rmd
    Deleted:    analysis/salmon-SatuRn.Rmd
    Deleted:    code/rnaseq-with-salmon-quant.groovy
    Deleted:    code/rnaseq.groovy
    Deleted:    code/salmon-quant-trimmed.groovy
    Deleted:    code/unmapped-pipe.groovy

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-ruvseq-edger.Rmd</code>) and HTML (<code>docs/salmon-ruvseq-edger.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/e20c62d21447b7a1e31bb69f57e0e415e80bcf8a/analysis/salmon-ruvseq-edger.Rmd" target="_blank">e20c62d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-25
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/salmon-limma-voom.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/2b6479ab40b7e78b559ce7d3912860d1423a98ca/analysis/salmon-ruvseq-edger.Rmd" target="_blank">2b6479a</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-26
</td>
<td>
Move/remove olds files.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/5a293f82d7268771d2cf4da55c4990a9f3d6b750/analysis/salmon-ruvseq-edger.Rmd" target="_blank">5a293f8</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
Old analyses.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tximport)
library(here)
library(tidyverse)
library(paletteer)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(EnsDb.Hsapiens.v86)
library(readr)
library(limma)
library(edgeR)
library(NMF)
library(EDASeq)
library(RUVSeq)

source(here(&quot;code/output.R&quot;))</code></pre>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>As with the <a href="salmon-limma-voom.html">limma-voom</a> analysis, the data was quasi-mapped and quantified using Salmon (v0.14.1) with the GENCODE (v31) human (hg38) transcriptome and decoy sequences. The mapping statistics can be found <a href="multiqc_report.html">here</a>. There appeared to be an issue with adapter contamination, with all samples having relatively low mapping percentages.</p>
<p>Load sample information and file names.</p>
<pre class="r"><code>samples &lt;- read.csv(here(&quot;data/samples.csv&quot;), stringsAsFactors = FALSE, header = TRUE)
files &lt;- list.files(here(&quot;data/salmon-pilot-analysis/190717_A00692_0021_AHLLHFDSXX/new_salmon/quants&quot;), 
                    recursive = TRUE, pattern = &quot;quant.sf&quot;, full.names = TRUE)
names(files) &lt;- strsplit2(files, &quot;_&quot;)[,10]

files &lt;- files[names(files) %in% samples$SampleId]
samples &lt;- samples[match(names(files), samples$SampleId),]

all(file.exists(files))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>metadat &lt;- read.csv(file = here(&quot;data/metadata.csv&quot;), header = TRUE, 
                    stringsAsFactors = FALSE)
joindat &lt;- read.csv(file = here(&quot;data/joindata.csv&quot;), header = TRUE, 
                    stringsAsFactors = FALSE)

metadat %&gt;% inner_join(joindat, by = c(&quot;Record.ID&quot; = &quot;UR&quot;)) %&gt;% 
  inner_join(samples, by = c(&quot;ID.post.extraction&quot; = &quot;SampleId&quot;)) -&gt; a
metadat %&gt;% inner_join(joindat, by = c(&quot;Tube.ID&quot; = &quot;UR&quot;)) %&gt;% 
  inner_join(samples, by = c(&quot;ID.post.extraction&quot; = &quot;SampleId&quot;)) -&gt; b

alldat &lt;- bind_rows(a,b)
alldat %&gt;% inner_join(samples, by = c(&quot;ID.post.extraction&quot; = &quot;SampleId&quot;)) -&gt; targets
targets &lt;- targets[match(samples$SampleId, targets$ID.post.extraction),]
targets$Sex &lt;- toupper(targets$Sex)</code></pre>
<p>Associate transcripts with gene IDs for gene-level summarization.</p>
<pre class="r"><code>edb &lt;- EnsDb.Hsapiens.v86
tx2gene &lt;- transcripts(edb, columns = c(&quot;tx_id&quot;, &quot;gene_id&quot;), return.type = &quot;DataFrame&quot;)</code></pre>
<p>Import gene-level counts and abundances.</p>
<pre class="r"><code>txiG &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene, ignoreTxVersion = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 
transcripts missing from tx2gene: 29910
summarizing abundance
summarizing counts
summarizing length</code></pre>
<pre class="r"><code>head(txiG$counts)</code></pre>
<pre><code>                CMV30   CMV31    CMV8    CMV9   CMV26 CMV27   CMV14   CMV15
ENSG00000000003   429 254.000 250.001 298.000 241.999   240 441.999 204.000
ENSG00000000005     0   0.000   5.000   0.000   1.000     0   6.000   1.000
ENSG00000000419    98 128.999 101.000 104.000 118.000   100 211.000 190.000
ENSG00000000457   794 515.000 367.001 321.999 501.000   405 523.000 419.006
ENSG00000000460   194 145.000 157.999  94.001 101.001    82  98.001 119.000
ENSG00000000938    20  10.000   4.000   2.000   2.000     0  10.000   3.000
                  CMV20   CMV21    CMV1 CMV2    CMV3    CMV4   CMV10   CMV11
ENSG00000000003 240.001 240.000 200.000  242 177.000 178.000 162.001  60.000
ENSG00000000005   0.000   0.000   0.000    0   0.000   0.000   0.000   0.000
ENSG00000000419 112.000 123.999 100.000  123  82.000  65.000 132.000  15.000
ENSG00000000457 504.999 355.000 319.001  300 251.999 237.001 280.000 159.999
ENSG00000000460 139.999 134.000  91.000   69  32.001  55.001  87.000  24.999
ENSG00000000938  69.000   4.000   7.000    0  17.000   7.000   7.000   0.000
                  CMV18   CMV19   CMV35
ENSG00000000003 475.000 439.999 178.000
ENSG00000000005   0.000   2.000   0.000
ENSG00000000419 138.000 124.000  78.999
ENSG00000000457 477.999 331.999 282.789
ENSG00000000460  87.000  92.000  55.000
ENSG00000000938   6.000   0.000  11.000</code></pre>
<p>Calculate offsets.</p>
<pre class="r"><code>ctsG &lt;- txiG$counts
normMat &lt;- txiG$length
normMat &lt;- normMat / exp(rowMeans(log(normMat)))
o &lt;- log(calcNormFactors(ctsG/normMat)) + log(colSums(ctsG/normMat))
z &lt;- DGEList(ctsG)
z &lt;- scaleOffset(z, t(t(log(normMat)) + o))</code></pre>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>Filter out lowly expressed genes as with the <a href="salmon-limma-voom.html">limma-voom</a> analysis.</p>
<pre class="r"><code>z$genes &lt;- ensembldb::genes(edb, filter = GeneIdFilter(rownames(z)), 
                 columns = c(&quot;gene_id&quot;, &quot;symbol&quot;, &quot;entrezid&quot;), return.type = &quot;DataFrame&quot;)
z$genes$entrezid &lt;- sapply(z$genes$entrezid, function(x) x[1])
z$genes$length &lt;- rowMedians(txiG$length)
z$group &lt;- ifelse(targets$TestResult.x == &quot;neg&quot;,&quot;neg&quot;,&quot;pos&quot;)

keep &lt;- filterByExpr(z, group = z$group)
y &lt;- z[keep,]
y</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                  CMV30   CMV31    CMV8    CMV9   CMV26   CMV27   CMV14   CMV15
ENSG00000000003 429.000 254.000 250.001 298.000 241.999 240.000 441.999 204.000
ENSG00000000419  98.000 128.999 101.000 104.000 118.000 100.000 211.000 190.000
ENSG00000000457 794.000 515.000 367.001 321.999 501.000 405.000 523.000 419.006
ENSG00000000460 194.000 145.000 157.999  94.001 101.001  82.000  98.001 119.000
ENSG00000000971  96.402 136.928 171.252 171.891  96.000  92.854 180.180 289.659
                  CMV20   CMV21    CMV1    CMV2    CMV3    CMV4   CMV10   CMV11
ENSG00000000003 240.001 240.000 200.000 242.000 177.000 178.000 162.001  60.000
ENSG00000000419 112.000 123.999 100.000 123.000  82.000  65.000 132.000  15.000
ENSG00000000457 504.999 355.000 319.001 300.000 251.999 237.001 280.000 159.999
ENSG00000000460 139.999 134.000  91.000  69.000  32.001  55.001  87.000  24.999
ENSG00000000971 165.633 122.309 141.339  87.812  38.622  38.448  97.371  45.159
                  CMV18   CMV19   CMV35
ENSG00000000003 475.000 439.999 178.000
ENSG00000000419 138.000 124.000  78.999
ENSG00000000457 477.999 331.999 282.789
ENSG00000000460  87.000  92.000  55.000
ENSG00000000971  47.101  99.917  85.000
14515 more rows ...

$samples
      group lib.size norm.factors
CMV30     1 12910571            1
CMV31     1  9882210            1
CMV8      1 10018352            1
CMV9      1  9653680            1
CMV26     1  8468332            1
14 more rows ...

$offset
                   CMV30    CMV31     CMV8     CMV9    CMV26    CMV27    CMV14
ENSG00000000003 16.54255 16.31769 15.92721 15.91117 16.10563 16.03860 16.34990
ENSG00000000419 16.49606 16.25633 15.94499 16.01463 15.97453 15.88564 16.41351
ENSG00000000457 16.29771 16.26280 16.10471 16.07727 16.19070 15.90347 16.35894
ENSG00000000460 16.81878 16.14598 16.50044 15.73052 15.60297 16.04805 16.44529
ENSG00000000971 16.64286 15.87519 16.25381 16.15640 15.86910 15.80674 16.57824
                   CMV15    CMV20    CMV21     CMV1     CMV2     CMV3     CMV4
ENSG00000000003 16.14024 16.08695 16.10387 15.95037 15.94678 15.86041 15.79804
ENSG00000000419 16.32179 16.15972 16.06037 15.85693 15.87109 15.76417 15.87444
ENSG00000000457 16.25268 16.09932 16.09221 16.02304 15.91295 15.73141 15.87941
ENSG00000000460 16.08153 15.85915 16.39924 16.26697 15.20041 16.36663 15.61240
ENSG00000000971 16.28544 16.17539 16.19212 15.84957 16.06087 15.95119 15.79255
                   CMV10    CMV11    CMV18    CMV19    CMV35
ENSG00000000003 15.89272 15.47791 16.05097 16.12641 15.83450
ENSG00000000419 16.05337 15.53544 16.18104 16.08051 15.71735
ENSG00000000457 16.12283 15.43731 15.98304 16.07545 15.65665
ENSG00000000460 16.56215 15.03883 16.38465 15.81182 15.58610
ENSG00000000971 15.85268 15.02966 15.96116 16.30960 15.81935
14515 more rows ...

$genes
DataFrame with 14520 rows and 4 columns
              gene_id        symbol  entrezid    length
          &lt;character&gt;   &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;
1     ENSG00000000003        TSPAN6      7105  1689.644
2     ENSG00000000419          DPM1      8813   807.898
3     ENSG00000000457         SCYL3     57147  3432.378
4     ENSG00000000460      C1orf112     55732  1817.522
5     ENSG00000000971           CFH      3075  3873.128
...               ...           ...       ...       ...
14516 ENSG00000283512 CTD-2547E10.6        NA   298.158
14517 ENSG00000283553    AL161905.1      8464   346.939
14518 ENSG00000283617    AC074141.1     84765  1894.318
14519 ENSG00000283620         ZNF48    197407  3021.887
14520 ENSG00000283633    AC005301.9        NA  1419.873

$group
 [1] &quot;pos&quot; &quot;neg&quot; &quot;neg&quot; &quot;pos&quot; &quot;pos&quot; &quot;neg&quot; &quot;neg&quot; &quot;pos&quot; &quot;pos&quot; &quot;neg&quot; &quot;pos&quot; &quot;neg&quot;
[13] &quot;neg&quot; &quot;pos&quot; &quot;neg&quot; &quot;pos&quot; &quot;neg&quot; &quot;pos&quot; &quot;neg&quot;</code></pre>
<p>The MDS plots are the same as for the <a href="salmon-limma-voom.html">limma-voom</a> analysis.</p>
<pre class="r"><code>pal &lt;- paletteer_d(&quot;LaCroixColoR::paired&quot;, 14)

colourBy &lt;- c(&quot;CMV_result&quot; = &quot;TestResult.x&quot;,&quot;GA_at_amnio&quot; = &quot;GA.at.amnio&quot;,&quot;Sex&quot; = &quot;Sex&quot;,
              &quot;Matched_pair&quot; = &quot;matchedpair&quot;, &quot;Site&quot; = &quot;Site.y&quot;)
lcpm &lt;- cpm(y, log = TRUE)

for(i in 1:length(colourBy)){
  
  par(mfrow=c(2,2), oma = c(0,0,2,0))
  plotMDS(lcpm, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])]) 
  legend(&quot;topleft&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.8)
  plotMDS(lcpm, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])], dim.plot = c(1,3))
  legend(&quot;right&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.8)
  plotMDS(lcpm, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])], dim.plot = c(2,3))
  legend(&quot;topright&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.8)
  plotMDS(lcpm, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])], dim.plot = c(3,4))
  legend(&quot;bottomleft&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.8)
  title(main = paste0(&quot;Coloured by &quot;, names(colourBy)[i]), outer = TRUE)
}</code></pre>
<p><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-8-2.png" width="864" style="display: block; margin: auto;" /><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-8-3.png" width="864" style="display: block; margin: auto;" /><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-8-4.png" width="864" style="display: block; margin: auto;" /><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-8-5.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="ruvseq-analysis" class="section level1">
<h1>RUVSeq analysis</h1>
<p>Given that there is a lot ov variability between the samples and it is not clearly associated with any of the known features of the data, we will use and <code>RUVSeq</code> analysis to remove some of the unwanted variation from the data. We will estimate the unwanted variation using the expression of known house-keeping genes (HKG), as defined using the Human BodyMap 2.0 <a href="https://www.tau.ac.il/~elieis/HKG/">data</a>, in our dataset.</p>
<p>The heatmap below shows the expression of the HKG in our samples. Although the expression is fairly uniform, some variability is apparent between the samples.</p>
<pre class="r"><code>hkGenes &lt;- read.delim(here(&quot;data/HK_genes.txt&quot;), stringsAsFactors = FALSE, 
                      header = FALSE, col.names = c(&quot;SYMBOL&quot;,&quot;REFSEQ&quot;), 
                      strip.white = TRUE)
y$genes &lt;- GenomicFeatures::genes(edb, filter = GeneIdFilter(rownames(y)), 
                 columns = c(&quot;gene_id&quot;, &quot;symbol&quot;, &quot;entrezid&quot;), 
                 return.type = &quot;DataFrame&quot;)

aheatmap(cpm(y[y$genes$symbol %in% hkGenes$SYMBOL,], log = TRUE),
         main = &quot;log2 CPM Expression of HKG&quot;, labRow = NA,  
         annCol = list(GA_at_amnio = targets$GA.at.amnio, 
                       CMV_result = targets$TestResult.x, 
                       Matched_pair = targets$matchedpair),
         scale = &quot;none&quot;)</code></pre>
<p><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Using the <code>RUVg</code> function, we will estimate the unwanted variation in our data using the HKG for k = 1 and k = 2.</p>
<pre class="r"><code>yRg1 &lt;- RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k=1)</code></pre>
<pre><code>Warning in RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k = 1): The expression matrix does not contain counts.
Please, pass a matrix of counts (not logged) or set isLog to TRUE to skip the log transformation</code></pre>
<pre class="r"><code>yRg2 &lt;- RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k=2)</code></pre>
<pre><code>Warning in RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k = 2): The expression matrix does not contain counts.
Please, pass a matrix of counts (not logged) or set isLog to TRUE to skip the log transformation</code></pre>
<p>The RLE plots below show that <code>RUVg</code> adjustment is a significant improvement over the raw data for both values of k.</p>
<pre class="r"><code>par(mfrow=c(1,3))
plotRLE(log2(y$counts + 2), ylim=c(-0.5,0.5), las=2, main=&quot;Raw&quot;)
plotRLE(yRg1$normalizedCounts, ylim=c(-0.5,0.5), las=2, main=&quot;RUVg k=1&quot;)</code></pre>
<pre><code>Warning in log(x + 1): NaNs produced</code></pre>
<pre class="r"><code>plotRLE(yRg2$normalizedCounts, ylim=c(-0.5,0.5), las=2, main=&quot;RUVg k=2&quot;)</code></pre>
<pre><code>Warning in log(x + 1): NaNs produced</code></pre>
<p><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-11-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>The MDS plots of the <code>RUVg</code> normalised counts show clearer separation between the CMV negative and CMV positive samples, particularly with k=2. However, the CMV3 “normal” sample continues to cluster with the CMV positive samples.</p>
<pre class="r"><code>i=1
par(mfrow=c(1,2))
  plotMDS(yRg1$normalizedCounts, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])]) 
  legend(&quot;bottomright&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.7)
  plotMDS(yRg2$normalizedCounts, gene.selection = &quot;common&quot;, 
          col = pal[factor(targets[,colourBy[i]])])
  legend(&quot;topleft&quot;, legend = levels(factor(targets[,colourBy[i]])), text.col = pal, 
         cex = 0.7)</code></pre>
<p><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-12-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<p>We will look for differentially expressed genes, using the negative binomial GLM approach implemented in <code>edgeR</code>. This is done by considering a design matrix that includes both the covariates of interest (CMV result) and the factors of unwanted variation calculated by <code>RUVg</code>. The CMV positive samples were compared to the CMV negative (normal) samples. A summary of the numbers of differentially expressed genes is shown below, as well as the top 20 differentially expressed genes. The full results tables were exported as csv files.</p>
<pre class="r"><code>design &lt;- model.matrix(~y$group + yRg2$W)
y &lt;- calcNormFactors(y)</code></pre>
<pre><code>Warning in calcNormFactors.DGEList(y): object contains offsets, which take precedence over library
sizes and norm factors (and which will not be recomputed).</code></pre>
<pre class="r"><code>y &lt;- estimateGLMCommonDisp(y, design)
y &lt;- estimateGLMTagwiseDisp(y, design)
fit &lt;- glmFit(y, design)
lrt &lt;- glmLRT(fit, coef=2)
topRE &lt;- topTags(lrt, n=100)$table
table(topRE$FDR &lt; 0.1)</code></pre>
<pre><code>
FALSE  TRUE 
   63    37 </code></pre>
<pre class="r"><code>write.csv(topRE, file = here(&quot;output/salmon-ruvseq-edger.csv&quot;))
head(as.data.frame(topRE), n = 20)</code></pre>
<pre><code>                        gene_id        symbol entrezid     logFC    logCPM
ENSG00000137959 ENSG00000137959        IFI44L    10964  2.863488 7.4121085
ENSG00000138646 ENSG00000138646         HERC5    51191  2.543027 2.6791812
ENSG00000138642 ENSG00000138642         HERC6    55008  1.895310 4.9941022
ENSG00000206337 ENSG00000206337          HCP5    10866  5.448222 2.7483610
ENSG00000206503 ENSG00000206503         HLA-A     3105  1.774618 5.3150397
ENSG00000240065 ENSG00000240065         PSMB9     5698  2.638340 1.7793430
ENSG00000234745 ENSG00000234745         HLA-B     3106  2.255485 5.6254056
ENSG00000137628 ENSG00000137628         DDX60    55601  1.613339 5.9448183
ENSG00000157601 ENSG00000157601           MX1     4599  1.950726 6.7805999
ENSG00000165949 ENSG00000165949         IFI27     3429  2.632822 5.2274739
ENSG00000107201 ENSG00000107201         DDX58    23586  1.285475 5.4324542
ENSG00000132530 ENSG00000132530          XAF1    54739  2.197606 5.2656988
ENSG00000115415 ENSG00000115415         STAT1     6772  1.147542 7.8061832
ENSG00000279296 ENSG00000279296 RP11-609D21.3       NA  2.405050 1.2491052
ENSG00000177409 ENSG00000177409        SAMD9L   219285  1.580570 4.9190395
ENSG00000137965 ENSG00000137965         IFI44    10561  1.906087 4.6386580
ENSG00000182871 ENSG00000182871       COL18A1    80781  1.966976 0.9720643
ENSG00000136514 ENSG00000136514          RTP4    64108  1.670539 1.5666840
ENSG00000180287 ENSG00000180287          PLD5   200150 -2.382127 1.7504029
ENSG00000204642 ENSG00000204642         HLA-F     3134  2.366739 2.2507458
                      LR       PValue          FDR
ENSG00000137959 40.90936 1.594557e-10 2.315297e-06
ENSG00000138646 36.50652 1.521564e-09 1.104655e-05
ENSG00000138642 32.78666 1.028482e-08 4.977854e-05
ENSG00000206337 30.56025 3.236580e-08 9.927456e-05
ENSG00000206503 30.45413 3.418546e-08 9.927456e-05
ENSG00000240065 30.06534 4.177309e-08 1.010909e-04
ENSG00000234745 29.60788 5.288942e-08 1.097078e-04
ENSG00000137628 28.29987 1.039028e-07 1.885836e-04
ENSG00000157601 27.06348 1.968826e-07 3.176373e-04
ENSG00000165949 25.63122 4.132983e-07 6.001091e-04
ENSG00000107201 25.13480 5.345930e-07 7.056628e-04
ENSG00000132530 24.67032 6.802379e-07 8.230878e-04
ENSG00000115415 24.32575 8.134456e-07 9.085561e-04
ENSG00000279296 22.57021 2.026016e-06 2.101268e-03
ENSG00000177409 20.55053 5.807773e-06 5.621924e-03
ENSG00000137965 19.77621 8.705954e-06 7.900653e-03
ENSG00000182871 19.64752 9.312427e-06 7.953908e-03
ENSG00000136514 19.44036 1.037905e-05 8.372432e-03
ENSG00000180287 18.70257 1.527762e-05 1.158137e-02
ENSG00000204642 18.62018 1.595231e-05 1.158137e-02</code></pre>
<p>The following plots show the expression of the top 9 ranked differentially expressed genes for CMV negative and CMV positive samples, using the <code>RUVg</code> normalised counts. Although there is significant variability within the groups and the log2 fold changes are not large, there are obvious differences in expression for the top ranked genes.</p>
<pre class="r"><code>lcpm&lt;- yRg2$normalizedCounts

par(mfrow=c(3,3), mar=c(3,4,4,2) + 0.1)
for(i in 1:9){
stripchart(lcpm[rownames(lcpm) == topRE$gene_id[i],] ~ y$group,
           vertical = TRUE, las = 2, cex.axis = 0.8, pch = 16, cex = 1.3,
           method = &quot;jitter&quot;, ylab = &quot;log2 CPM&quot;, main = topRE$symbol[i],
           col = pal[factor(y$group)])
}</code></pre>
<p><img src="figure/salmon-ruvseq-edger.Rmd/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-testing" class="section level1">
<h1>Gene set testing</h1>
<p>Testing for over-representation of gene ontology (GO) terms in the significantly differentially expressed genes (FDR &lt; 0.1) again highlights the reponse to interferon.</p>
<pre class="r"><code>go &lt;- goana(unlist(topRE$entrezid[topRE$FDR &lt; 0.1]), trend = &quot;length&quot;)
topGO(go)</code></pre>
<pre><code>                                                  Term Ont    N DE         P.DE
GO:0071357      cellular response to type I interferon  BP   95 13 9.109810e-23
GO:0060337         type I interferon signaling pathway  BP   95 13 9.109810e-23
GO:0034340               response to type I interferon  BP   99 13 1.609062e-22
GO:0051707                  response to other organism  BP 1591 24 3.032826e-20
GO:0043207        response to external biotic stimulus  BP 1593 24 3.123533e-20
GO:0009607                 response to biotic stimulus  BP 1625 24 4.979422e-20
GO:0051607                   defense response to virus  BP  258 14 1.021484e-18
GO:0009615                           response to virus  BP  349 15 1.484608e-18
GO:0098542          defense response to other organism  BP 1213 21 2.056318e-18
GO:0045087                      innate immune response  BP  996 19 3.382725e-17
GO:0044419  interspecies interaction between organisms  BP 2250 24 9.668112e-17
GO:0006952                            defense response  BP 1824 21 7.965926e-15
GO:0009605               response to external stimulus  BP 2976 24 5.803162e-14
GO:0019221         cytokine-mediated signaling pathway  BP  808 15 3.543369e-13
GO:0034341                response to interferon-gamma  BP  202 10 4.785502e-13
GO:0034097                        response to cytokine  BP 1227 17 5.868682e-13
GO:0006955                             immune response  BP 2306 21 8.365729e-13
GO:0060333 interferon-gamma-mediated signaling pathway  BP   91  8 1.296475e-12
GO:0002252                     immune effector process  BP 1299 17 1.472525e-12
GO:0071345      cellular response to cytokine stimulus  BP 1136 16 2.956439e-12</code></pre>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>RUVSeq analysis of the data reduced some of the variability present and resulted in 37 significantly differentially expressed genes (FDR &lt; 0.1) that were over-represented for GO terms associated with reponse to interferon, which was consistent with the previous <a href="salmon-limma-voom.html">limma-voom</a> analysis.</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2                      
 [2] RUVSeq_1.24.0                           
 [3] EDASeq_2.24.0                           
 [4] ShortRead_1.48.0                        
 [5] GenomicAlignments_1.26.0                
 [6] SummarizedExperiment_1.20.0             
 [7] MatrixGenerics_1.2.1                    
 [8] matrixStats_0.59.0                      
 [9] Rsamtools_2.6.0                         
[10] Biostrings_2.58.0                       
[11] XVector_0.30.0                          
[12] BiocParallel_1.24.1                     
[13] NMF_0.23.0                              
[14] cluster_2.1.0                           
[15] rngtools_1.5                            
[16] pkgmaker_0.32.2                         
[17] registry_0.5-1                          
[18] edgeR_3.32.1                            
[19] limma_3.46.0                            
[20] EnsDb.Hsapiens.v86_2.99.0               
[21] ensembldb_2.14.0                        
[22] AnnotationFilter_1.14.0                 
[23] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
[24] GenomicFeatures_1.42.1                  
[25] AnnotationDbi_1.52.0                    
[26] Biobase_2.50.0                          
[27] GenomicRanges_1.42.0                    
[28] GenomeInfoDb_1.26.7                     
[29] IRanges_2.24.1                          
[30] S4Vectors_0.28.1                        
[31] BiocGenerics_0.36.1                     
[32] paletteer_1.3.0                         
[33] forcats_0.5.1                           
[34] stringr_1.4.0                           
[35] dplyr_1.0.4                             
[36] purrr_0.3.4                             
[37] readr_1.4.0                             
[38] tidyr_1.1.2                             
[39] tibble_3.1.2                            
[40] ggplot2_3.3.5                           
[41] tidyverse_1.3.0                         
[42] here_1.0.1                              
[43] tximport_1.18.0                         
[44] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        aroma.light_3.20.0    
  [4] BiocFileCache_1.14.0   plyr_1.8.6             lazyeval_0.2.2        
  [7] gridBase_0.4-7         digest_0.6.27          foreach_1.5.1         
 [10] htmltools_0.5.1.1      GO.db_3.12.1           fansi_0.5.0           
 [13] magrittr_2.0.1         memoise_2.0.0.9000     doParallel_1.0.16     
 [16] modelr_0.1.8           R.utils_2.10.1         askpass_1.1           
 [19] prettyunits_1.1.1      jpeg_0.1-8.1           colorspace_2.0-2      
 [22] blob_1.2.1             rvest_0.3.6            rappdirs_0.3.3        
 [25] haven_2.3.1            xfun_0.23              prismatic_1.0.0       
 [28] crayon_1.4.1           RCurl_1.98-1.3         jsonlite_1.7.2        
 [31] iterators_1.0.13       glue_1.4.2             gtable_0.3.0          
 [34] zlibbioc_1.36.0        DelayedArray_0.16.3    scales_1.1.1          
 [37] DBI_1.1.1              Rcpp_1.0.6             xtable_1.8-4          
 [40] progress_1.2.2         bit_4.0.4              httr_1.4.2            
 [43] ellipsis_0.3.2         farver_2.1.0           R.methodsS3_1.8.1     
 [46] pkgconfig_2.0.3        XML_3.99-0.5           dbplyr_2.1.0          
 [49] locfit_1.5-9.4         utf8_1.2.1             tidyselect_1.1.0      
 [52] rlang_0.4.11           reshape2_1.4.4         later_1.1.0.1         
 [55] munsell_0.5.0          cellranger_1.1.0       tools_4.0.2           
 [58] cachem_1.0.4           cli_3.0.0              generics_0.1.0        
 [61] RSQLite_2.2.5          broom_0.7.4            evaluate_0.14         
 [64] fastmap_1.1.0          yaml_2.2.1             rematch2_2.1.2        
 [67] org.Hs.eg.db_3.12.0    knitr_1.31             bit64_4.0.5           
 [70] fs_1.5.0               whisker_0.4            R.oo_1.24.0           
 [73] xml2_1.3.2             biomaRt_2.46.3         compiler_4.0.2        
 [76] rstudioapi_0.13        png_0.1-7              curl_4.3              
 [79] reprex_1.0.0           stringi_1.5.3          highr_0.8             
 [82] lattice_0.20-41        ProtGenerics_1.22.0    Matrix_1.3-2          
 [85] vctrs_0.3.8            pillar_1.6.1           lifecycle_1.0.0       
 [88] bitops_1.0-7           httpuv_1.5.5           rtracklayer_1.50.0    
 [91] latticeExtra_0.6-29    hwriter_1.3.2          R6_2.5.0              
 [94] promises_1.2.0.1       codetools_0.2-18       MASS_7.3-53.1         
 [97] assertthat_0.2.1       openssl_1.4.3          rprojroot_2.0.2       
[100] withr_2.4.2            GenomeInfoDbData_1.2.4 hms_1.0.0             
[103] grid_4.0.2             rmarkdown_2.6          git2r_0.28.0          
[106] lubridate_1.7.9.2     </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2                      
 [2] RUVSeq_1.24.0                           
 [3] EDASeq_2.24.0                           
 [4] ShortRead_1.48.0                        
 [5] GenomicAlignments_1.26.0                
 [6] SummarizedExperiment_1.20.0             
 [7] MatrixGenerics_1.2.1                    
 [8] matrixStats_0.59.0                      
 [9] Rsamtools_2.6.0                         
[10] Biostrings_2.58.0                       
[11] XVector_0.30.0                          
[12] BiocParallel_1.24.1                     
[13] NMF_0.23.0                              
[14] cluster_2.1.0                           
[15] rngtools_1.5                            
[16] pkgmaker_0.32.2                         
[17] registry_0.5-1                          
[18] edgeR_3.32.1                            
[19] limma_3.46.0                            
[20] EnsDb.Hsapiens.v86_2.99.0               
[21] ensembldb_2.14.0                        
[22] AnnotationFilter_1.14.0                 
[23] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
[24] GenomicFeatures_1.42.1                  
[25] AnnotationDbi_1.52.0                    
[26] Biobase_2.50.0                          
[27] GenomicRanges_1.42.0                    
[28] GenomeInfoDb_1.26.7                     
[29] IRanges_2.24.1                          
[30] S4Vectors_0.28.1                        
[31] BiocGenerics_0.36.1                     
[32] paletteer_1.3.0                         
[33] forcats_0.5.1                           
[34] stringr_1.4.0                           
[35] dplyr_1.0.4                             
[36] purrr_0.3.4                             
[37] readr_1.4.0                             
[38] tidyr_1.1.2                             
[39] tibble_3.1.2                            
[40] ggplot2_3.3.5                           
[41] tidyverse_1.3.0                         
[42] here_1.0.1                              
[43] tximport_1.18.0                         
[44] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        aroma.light_3.20.0    
  [4] BiocFileCache_1.14.0   plyr_1.8.6             lazyeval_0.2.2        
  [7] gridBase_0.4-7         digest_0.6.27          foreach_1.5.1         
 [10] htmltools_0.5.1.1      GO.db_3.12.1           fansi_0.5.0           
 [13] magrittr_2.0.1         memoise_2.0.0.9000     doParallel_1.0.16     
 [16] modelr_0.1.8           R.utils_2.10.1         askpass_1.1           
 [19] prettyunits_1.1.1      jpeg_0.1-8.1           colorspace_2.0-2      
 [22] blob_1.2.1             rvest_0.3.6            rappdirs_0.3.3        
 [25] haven_2.3.1            xfun_0.23              prismatic_1.0.0       
 [28] crayon_1.4.1           RCurl_1.98-1.3         jsonlite_1.7.2        
 [31] iterators_1.0.13       glue_1.4.2             gtable_0.3.0          
 [34] zlibbioc_1.36.0        DelayedArray_0.16.3    scales_1.1.1          
 [37] DBI_1.1.1              Rcpp_1.0.6             xtable_1.8-4          
 [40] progress_1.2.2         bit_4.0.4              httr_1.4.2            
 [43] ellipsis_0.3.2         farver_2.1.0           R.methodsS3_1.8.1     
 [46] pkgconfig_2.0.3        XML_3.99-0.5           dbplyr_2.1.0          
 [49] locfit_1.5-9.4         utf8_1.2.1             tidyselect_1.1.0      
 [52] rlang_0.4.11           reshape2_1.4.4         later_1.1.0.1         
 [55] munsell_0.5.0          cellranger_1.1.0       tools_4.0.2           
 [58] cachem_1.0.4           cli_3.0.0              generics_0.1.0        
 [61] RSQLite_2.2.5          broom_0.7.4            evaluate_0.14         
 [64] fastmap_1.1.0          yaml_2.2.1             rematch2_2.1.2        
 [67] org.Hs.eg.db_3.12.0    knitr_1.31             bit64_4.0.5           
 [70] fs_1.5.0               whisker_0.4            R.oo_1.24.0           
 [73] xml2_1.3.2             biomaRt_2.46.3         compiler_4.0.2        
 [76] rstudioapi_0.13        png_0.1-7              curl_4.3              
 [79] reprex_1.0.0           stringi_1.5.3          highr_0.8             
 [82] lattice_0.20-41        ProtGenerics_1.22.0    Matrix_1.3-2          
 [85] vctrs_0.3.8            pillar_1.6.1           lifecycle_1.0.0       
 [88] bitops_1.0-7           httpuv_1.5.5           rtracklayer_1.50.0    
 [91] latticeExtra_0.6-29    hwriter_1.3.2          R6_2.5.0              
 [94] promises_1.2.0.1       codetools_0.2-18       MASS_7.3-53.1         
 [97] assertthat_0.2.1       openssl_1.4.3          rprojroot_2.0.2       
[100] withr_2.4.2            GenomeInfoDbData_1.2.4 hms_1.0.0             
[103] grid_4.0.2             rmarkdown_2.6          git2r_0.28.0          
[106] lubridate_1.7.9.2     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
