<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Alternative splicing analysis</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Alternative splicing analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree382671cd4607b48ba3f75da5f7c50b043e5fdcf5targetblank382671ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/382671cd4607b48ba3f75da5f7c50b043e5fdcf5" target="_blank">382671c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree382671cd4607b48ba3f75da5f7c50b043e5fdcf5targetblank382671ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/382671cd4607b48ba3f75da5f7c50b043e5fdcf5" target="_blank">382671c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  .bpipe/
    Untracked:  0.99.7.tar.gz
    Untracked:  analysis/STAR-diffSplice.Rmd
    Untracked:  analysis/salmon-DRIMseq.Rmd
    Untracked:  analysis/saturn-dtu.Rmd
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-FTEST-0.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-FTEST-1.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-FTEST-2.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-FTEST-3.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-FTEST-4.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-MERGE-TMP.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH00-BK000016.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH01-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH02-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH03-BK000016.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH04-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH05-BK000016.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH06-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH07-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH08-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH09-BK000016.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH10-BK000015.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH11-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000000.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000001.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000002.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000003.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000004.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000005.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000006.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000007.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000008.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000009.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000010.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000011.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000012.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000013.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000014.tmp
    Untracked:  analysis/temp-core-081656-0025B5042A3E.sam-TH12-BK000015.tmp
    Untracked:  renv.lock
    Untracked:  satuRn-0.99.7/
    Untracked:  satuRn_0.99.7.tar.gz

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-SatuRn.Rmd</code>) and HTML (<code>docs/salmon-SatuRn.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/382671cd4607b48ba3f75da5f7c50b043e5fdcf5/analysis/salmon-SatuRn.Rmd" target="_blank">382671c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-09
</td>
<td>
wflow_publish(“analysis/salmon-SatuRn.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
</tbody>
</table>
<pre class="r"><code>int &lt;- intersect(names(files), targets$id)
targets &lt;- targets[match(int, targets$id),]
files &lt;- files[match(int, names(files))]
files</code></pre>
<pre><code>                                                                                                                                                                                           CMV30 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV31 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV8 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV9 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV26 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV14 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV15 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV20 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV21 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV1 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV2 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV3 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV4 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV10 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV11 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV19 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV35 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV51 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV52 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV53 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV54 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV56 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV57 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV58 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV60 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV61 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; </code></pre>
<p>Import transcript information First, we need an object that links transcripts to their corresponding genes. We suggest using the BioConductor R packages AnnotationHub and ensembldb for this purpose.</p>
<pre class="r"><code>edb &lt;- EnsDb.Hsapiens.v86
txs &lt;- transcripts(edb) 
head(txs)</code></pre>
<pre><code>GRanges object with 6 ranges and 6 metadata columns:
                  seqnames      ranges strand |           tx_id
                     &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;
  ENST00000456328        1 11869-14409      + | ENST00000456328
  ENST00000450305        1 12010-13670      + | ENST00000450305
  ENST00000488147        1 14404-29570      - | ENST00000488147
  ENST00000619216        1 17369-17436      - | ENST00000619216
  ENST00000473358        1 29554-31097      + | ENST00000473358
  ENST00000469289        1 30267-31109      + | ENST00000469289
                              tx_biotype tx_cds_seq_start tx_cds_seq_end
                             &lt;character&gt;        &lt;integer&gt;      &lt;integer&gt;
  ENST00000456328   processed_transcript             &lt;NA&gt;           &lt;NA&gt;
  ENST00000450305 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt;
  ENST00000488147 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt;
  ENST00000619216                  miRNA             &lt;NA&gt;           &lt;NA&gt;
  ENST00000473358                lincRNA             &lt;NA&gt;           &lt;NA&gt;
  ENST00000469289                lincRNA             &lt;NA&gt;           &lt;NA&gt;
                          gene_id         tx_name
                      &lt;character&gt;     &lt;character&gt;
  ENST00000456328 ENSG00000223972 ENST00000456328
  ENST00000450305 ENSG00000223972 ENST00000450305
  ENST00000488147 ENSG00000227232 ENST00000488147
  ENST00000619216 ENSG00000278267 ENST00000619216
  ENST00000473358 ENSG00000243485 ENST00000473358
  ENST00000469289 ENSG00000243485 ENST00000469289
  -------
  seqinfo: 357 sequences from GRCh38 genome</code></pre>
<p>Data wrangling Next, we perform some data wrangling steps to get the data in a format that is suited for satuRn. First, we create a DataFrame or Matrix linking transcripts to their corresponding genes.</p>
<p>! Important: satuRn is implemented such that the columns with transcript identifiers is names isoform_id, while the column containing gene identifiers should be named gene_id. In addition, following chunk removes transcripts that are the only isoform expressed of a certain gene, as they cannot be used in a DTU analysis.</p>
<pre class="r"><code># Get the transcript information in correct format
txInfo &lt;- as.data.frame(matrix(data = NA, nrow = length(txs), ncol = 2))
colnames(txInfo) &lt;- c(&quot;isoform_id&quot;, &quot;gene_id&quot;)
txInfo$isoform_id &lt;- txs$tx_id
txInfo$gene_id &lt;- txs$gene_id
rownames(txInfo) &lt;- txInfo$isoform_id
head(txInfo)</code></pre>
<pre><code>                     isoform_id         gene_id
ENST00000456328 ENST00000456328 ENSG00000223972
ENST00000450305 ENST00000450305 ENSG00000223972
ENST00000488147 ENST00000488147 ENSG00000227232
ENST00000619216 ENST00000619216 ENSG00000278267
ENST00000473358 ENST00000473358 ENSG00000243485
ENST00000469289 ENST00000469289 ENSG00000243485</code></pre>
<p>Import transcript counts, match with gene IDs and remove transcripts that are the only isoform of a gene.</p>
<pre class="r"><code>txi &lt;- tximport(files, type = &quot;salmon&quot;, txOut = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 </code></pre>
<pre class="r"><code>cts &lt;- txi$counts

rownames(cts) &lt;- sub(&quot;\\..*&quot;, &quot;&quot;, rownames(cts)) 
txInfo &lt;- txInfo[txInfo$isoform_id %in% rownames(cts), ]
txInfo &lt;- subset(txInfo, 
                 duplicated(gene_id) | duplicated(gene_id, fromLast = TRUE))
cts &lt;- cts[which(rownames(cts) %in% txInfo$isoform_id), ]
dim(cts)</code></pre>
<pre><code>[1] 160351     26</code></pre>
<pre class="r"><code>colSums(cts) %&gt;%
  data.frame %&gt;%
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  dplyr::rename(&quot;libsize&quot; = &quot;.&quot;) %&gt;%
ggplot(aes(x = sample, y = libsize)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Filtering Here we perform some feature-level filtering. For this task, we adopt the filtering criterion that is implemented in the R package edgeR. Alternatively, one could adopt the dmFilter criterion from the DRIMSeq R package, which provides a more stringent filtering when both methods are run in default settings. After filtering, we again remove transcripts that are the only isoform expressed of a certain gene.</p>
<pre class="r"><code>filter_edgeR &lt;- filterByExpr(cts,
    design = NULL,
    group = targets$CMV_status,
) 

table(filter_edgeR)</code></pre>
<pre><code>filter_edgeR
 FALSE   TRUE 
134588  25763 </code></pre>
<pre class="r"><code>cts &lt;- cts[filter_edgeR, ]

# Update txInfo according to the filtering procedure
txInfo &lt;- txInfo[which(
  txInfo$isoform_id %in% rownames(cts)), ]

# remove txs that are the only isoform expressed within a gene (after filtering)
txInfo &lt;- subset(txInfo, 
                 duplicated(gene_id) | duplicated(gene_id, fromLast = TRUE))
cts &lt;- cts[which(rownames(cts) %in% txInfo$isoform_id), ]

# satuRn requires the transcripts in the rowData and 
# the transcripts in the count matrix to be in the same order.
txInfo &lt;- txInfo[match(rownames(cts), txInfo$isoform_id), ]
dim(txInfo)</code></pre>
<pre><code>[1] 20418     2</code></pre>
<p>Generate SummarizedExperiment All three main functions of satuRn require a SummarizedExperiment object as an input class. See the SummarizedExperiment vignette (Morgan Martin, n.d.) for more information on this object class.</p>
<p>Do not forget to include the design matrix formula (see above) to the SummarizedExperiment as indicated below. As such, the object contains all the information required for the downstream DTU analysis.</p>
<pre class="r"><code>sumExp &lt;- SummarizedExperiment::SummarizedExperiment(
    assays = list(counts = cts),
    colData = targets,
    rowData = txInfo
)

# specify design formula from colData
metadata(sumExp)$formula &lt;- ~ 0 + as.factor(colData(sumExp)$CMV_status) + as.factor(colData(sumExp)$pair)
sumExp</code></pre>
<pre><code>class: SummarizedExperiment 
dim: 20418 26 
metadata(1): formula
assays(1): counts
rownames(20418): ENST00000447500 ENST00000590817 ... ENST00000496011
  ENST00000479401
rowData names(2): isoform_id gene_id
colnames(26): CMV30 CMV31 ... CMV60 CMV61
colData names(6): id CMV_status ... GA_at_amnio indication</code></pre>
<p>Fit quasi-binomial generalized linear models models The fitDTU function of satuRn is used to model transcript usage in different groups of samples or cells. Here we adopt the default settings of the function. Without parallelized execution, this code runs for approximately 15 seconds on a 2018 macbook pro laptop.</p>
<pre class="r"><code>system.time({
sumExp &lt;- satuRn::fitDTU(
    object = sumExp,
    formula = ~ 0 + CMV_status + pair,
    parallel = FALSE,
    BPPARAM = BiocParallel::bpparam(),
    verbose = TRUE
)
})</code></pre>
<pre><code>Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge

Warning: glm.fit: algorithm did not converge</code></pre>
<pre><code>   user  system elapsed 
 36.578   0.972  37.554 </code></pre>
<p>Set up contrast matrix First, we set up a contrast matrix. This allows us to test for differential transcript usage between groups of interest. The group factor in this toy example contains three levels; (1) ALM.L5_IT_ALM_Tmem163_Dmrtb1, (2) ALM.L5_IT_ALM_Tnc, (3) VISp.L5_IT_VISp_Hsd11b1_Endou. Here we show to assess DTU between cells of the groups 1 and 3 and between cells of groups 2 and 3.</p>
<pre class="r"><code>group &lt;- as.factor(targets$CMV_status)
pair &lt;- as.factor(targets$pair)
design &lt;- model.matrix(~ 0 + group + pair) # construct design matrix
colnames(design)[1:2] &lt;- levels(group)

L &lt;- limma::makeContrasts(
    Contrast1 = neg - pos,
    levels = design
)
L # contrast matrix</code></pre>
<pre><code>         Contrasts
Levels    Contrast1
  neg             1
  pos            -1
  pairL2          0
  pairL3          0
  pairL4          0
  pairL5          0
  pairL6          0
  pairL7          0
  pairL8          0
  pairL9          0
  pairM1          0
  pairM2          0
  pairNC1         0
  pairNC2         0</code></pre>
<p>Perform the test Next we can perform differential usage testing using testDTU. We again adopt default settings. For more information on the parameter settings, please comsult the help file of the testDTU function.</p>
<pre class="r"><code>sumExp &lt;- satuRn::testDTU(
    object = sumExp,
    contrasts = L,
    plot = FALSE,
    sort = TRUE
)</code></pre>
<p>The test results are now saved into the rowData of our SummarizedExperiment object under the name fitDTUResult_ followed by the name of the contrast of interest (i.e. the column names of the contrast matrix). The results can be accessed as follows:</p>
<pre class="r"><code>head(rowData(sumExp)[[&quot;fitDTUResult_Contrast1&quot;]])</code></pre>
<pre><code>                 estimates         se       df         t         pval
ENST00000334744  0.7551322 0.14028475 12.88846  5.382853 0.0001285312
ENST00000395529  0.5284519 0.09881913 12.88846  5.347668 0.0001364873
ENST00000394450 -0.7551322 0.14028475 12.88846 -5.382853 0.0001285312
ENST00000393514  1.3760246 0.26381365 12.88846  5.215896 0.0001711735
ENST00000261332 -2.1028953 0.41375757 12.88846 -5.082433 0.0002158215
ENST00000339561  1.7050454 0.34349751 12.88846  4.963778 0.0002657436
                regular_FDR empirical_pval empirical_FDR
ENST00000334744   0.7544186   0.0004076967     0.9988623
ENST00000395529   0.7544186   0.0004291954     0.9988623
ENST00000394450   0.7544186   0.0004651809     0.9988623
ENST00000393514   0.7544186   0.0005209740     0.9988623
ENST00000261332   0.7544186   0.0007220416     0.9988623
ENST00000339561   0.7544186   0.0007592072     0.9988623</code></pre>
<p>Visualize DTU Finally, we may visualize the usage of select transcripts in select groups of interest.</p>
<pre class="r"><code>group1 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;neg&quot;]
group2 &lt;- rownames(colData(sumExp))[colData(sumExp)$CMV_status == 
                                      &quot;pos&quot;]

plots &lt;- satuRn::plotDTU(
    object = sumExp,
    contrast = &quot;Contrast1&quot;,
    groups = list(group1, group2),
    coefficients = list(c(1, rep(0, 13)), 
                        c(0, 1, rep(0,12))),
    summaryStat = &quot;model&quot;,
    top.n = 6
)

plots</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/salmon-SatuRn.Rmd/unnamed-chunk-16-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>Two-stage testing procedure satuRn returns transcript-level p-values for each of the specified contrasts. While we have shown that satuRn is able to adequately control the false discovery rate (FDR) at the transcript level (Gilis Jeroen 2021), (Van den Berge Koen 2017) argued that it is often desirable to control the FDR at the gene level. This boosts statistical power and eases downstream biological interpretation and validation, which typically occur at the gene level.</p>
<p>To this end, (Van den Berge Koen 2017) developed a testing procedure that is implemented in the BioConductor R package stageR. The procedure consists of two stages; a screening stage and a confirmation stage.</p>
<p>In the screening stage, gene-level FDR-adjusted p-values are computed, which aggregate the evidence for differential transcript usage over all transcripts within the gene. Only genes with an FDR below the desired nominal level are further considered in the second stage. In the confirmation stage, transcript-level p-values are adjusted for those genes, using a FWER-controlling method on the FDR-adjusted significance level.</p>
<p>In its current implementation, stageR can only perform stage-wise testing if only one contrast is of interest in a DTU setting. An analogous correction for the assessment of multiple contrasts for multiple transcripts per gene has not yet been implemented.</p>
<p>Below, we demonstrate how the transcript-level p-values for the first contrast as returned by satuRn can be post-processed using stageR. We rely on the perGeneQValue function:</p>
<pre class="r"><code># transcript level p-values from satuRn
pvals &lt;- rowData(sumExp)[[&quot;fitDTUResult_Contrast1&quot;]]$empirical_pval

# compute gene level q-values
geneID &lt;- factor(rowData(sumExp)$gene_id)
geneSplit &lt;- split(seq(along = geneID), geneID)
pGene &lt;- sapply(geneSplit, function(i) min(pvals[i]))
pGene[is.na(pGene)] &lt;- 1
theta &lt;- unique(sort(pGene))

# gene-level significance testing
q &lt;- DEXSeq:::perGeneQValueExact(pGene, theta, geneSplit) 
qScreen &lt;- rep(NA_real_, length(pGene))
qScreen &lt;- q[match(pGene, theta)]
qScreen &lt;- pmin(1, qScreen)
names(qScreen) &lt;- names(geneSplit)

# prepare stageR input
tx2gene &lt;- as.data.frame(rowData(sumExp)[c(&quot;isoform_id&quot;, &quot;gene_id&quot;)])
colnames(tx2gene) &lt;- c(&quot;transcript&quot;, &quot;gene&quot;)

pConfirmation &lt;- matrix(matrix(pvals),
    ncol = 1,
    dimnames = list(rownames(tx2gene), &quot;transcript&quot;)
)

# create a stageRTx object
stageRObj &lt;- stageR::stageRTx(
    pScreen = qScreen,
    pConfirmation = pConfirmation,
    pScreenAdjusted = TRUE,
    tx2gene = tx2gene
)

# perform the two-stage testing procedure
stageRObj &lt;- stageR::stageWiseAdjustment(
    object = stageRObj,
    method = &quot;dtu&quot;,
    alpha = 0.05,
    allowNA = TRUE
)

# retrieves the adjusted p-values from the stageRTx object
padj &lt;- stageR::getAdjustedPValues(stageRObj,
    order = TRUE,
    onlySignificantGenes = FALSE
)</code></pre>
<pre><code>The returned adjusted p-values are based on a stage-wise testing approach and are only valid for the provided target OFDR level of 5%. If a different target OFDR level is of interest,the entire adjustment should be re-run. </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] stageR_1.12.0               DEXSeq_1.36.0              
 [3] RColorBrewer_1.1-2          DESeq2_1.30.1              
 [5] BiocParallel_1.24.1         SummarizedExperiment_1.20.0
 [7] MatrixGenerics_1.2.1        matrixStats_0.59.0         
 [9] edgeR_3.32.1                limma_3.46.0               
[11] EnsDb.Hsapiens.v86_2.99.0   ensembldb_2.14.0           
[13] AnnotationFilter_1.14.0     GenomicFeatures_1.42.1     
[15] AnnotationDbi_1.52.0        GenomicRanges_1.42.0       
[17] GenomeInfoDb_1.26.7         IRanges_2.24.1             
[19] S4Vectors_0.28.1            AnnotationHub_2.22.0       
[21] BiocFileCache_1.14.0        dbplyr_2.1.0               
[23] satuRn_0.99.7               tximport_1.18.0            
[25] patchwork_1.1.1             NMF_0.23.0                 
[27] Biobase_2.50.0              BiocGenerics_0.36.1        
[29] cluster_2.1.0               rngtools_1.5               
[31] pkgmaker_0.32.2             registry_0.5-1             
[33] forcats_0.5.1               stringr_1.4.0              
[35] dplyr_1.0.4                 purrr_0.3.4                
[37] readr_1.4.0                 tidyr_1.1.2                
[39] tibble_3.1.2                ggplot2_3.3.5              
[41] tidyverse_1.3.0             here_1.0.1                 
[43] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] plyr_1.8.6                    lazyeval_0.2.2               
  [5] splines_4.0.2                 gridBase_0.4-7               
  [7] digest_0.6.27                 foreach_1.5.1                
  [9] htmltools_0.5.1.1             fansi_0.5.0                  
 [11] magrittr_2.0.1                memoise_2.0.0                
 [13] doParallel_1.0.16             annotate_1.68.0              
 [15] Biostrings_2.58.0             modelr_0.1.8                 
 [17] askpass_1.1                   prettyunits_1.1.1            
 [19] colorspace_2.0-2              blob_1.2.1                   
 [21] rvest_0.3.6                   rappdirs_0.3.3               
 [23] haven_2.3.1                   xfun_0.23                    
 [25] crayon_1.4.1                  RCurl_1.98-1.3               
 [27] jsonlite_1.7.2                genefilter_1.72.1            
 [29] survival_3.2-7                iterators_1.0.13             
 [31] glue_1.4.2                    gtable_0.3.0                 
 [33] zlibbioc_1.36.0               XVector_0.30.0               
 [35] DelayedArray_0.16.3           scales_1.1.1                 
 [37] DBI_1.1.1                     Rcpp_1.0.6                   
 [39] xtable_1.8-4                  progress_1.2.2               
 [41] bit_4.0.4                     httr_1.4.2                   
 [43] ellipsis_0.3.2                farver_2.1.0                 
 [45] pkgconfig_2.0.3               XML_3.99-0.5                 
 [47] locfit_1.5-9.4                utf8_1.2.1                   
 [49] labeling_0.4.2                tidyselect_1.1.0             
 [51] rlang_0.4.11                  reshape2_1.4.4               
 [53] later_1.1.0.1                 munsell_0.5.0                
 [55] BiocVersion_3.12.0            cellranger_1.1.0             
 [57] tools_4.0.2                   cachem_1.0.4                 
 [59] cli_3.0.0                     generics_0.1.0               
 [61] RSQLite_2.2.5                 broom_0.7.4                  
 [63] evaluate_0.14                 fastmap_1.1.0                
 [65] yaml_2.2.1                    knitr_1.31                   
 [67] bit64_4.0.5                   fs_1.5.0                     
 [69] pbapply_1.4-3                 whisker_0.4                  
 [71] mime_0.10                     xml2_1.3.2                   
 [73] biomaRt_2.46.3                compiler_4.0.2               
 [75] rstudioapi_0.13               curl_4.3                     
 [77] interactiveDisplayBase_1.28.0 reprex_1.0.0                 
 [79] statmod_1.4.35                geneplotter_1.68.0           
 [81] stringi_1.5.3                 highr_0.8                    
 [83] lattice_0.20-41               ProtGenerics_1.22.0          
 [85] Matrix_1.3-2                  vctrs_0.3.8                  
 [87] pillar_1.6.1                  lifecycle_1.0.0              
 [89] BiocManager_1.30.10           bitops_1.0-7                 
 [91] rtracklayer_1.50.0            httpuv_1.5.5                 
 [93] hwriter_1.3.2                 R6_2.5.0                     
 [95] promises_1.2.0.1              codetools_0.2-18             
 [97] boot_1.3-26                   assertthat_0.2.1             
 [99] openssl_1.4.3                 rprojroot_2.0.2              
[101] withr_2.4.2                   GenomicAlignments_1.26.0     
[103] Rsamtools_2.6.0               GenomeInfoDbData_1.2.4       
[105] locfdr_1.1-8                  hms_1.0.0                    
[107] grid_4.0.2                    rmarkdown_2.6                
[109] git2r_0.28.0                  shiny_1.6.0                  
[111] lubridate_1.7.9.2            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
