<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Limma-voom paired analysis of ALL samples</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Limma-voom paired analysis of ALL samples</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreecccf312d4bec46ee08bb315ec42ac751092a165dtargetblankcccf312a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/cccf312d4bec46ee08bb315ec42ac751092a165d" target="_blank">cccf312</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreecccf312d4bec46ee08bb315ec42ac751092a165dtargetblankcccf312a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/cccf312d4bec46ee08bb315ec42ac751092a165d" target="_blank">cccf312</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/rds/
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruv-all-gsea-c2.csv
    Ignored:    output/salmon-ruv-all-gsea-c5.csv
    Ignored:    output/salmon-ruv-all.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  analysis/old_analysis/
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/salmon-SatuRn.Rmd
    Deleted:    analysis/salmon-bandits.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-RUV-all.Rmd</code>) and HTML (<code>docs/salmon-RUV-all.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/cccf312d4bec46ee08bb315ec42ac751092a165d/analysis/salmon-RUV-all.Rmd" target="_blank">cccf312</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-26
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/salmon-RUV-all.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>A total of 28 amniotic fluid cfRNA samples were sequenced (150bp PE ). Due to the degraded nature of the cfRNA, most of the read pairs showed considerable overlap. There was also significant adapter contamination. Adapter were trimmed and the reads were quality trimmed and filtered using <strong>fastp (0.20.1)</strong>. The reads were then aligned and quantified using <strong>Salmon (1.3.0)</strong>. The <strong>MultiQC (1.8)</strong> summary of the read quality control and alignment can be view <a href="multiqc_report.html">here</a>.</p>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Get file names for <strong>Salmon</strong> quantification output.</p>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
</tbody>
</table>
<p>Select only paired samples with adequate clinical information for downstream analysis.</p>
<pre class="r"><code>int &lt;- intersect(names(files), targets$id)
targets &lt;- targets[match(int, targets$id),]
files &lt;- files[match(int, names(files))]
targets$batch &lt;- factor(strsplit2(files, &quot;_&quot;)[,2], labels = 1:2)
targets %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
<th align="left">batch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
<td align="left">2</td>
</tr>
</tbody>
</table>
<p>Get gene and transcript information.</p>
<pre><code>DataFrame with 216741 rows and 2 columns
                 tx_id         gene_id
           &lt;character&gt;     &lt;character&gt;
1      ENST00000000233 ENSG00000004059
2      ENST00000000412 ENSG00000003056
3      ENST00000000442 ENSG00000173153
4      ENST00000001008 ENSG00000004478
5      ENST00000001146 ENSG00000003137
...                ...             ...
216737        LRG_96t1          LRG_96
216738        LRG_97t1          LRG_97
216739        LRG_98t1          LRG_98
216740       LRG_992t1         LRG_992
216741        LRG_99t1          LRG_99</code></pre>
<p>Import gene-level counts and abundances.</p>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 
transcripts missing from tx2gene: 31281
summarizing abundance
summarizing counts
summarizing length</code></pre>
<table style="width:100%;">
<colgroup>
<col width="5%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="4%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CMV30</th>
<th align="right">CMV31</th>
<th align="right">CMV8</th>
<th align="right">CMV9</th>
<th align="right">CMV26</th>
<th align="right">CMV14</th>
<th align="right">CMV15</th>
<th align="right">CMV20</th>
<th align="right">CMV21</th>
<th align="right">CMV1</th>
<th align="right">CMV2</th>
<th align="right">CMV3</th>
<th align="right">CMV4</th>
<th align="right">CMV10</th>
<th align="right">CMV11</th>
<th align="right">CMV19</th>
<th align="right">CMV35</th>
<th align="right">CMV51</th>
<th align="right">CMV52</th>
<th align="right">CMV53</th>
<th align="right">CMV54</th>
<th align="right">CMV56</th>
<th align="right">CMV57</th>
<th align="right">CMV58</th>
<th align="right">CMV60</th>
<th align="right">CMV61</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003</td>
<td align="right">351.24904</td>
<td align="right">215.47855</td>
<td align="right">400.569364</td>
<td align="right">463.664090</td>
<td align="right">291.617270</td>
<td align="right">517.93828</td>
<td align="right">286.237271</td>
<td align="right">247.64989</td>
<td align="right">331.090235</td>
<td align="right">237.40134</td>
<td align="right">236.5757</td>
<td align="right">153.77526</td>
<td align="right">272.093888</td>
<td align="right">242.67002</td>
<td align="right">66.435707</td>
<td align="right">580.044036</td>
<td align="right">184.54126</td>
<td align="right">375.780680</td>
<td align="right">156.17149</td>
<td align="right">593.638661</td>
<td align="right">414.0123</td>
<td align="right">455.437951</td>
<td align="right">544.679577</td>
<td align="right">424.2379</td>
<td align="right">278.0470012</td>
<td align="right">272.546693</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">3.105802</td>
<td align="right">0.000000</td>
<td align="right">1.891432</td>
<td align="right">12.55298</td>
<td align="right">1.844773</td>
<td align="right">0.00000</td>
<td align="right">0.000000</td>
<td align="right">0.00000</td>
<td align="right">0.0000</td>
<td align="right">0.00000</td>
<td align="right">0.000000</td>
<td align="right">0.00000</td>
<td align="right">2.317651</td>
<td align="right">1.251864</td>
<td align="right">0.00000</td>
<td align="right">0.000000</td>
<td align="right">0.00000</td>
<td align="right">1.820226</td>
<td align="right">0.0000</td>
<td align="right">0.000000</td>
<td align="right">0.000000</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419</td>
<td align="right">108.63348</td>
<td align="right">146.95853</td>
<td align="right">141.102056</td>
<td align="right">133.606648</td>
<td align="right">147.259185</td>
<td align="right">224.91166</td>
<td align="right">207.341575</td>
<td align="right">139.48264</td>
<td align="right">164.499226</td>
<td align="right">116.13259</td>
<td align="right">158.0801</td>
<td align="right">91.04703</td>
<td align="right">83.424213</td>
<td align="right">143.02970</td>
<td align="right">24.064450</td>
<td align="right">128.448849</td>
<td align="right">95.69470</td>
<td align="right">142.606856</td>
<td align="right">119.75518</td>
<td align="right">155.686491</td>
<td align="right">142.2288</td>
<td align="right">236.574036</td>
<td align="right">187.065337</td>
<td align="right">189.0048</td>
<td align="right">218.6861019</td>
<td align="right">67.422475</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000457</td>
<td align="right">1037.82768</td>
<td align="right">602.04979</td>
<td align="right">457.562351</td>
<td align="right">435.564805</td>
<td align="right">551.057510</td>
<td align="right">581.58874</td>
<td align="right">524.433334</td>
<td align="right">625.88743</td>
<td align="right">483.680976</td>
<td align="right">350.12584</td>
<td align="right">310.0534</td>
<td align="right">290.20409</td>
<td align="right">327.928063</td>
<td align="right">360.60862</td>
<td align="right">181.961437</td>
<td align="right">463.863077</td>
<td align="right">367.57280</td>
<td align="right">615.260249</td>
<td align="right">234.18632</td>
<td align="right">378.767591</td>
<td align="right">645.9281</td>
<td align="right">852.742720</td>
<td align="right">578.638345</td>
<td align="right">1081.7101</td>
<td align="right">318.3343798</td>
<td align="right">438.913930</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000460</td>
<td align="right">161.34445</td>
<td align="right">194.18409</td>
<td align="right">123.620807</td>
<td align="right">158.016381</td>
<td align="right">181.365755</td>
<td align="right">122.64404</td>
<td align="right">168.124240</td>
<td align="right">210.56054</td>
<td align="right">112.496678</td>
<td align="right">94.36101</td>
<td align="right">173.6230</td>
<td align="right">50.28069</td>
<td align="right">92.437809</td>
<td align="right">65.18898</td>
<td align="right">33.311858</td>
<td align="right">157.864419</td>
<td align="right">58.78816</td>
<td align="right">105.980425</td>
<td align="right">94.78877</td>
<td align="right">105.073735</td>
<td align="right">101.4636</td>
<td align="right">112.049466</td>
<td align="right">64.166866</td>
<td align="right">272.3186</td>
<td align="right">106.4651811</td>
<td align="right">84.707930</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000938</td>
<td align="right">19.80796</td>
<td align="right">10.08111</td>
<td align="right">3.802534</td>
<td align="right">4.130139</td>
<td align="right">5.844901</td>
<td align="right">13.13771</td>
<td align="right">4.901726</td>
<td align="right">91.29149</td>
<td align="right">4.967891</td>
<td align="right">13.94746</td>
<td align="right">0.0000</td>
<td align="right">12.66172</td>
<td align="right">7.828971</td>
<td align="right">11.30756</td>
<td align="right">9.060634</td>
<td align="right">2.239946</td>
<td align="right">14.87506</td>
<td align="right">5.736005</td>
<td align="right">22.87352</td>
<td align="right">2.122864</td>
<td align="right">0.0000</td>
<td align="right">3.351409</td>
<td align="right">7.705324</td>
<td align="right">28.1080</td>
<td align="right">0.9679327</td>
<td align="right">9.189123</td>
</tr>
</tbody>
</table>
<p>Set up <code>DGElist</code> object for downstream analysis.</p>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                    CMV30    CMV31       CMV8     CMV9      CMV26     CMV14
ENSG00000000003  351.2490 215.4786 400.569364 463.6641 291.617270 517.93828
ENSG00000000005    0.0000   0.0000   3.105802   0.0000   1.891432  12.55298
ENSG00000000419  108.6335 146.9585 141.102056 133.6066 147.259185 224.91166
ENSG00000000457 1037.8277 602.0498 457.562351 435.5648 551.057510 581.58874
ENSG00000000460  161.3444 194.1841 123.620807 158.0164 181.365755 122.64404
                     CMV15    CMV20    CMV21      CMV1     CMV2      CMV3
ENSG00000000003 286.237271 247.6499 331.0902 237.40134 236.5757 153.77526
ENSG00000000005   1.844773   0.0000   0.0000   0.00000   0.0000   0.00000
ENSG00000000419 207.341575 139.4826 164.4992 116.13259 158.0801  91.04703
ENSG00000000457 524.433334 625.8874 483.6810 350.12584 310.0534 290.20409
ENSG00000000460 168.124240 210.5605 112.4967  94.36101 173.6230  50.28069
                     CMV4     CMV10      CMV11      CMV19     CMV35    CMV51
ENSG00000000003 272.09389 242.67002  66.435707 580.044036 184.54126 375.7807
ENSG00000000005   0.00000   0.00000   2.317651   1.251864   0.00000   0.0000
ENSG00000000419  83.42421 143.02970  24.064450 128.448849  95.69470 142.6069
ENSG00000000457 327.92806 360.60862 181.961437 463.863077 367.57280 615.2602
ENSG00000000460  92.43781  65.18898  33.311858 157.864419  58.78816 105.9804
                    CMV52      CMV53    CMV54    CMV56     CMV57     CMV58
ENSG00000000003 156.17149 593.638661 414.0123 455.4380 544.67958  424.2379
ENSG00000000005   0.00000   1.820226   0.0000   0.0000   0.00000    0.0000
ENSG00000000419 119.75518 155.686491 142.2288 236.5740 187.06534  189.0048
ENSG00000000457 234.18632 378.767591 645.9281 852.7427 578.63835 1081.7101
ENSG00000000460  94.78877 105.073736 101.4636 112.0495  64.16687  272.3186
                   CMV60     CMV61
ENSG00000000003 278.0470 272.54669
ENSG00000000005   0.0000   0.00000
ENSG00000000419 218.6861  67.42247
ENSG00000000457 318.3344 438.91393
ENSG00000000460 106.4652  84.70793
57235 more rows ...

$samples
      group lib.size norm.factors
CMV30     1 14533210            1
CMV31     1 12415440            1
CMV8      1 11399505            1
CMV9      1 11940053            1
CMV26     1 10869683            1
21 more rows ...

$genes
DataFrame with 57240 rows and 4 columns
              gene_id         symbol  entrezid    length
          &lt;character&gt;    &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;
1     ENSG00000000003         TSPAN6      7105  2387.631
2     ENSG00000000005           TNMD     64102   706.665
3     ENSG00000000419           DPM1      8813   905.483
4     ENSG00000000457          SCYL3     57147  3387.525
5     ENSG00000000460       C1orf112     55732  2128.460
...               ...            ...       ...       ...
57236 ENSG00000283695  CTD-2331H12.9        NA   14.4615
57237 ENSG00000283696 RP11-122G18.12        NA  819.7439
57238 ENSG00000283697   RP5-937E21.8 101928917 1177.1967
57239 ENSG00000283698  RP11-606M12.1        NA  381.1154
57240 ENSG00000283699        MIR4481 100616320   18.0385

$group
[1] &quot;pos&quot; &quot;neg&quot; &quot;neg&quot; &quot;pos&quot; &quot;pos&quot;
21 more elements ...</code></pre>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>Genes that do not have an adequate number of reads in any sample should be filtered out prior to downstream analyses. From a biological perspective, genes that are not expressed at a biologically meaningful level in any condition are not of interest. Statistically, we get a better estimate of the mean-variance relationship in the data and reduce the number of statistical tests that are performed during differential expression analyses.</p>
<p>Filter out lowly expressed genes and genes without Entrez IDs and calculate TMM normalization factors.</p>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                     CMV30    CMV31     CMV8     CMV9    CMV26    CMV14
ENSG00000000003  351.24904 215.4786 400.5694 463.6641 291.6173 517.9383
ENSG00000000419  108.63348 146.9585 141.1021 133.6066 147.2592 224.9117
ENSG00000000457 1037.82768 602.0498 457.5624 435.5648 551.0575 581.5887
ENSG00000000460  161.34445 194.1841 123.6208 158.0164 181.3658 122.6440
ENSG00000000971   91.76282 151.6092 212.2621 229.8054 106.4742 203.3716
                   CMV15    CMV20    CMV21      CMV1     CMV2      CMV3
ENSG00000000003 286.2373 247.6499 331.0902 237.40134 236.5757 153.77526
ENSG00000000419 207.3416 139.4826 164.4992 116.13259 158.0801  91.04703
ENSG00000000457 524.4333 625.8874 483.6810 350.12584 310.0534 290.20409
ENSG00000000460 168.1242 210.5605 112.4967  94.36101 173.6230  50.28069
ENSG00000000971 343.2359 191.7684 183.3917 189.24105 117.4170  46.69618
                     CMV4     CMV10     CMV11    CMV19     CMV35    CMV51
ENSG00000000003 272.09389 242.67002  66.43571 580.0440 184.54126 375.7807
ENSG00000000419  83.42421 143.02970  24.06445 128.4488  95.69470 142.6069
ENSG00000000457 327.92806 360.60862 181.96144 463.8631 367.57280 615.2602
ENSG00000000460  92.43781  65.18898  33.31186 157.8644  58.78816 105.9804
ENSG00000000971  57.43046 115.75535  75.66333 111.4139 107.56176 233.0117
                    CMV52     CMV53     CMV54    CMV56     CMV57     CMV58
ENSG00000000003 156.17149 593.63866 414.01232 455.4380 544.67958  424.2379
ENSG00000000419 119.75518 155.68649 142.22878 236.5740 187.06534  189.0048
ENSG00000000457 234.18632 378.76759 645.92807 852.7427 578.63835 1081.7101
ENSG00000000460  94.78877 105.07374 101.46359 112.0495  64.16687  272.3186
ENSG00000000971 195.46329  91.30424  78.75588 325.2382 351.35402  189.9438
                   CMV60     CMV61
ENSG00000000003 278.0470 272.54669
ENSG00000000419 218.6861  67.42247
ENSG00000000457 318.3344 438.91393
ENSG00000000460 106.4652  84.70793
ENSG00000000971 131.0963 135.75378
13805 more rows ...

$samples
      group lib.size norm.factors
CMV30     1 14533210    1.0981136
CMV31     1 12415440    1.0566720
CMV8      1 11399505    1.0135492
CMV9      1 11940053    0.9762053
CMV26     1 10869683    1.0615269
21 more rows ...

$genes
DataFrame with 13810 rows and 4 columns
              gene_id      symbol  entrezid    length
          &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;numeric&gt;
1     ENSG00000000003      TSPAN6      7105  2387.631
2     ENSG00000000419        DPM1      8813   905.483
3     ENSG00000000457       SCYL3     57147  3387.525
4     ENSG00000000460    C1orf112     55732  2128.460
5     ENSG00000000971         CFH      3075  4473.834
...               ...         ...       ...       ...
13806 ENSG00000283430  AL513122.2      2934   2334.43
13807 ENSG00000283472  AL109923.1     83737   3268.92
13808 ENSG00000283485  AC090094.1       444   2974.69
13809 ENSG00000283617  AC074141.1     84765   1833.10
13810 ENSG00000283620       ZNF48    197407   2463.84

$group
[1] &quot;pos&quot; &quot;neg&quot; &quot;neg&quot; &quot;pos&quot; &quot;pos&quot;
21 more elements ...</code></pre>
<p>Plotting the distribution log-CPM values shows that a majority of genes within each sample are either not expressed or lowly-expressed with log-CPM values that are small or negative.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Although in excess of 30 million reads were obtained per sample, we can see that after quantification of gene expression the median library size is <code>median(y$samples$lib.size)</code> million reads. This suggests that we are likely to only be capturing the most abundant cfRNAs.</p>
<p>It is assumed that all samples should have a similar range and distribution of expression values. The raw data looks fairly uniform between samples, although TMM normalization further improves this.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-10-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of exploring the relationships between the samples and identifying structure in the data. The following series of MDS plots examines the first four principal components. The samples are coloured by various known features of the samples such as CMV Status and foetal sex. The MDS plots do not show the samples strongly clustering by any of the known features of the data, although there does seem to be some separation between the CMV positive and negative samples in the 1st principal component. This indicates that there are possibly some differentially expressed genes between CMV positive and negative samples.</p>
<p>A weak batch effect is also evident in the 3rd principal component, when we examine the plots coloured by batch.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ruvseq-analysis" class="section level1">
<h1>RUVSeq analysis</h1>
<p>Given that there is a lot ov variability between the samples and it is not clearly associated with any of the known features of the data, we will use and <code>RUVSeq</code> analysis to remove some of the unwanted variation from the data. We will estimate the unwanted variation using the expression of known house-keeping genes (HKG), as defined using the Human BodyMap 2.0 <a href="https://www.tau.ac.il/~elieis/HKG/">data</a>, in our dataset.</p>
<p>The heatmap below shows the expression of the HKG in our samples. Although the expression is fairly uniform, some variability is apparent between the samples.</p>
<pre class="r"><code>hkGenes &lt;- read.delim(here(&quot;data/HK_genes.txt&quot;), stringsAsFactors = FALSE, 
                      header = FALSE, col.names = c(&quot;SYMBOL&quot;,&quot;REFSEQ&quot;), 
                      strip.white = TRUE)

aheatmap(cpm(y[y$genes$symbol %in% hkGenes$SYMBOL,], log = TRUE),
         main = &quot;log2 CPM Expression of HKG&quot;, labRow = NA,  
         annCol = list(GA_at_amnio = targets$GA_at_amnio, 
                       CMV_result = targets$CMV_status, 
                       Pair = targets$pair,
                       Sex = targets$sex),
         annColors = list(scales::brewer_pal(palette = &quot;Dark2&quot;)(length(unique(targets$GA_at_amnio))),
                          scales::hue_pal()(length(unique(targets$CMV_status))),
                          scales::hue_pal()(length(unique(targets$pair))),
                          scales::brewer_pal(palette = &quot;Set1&quot;)(length(unique(targets$sex)))),
         scale = &quot;none&quot;)</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The scree plot shows that most of the variation in the negative control genes is captured by the first principal component.</p>
<pre class="r"><code>PCs &lt;- prcomp(t(y$counts[which(y$genes$symbol %in% hkGenes$SYMBOL),]), 
              center = TRUE, scale = TRUE, retx=TRUE)
loadings = PCs$x # pc loadings
plot(PCs, type=&quot;lines&quot;) # scree plot</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Using the <code>RUVg</code> function, we will estimate the unwanted variation in our data using the HKG for k = 1, 2 and 3.</p>
<pre class="r"><code>yRg1 &lt;- RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k=1)</code></pre>
<pre><code>Warning in RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k = 1): The expression matrix does not contain counts.
Please, pass a matrix of counts (not logged) or set isLog to TRUE to skip the log transformation</code></pre>
<pre class="r"><code>yRg2 &lt;- RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k=2)</code></pre>
<pre><code>Warning in RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k = 2): The expression matrix does not contain counts.
Please, pass a matrix of counts (not logged) or set isLog to TRUE to skip the log transformation</code></pre>
<pre class="r"><code>yRg3 &lt;- RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k=3)</code></pre>
<pre><code>Warning in RUVg(y$counts, which(y$genes$symbol %in% hkGenes$SYMBOL), k = 3): The expression matrix does not contain counts.
Please, pass a matrix of counts (not logged) or set isLog to TRUE to skip the log transformation</code></pre>
<p>The RLE plots below show that <code>RUVg</code> adjustment is a significant improvement over the raw data with k=1. The improvements with k=2 and k=3 are relatively minor in comparison.</p>
<pre class="r"><code>par(mfrow=c(2,2))
plotRLE(y$counts, ylim=c(-1,1), las=2, main=&quot;Raw&quot;, 
        cex.axis = 0.6)
plotRLE(yRg1$normalizedCounts, ylim=c(-0.2,0.2), las=2, main=&quot;RUVg k=1&quot;, 
        cex.axis = 0.6)
plotRLE(yRg2$normalizedCounts, ylim=c(-0.2,0.2), las=2, main=&quot;RUVg k=2&quot;, 
        cex.axis = 0.6)
plotRLE(yRg3$normalizedCounts, ylim=c(-0.2,0.2), las=2, main=&quot;RUVg k=3&quot;, 
        cex.axis = 0.6)</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /> The MDS plots of the <code>RUVg</code> normalised counts show clearer separation between the CMV positive and CMV negative samples.</p>
<pre class="r"><code>mds &lt;- plotMDS(cpm(y$counts, log = TRUE), top = 1000, gene.selection=&quot;common&quot;, 
               plot = FALSE)
dat &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = targets$id,
                      status = targets$CMV_status)

p &lt;- ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;CMV status&quot;) +
  ggtitle(&quot;Raw&quot;)

mds &lt;- plotMDS(yRg1$normalizedCounts, top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat1 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = targets$id,
                      status = targets$CMV_status)

p1 &lt;- ggplot(dat1, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;CMV status&quot;) +
  ggtitle(&quot;RUVg k=1&quot;)

mds &lt;- plotMDS(yRg2$normalizedCounts, top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat2 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = targets$id,
                      status = targets$CMV_status)

p2 &lt;- ggplot(dat2, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;CMV status&quot;) +
  ggtitle(&quot;RUVg k=2&quot;)

mds &lt;- plotMDS(yRg3$normalizedCounts, top = 1000, 
               gene.selection=&quot;common&quot;, plot = FALSE)
dat3 &lt;- tibble::tibble(x = mds$x, y = mds$y,
                      sample = targets$id,
                      status = targets$CMV_status)

p3 &lt;- ggplot(dat3, aes(x = x, y = y, colour = status)) +
  geom_text(aes(label = sample), size = 2.5) +
  labs(x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;,
       colour = &quot;CMV status&quot;) +
  ggtitle(&quot;RUVg k=3&quot;)

((p | p1) / (p2 | p3)) + 
  plot_layout(guides = &quot;collect&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<p>We will look for differentially expressed genes, using the negative binomial GLM approach implemented in <code>edgeR</code>. This is done by considering a design matrix that includes both the covariates of interest (CMV result) and the factors of unwanted variation calculated by <code>RUVg</code>. The CMV positive samples were compared to the CMV negative (normal) samples. A summary of the numbers of differentially expressed genes is shown below, as well as the top 20 differentially expressed genes. The full results tables were exported as csv files.</p>
<pre class="r"><code>norm &lt;- yRg1
design &lt;- model.matrix(~targets$CMV_status + norm$W)
colnames(design)[2] &lt;- &quot;CMV_status&quot;
y &lt;- estimateGLMCommonDisp(y, design)
y &lt;- estimateGLMTagwiseDisp(y, design)
plotBCV(y)</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- glmFit(y, design)
lrt &lt;- glmLRT(fit, coef = 2)
fitSum &lt;- summary(decideTests(lrt, p.value = 0.05))
fitSum</code></pre>
<pre><code>       CMV_status
Down           15
NotSig      13452
Up            343</code></pre>
<pre class="r"><code>plotMD(lrt)</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There were 15 down-regulated and 343 up-regulated genes between CMV positive and CMV negative samples at FDR &lt; 0.05.</p>
<p>These are the top 10 differentially expressed genes.</p>
<pre><code>Warning: The `path` argument of `write_csv()` is deprecated as of readr 1.4.0.
Please use the `file` argument instead.</code></pre>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="16%" />
<col width="7%" />
<col width="9%" />
<col width="10%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="7%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">gene_id</th>
<th align="left">symbol</th>
<th align="right">entrezid</th>
<th align="right">length</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">LR</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000138646</td>
<td align="left">ENSG00000138646</td>
<td align="left">HERC5</td>
<td align="right">51191</td>
<td align="right">3554.2110</td>
<td align="right">3.240035</td>
<td align="right">2.575787</td>
<td align="right">63.65871</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000132530</td>
<td align="left">ENSG00000132530</td>
<td align="left">XAF1</td>
<td align="right">54739</td>
<td align="right">2905.0690</td>
<td align="right">3.263995</td>
<td align="right">5.531925</td>
<td align="right">59.67894</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000137959</td>
<td align="left">ENSG00000137959</td>
<td align="left">IFI44L</td>
<td align="right">10964</td>
<td align="right">3965.1450</td>
<td align="right">3.249470</td>
<td align="right">7.646582</td>
<td align="right">58.76277</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000165949</td>
<td align="left">ENSG00000165949</td>
<td align="left">IFI27</td>
<td align="right">3429</td>
<td align="right">271.2877</td>
<td align="right">3.845441</td>
<td align="right">5.122879</td>
<td align="right">56.95270</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000137628</td>
<td align="left">ENSG00000137628</td>
<td align="left">DDX60</td>
<td align="right">55601</td>
<td align="right">5949.1480</td>
<td align="right">2.027290</td>
<td align="right">6.133818</td>
<td align="right">53.82061</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000206337</td>
<td align="left">ENSG00000206337</td>
<td align="left">HCP5</td>
<td align="right">10866</td>
<td align="right">8208.1855</td>
<td align="right">4.423610</td>
<td align="right">2.705010</td>
<td align="right">52.12646</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000234745</td>
<td align="left">ENSG00000234745</td>
<td align="left">HLA-B</td>
<td align="right">3106</td>
<td align="right">1037.5684</td>
<td align="right">3.330680</td>
<td align="right">5.478258</td>
<td align="right">49.46783</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000240065</td>
<td align="left">ENSG00000240065</td>
<td align="left">PSMB9</td>
<td align="right">5698</td>
<td align="right">617.1136</td>
<td align="right">3.269509</td>
<td align="right">1.626930</td>
<td align="right">48.92831</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000206503</td>
<td align="left">ENSG00000206503</td>
<td align="left">HLA-A</td>
<td align="right">3105</td>
<td align="right">1392.9326</td>
<td align="right">2.390062</td>
<td align="right">5.132018</td>
<td align="right">45.91962</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000157601</td>
<td align="left">ENSG00000157601</td>
<td align="left">MX1</td>
<td align="right">4599</td>
<td align="right">2588.8012</td>
<td align="right">2.108591</td>
<td align="right">6.771776</td>
<td align="right">45.83015</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000137965</td>
<td align="left">ENSG00000137965</td>
<td align="left">IFI44</td>
<td align="right">10561</td>
<td align="right">1240.4006</td>
<td align="right">2.238469</td>
<td align="right">4.684852</td>
<td align="right">45.24820</td>
<td align="right">0</td>
<td align="right">0e+00</td>
</tr>
<tr class="even">
<td align="left">ENSG00000115415</td>
<td align="left">ENSG00000115415</td>
<td align="left">STAT1</td>
<td align="right">6772</td>
<td align="right">3870.7760</td>
<td align="right">1.495069</td>
<td align="right">7.669598</td>
<td align="right">43.39567</td>
<td align="right">0</td>
<td align="right">1e-07</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000138642</td>
<td align="left">ENSG00000138642</td>
<td align="left">HERC6</td>
<td align="right">55008</td>
<td align="right">3160.1320</td>
<td align="right">2.294971</td>
<td align="right">5.096975</td>
<td align="right">43.11905</td>
<td align="right">0</td>
<td align="right">1e-07</td>
</tr>
<tr class="even">
<td align="left">ENSG00000134042</td>
<td align="left">ENSG00000134042</td>
<td align="left">MRO</td>
<td align="right">83876</td>
<td align="right">4700.5555</td>
<td align="right">3.157659</td>
<td align="right">1.303776</td>
<td align="right">42.86281</td>
<td align="right">0</td>
<td align="right">1e-07</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000111335</td>
<td align="left">ENSG00000111335</td>
<td align="left">OAS2</td>
<td align="right">4939</td>
<td align="right">3920.9828</td>
<td align="right">2.651270</td>
<td align="right">5.958775</td>
<td align="right">40.66191</td>
<td align="right">0</td>
<td align="right">2e-07</td>
</tr>
<tr class="even">
<td align="left">ENSG00000134326</td>
<td align="left">ENSG00000134326</td>
<td align="left">CMPK2</td>
<td align="right">129607</td>
<td align="right">2060.8304</td>
<td align="right">2.927074</td>
<td align="right">4.330283</td>
<td align="right">39.91885</td>
<td align="right">0</td>
<td align="right">2e-07</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000107201</td>
<td align="left">ENSG00000107201</td>
<td align="left">DDX58</td>
<td align="right">23586</td>
<td align="right">4656.9714</td>
<td align="right">1.347560</td>
<td align="right">5.378869</td>
<td align="right">39.91836</td>
<td align="right">0</td>
<td align="right">2e-07</td>
</tr>
<tr class="even">
<td align="left">ENSG00000135114</td>
<td align="left">ENSG00000135114</td>
<td align="left">OASL</td>
<td align="right">8638</td>
<td align="right">2063.3455</td>
<td align="right">3.808902</td>
<td align="right">1.959642</td>
<td align="right">39.41942</td>
<td align="right">0</td>
<td align="right">3e-07</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000130303</td>
<td align="left">ENSG00000130303</td>
<td align="left">BST2</td>
<td align="right">684</td>
<td align="right">618.6255</td>
<td align="right">3.639003</td>
<td align="right">3.460692</td>
<td align="right">38.97830</td>
<td align="right">0</td>
<td align="right">3e-07</td>
</tr>
<tr class="even">
<td align="left">ENSG00000111331</td>
<td align="left">ENSG00000111331</td>
<td align="left">OAS3</td>
<td align="right">4940</td>
<td align="right">4752.7137</td>
<td align="right">2.025545</td>
<td align="right">6.338771</td>
<td align="right">38.37953</td>
<td align="right">0</td>
<td align="right">4e-07</td>
</tr>
</tbody>
</table>
<p>The following plots show the expression of the top 12 ranked differentially expressed genes for CMV positive and CMV negative samples. Although there is significant variability within the groups and the log2 fold changes are not large, there are obvious differences in expression for the top ranked genes.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level1">
<h1>Gene set enrichment analysis (GSEA)</h1>
<p>GSEA helps us to interpret the results of a differential expression analysis.</p>
<div id="go-terms" class="section level2">
<h2>GO Terms</h2>
<p>Testing for enrichment of Gene Ontology (GO) categories among statistically significant differentially expressed genes.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Term</th>
<th align="left">Ont</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0034340" class="uri">GO:0034340</a></td>
<td align="left">response to type I interferon</td>
<td align="left">BP</td>
<td align="right">78</td>
<td align="right">31</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0071357" class="uri">GO:0071357</a></td>
<td align="left">cellular response to type I interferon</td>
<td align="left">BP</td>
<td align="right">74</td>
<td align="right">30</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0060337" class="uri">GO:0060337</a></td>
<td align="left">type I interferon signaling pathway</td>
<td align="left">BP</td>
<td align="right">74</td>
<td align="right">30</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0045087" class="uri">GO:0045087</a></td>
<td align="left">innate immune response</td>
<td align="left">BP</td>
<td align="right">569</td>
<td align="right">67</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006955" class="uri">GO:0006955</a></td>
<td align="left">immune response</td>
<td align="left">BP</td>
<td align="right">1279</td>
<td align="right">101</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0043207" class="uri">GO:0043207</a></td>
<td align="left">response to external biotic stimulus</td>
<td align="left">BP</td>
<td align="right">915</td>
<td align="right">84</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051707" class="uri">GO:0051707</a></td>
<td align="left">response to other organism</td>
<td align="left">BP</td>
<td align="right">915</td>
<td align="right">84</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0009607" class="uri">GO:0009607</a></td>
<td align="left">response to biotic stimulus</td>
<td align="left">BP</td>
<td align="right">942</td>
<td align="right">85</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0098542" class="uri">GO:0098542</a></td>
<td align="left">defense response to other organism</td>
<td align="left">BP</td>
<td align="right">692</td>
<td align="right">71</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0031224" class="uri">GO:0031224</a></td>
<td align="left">intrinsic component of membrane</td>
<td align="left">CC</td>
<td align="right">2915</td>
<td align="right">159</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0016021" class="uri">GO:0016021</a></td>
<td align="left">integral component of membrane</td>
<td align="left">CC</td>
<td align="right">2832</td>
<td align="right">154</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">BP</td>
<td align="right">2026</td>
<td align="right">125</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006952" class="uri">GO:0006952</a></td>
<td align="left">defense response</td>
<td align="left">BP</td>
<td align="right">998</td>
<td align="right">83</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0009615" class="uri">GO:0009615</a></td>
<td align="left">response to virus</td>
<td align="left">BP</td>
<td align="right">243</td>
<td align="right">40</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051607" class="uri">GO:0051607</a></td>
<td align="left">defense response to virus</td>
<td align="left">BP</td>
<td align="right">187</td>
<td align="right">34</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0034341" class="uri">GO:0034341</a></td>
<td align="left">response to interferon-gamma</td>
<td align="left">BP</td>
<td align="right">140</td>
<td align="right">30</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0009605" class="uri">GO:0009605</a></td>
<td align="left">response to external stimulus</td>
<td align="left">BP</td>
<td align="right">1795</td>
<td align="right">110</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0019221" class="uri">GO:0019221</a></td>
<td align="left">cytokine-mediated signaling pathway</td>
<td align="left">BP</td>
<td align="right">539</td>
<td align="right">55</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0071346" class="uri">GO:0071346</a></td>
<td align="left">cellular response to interferon-gamma</td>
<td align="left">BP</td>
<td align="right">125</td>
<td align="right">27</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0034097" class="uri">GO:0034097</a></td>
<td align="left">response to cytokine</td>
<td align="left">BP</td>
<td align="right">849</td>
<td align="right">68</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="reactome-pathways" class="section level2">
<h2>Reactome Pathways</h2>
<p>Over-representation analysis of Reactome Pathways.</p>
<pre class="r"><code>rct &lt;- enrichPathway(top$entrezid[top$FDR &lt; 0.05], 
                     universe = as.character((top$entrezid)))</code></pre>
<pre><code>Loading required package: org.Hs.eg.db</code></pre>
<pre class="r"><code>rct %&gt;% data.frame %&gt;%
  dplyr::filter(p.adjust &lt; 0.05) %&gt;%
  dplyr::select(-geneID) -&gt; rct
rct %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="8%" />
<col width="47%" />
<col width="6%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">R-HSA-909733</td>
<td align="left">R-HSA-909733</td>
<td align="left">Interferon alpha/beta signaling</td>
<td align="left">28/238</td>
<td align="left">54/7629</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">28</td>
</tr>
<tr class="even">
<td align="left">R-HSA-913531</td>
<td align="left">R-HSA-913531</td>
<td align="left">Interferon Signaling</td>
<td align="left">39/238</td>
<td align="left">159/7629</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">39</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-877300</td>
<td align="left">R-HSA-877300</td>
<td align="left">Interferon gamma signaling</td>
<td align="left">19/238</td>
<td align="left">70/7629</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">19</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1236977</td>
<td align="left">R-HSA-1236977</td>
<td align="left">Endosomal/Vacuolar pathway</td>
<td align="left">7/238</td>
<td align="left">10/7629</td>
<td align="right">0.0000000</td>
<td align="right">0.0000004</td>
<td align="right">0.0000004</td>
<td align="right">7</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-425407</td>
<td align="left">R-HSA-425407</td>
<td align="left">SLC-mediated transmembrane transport</td>
<td align="left">20/238</td>
<td align="left">154/7629</td>
<td align="right">0.0000001</td>
<td align="right">0.0000057</td>
<td align="right">0.0000052</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1169410</td>
<td align="left">R-HSA-1169410</td>
<td align="left">Antiviral mechanism by IFN-stimulated genes</td>
<td align="left">13/238</td>
<td align="left">74/7629</td>
<td align="right">0.0000004</td>
<td align="right">0.0000329</td>
<td align="right">0.0000298</td>
<td align="right">13</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-8963898</td>
<td align="left">R-HSA-8963898</td>
<td align="left">Plasma lipoprotein assembly</td>
<td align="left">6/238</td>
<td align="left">15/7629</td>
<td align="right">0.0000034</td>
<td align="right">0.0002615</td>
<td align="right">0.0002372</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">R-HSA-211859</td>
<td align="left">R-HSA-211859</td>
<td align="left">Biological oxidations</td>
<td align="left">15/238</td>
<td align="left">121/7629</td>
<td align="right">0.0000047</td>
<td align="right">0.0003148</td>
<td align="right">0.0002856</td>
<td align="right">15</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-983170</td>
<td align="left">R-HSA-983170</td>
<td align="left">Antigen Presentation: Folding, assembly and peptide loading of class I MHC</td>
<td align="left">7/238</td>
<td align="left">24/7629</td>
<td align="right">0.0000058</td>
<td align="right">0.0003441</td>
<td align="right">0.0003121</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">R-HSA-425393</td>
<td align="left">R-HSA-425393</td>
<td align="left">Transport of inorganic cations/anions and amino acids/oligopeptides</td>
<td align="left">11/238</td>
<td align="left">71/7629</td>
<td align="right">0.0000106</td>
<td align="right">0.0005461</td>
<td align="right">0.0004953</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-8963743</td>
<td align="left">R-HSA-8963743</td>
<td align="left">Digestion and absorption</td>
<td align="left">5/238</td>
<td align="left">11/7629</td>
<td align="right">0.0000112</td>
<td align="right">0.0005461</td>
<td align="right">0.0004953</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">R-HSA-8963899</td>
<td align="left">R-HSA-8963899</td>
<td align="left">Plasma lipoprotein remodeling</td>
<td align="left">6/238</td>
<td align="left">19/7629</td>
<td align="right">0.0000167</td>
<td align="right">0.0007443</td>
<td align="right">0.0006751</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-198933</td>
<td align="left">R-HSA-198933</td>
<td align="left">Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</td>
<td align="left">8/238</td>
<td align="left">40/7629</td>
<td align="right">0.0000258</td>
<td align="right">0.0010635</td>
<td align="right">0.0009646</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1169408</td>
<td align="left">R-HSA-1169408</td>
<td align="left">ISG15 antiviral mechanism</td>
<td align="left">9/238</td>
<td align="left">66/7629</td>
<td align="right">0.0001887</td>
<td align="right">0.0072108</td>
<td align="right">0.0065405</td>
<td align="right">9</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-375276</td>
<td align="left">R-HSA-375276</td>
<td align="left">Peptide ligand-binding receptors</td>
<td align="left">7/238</td>
<td align="left">41/7629</td>
<td align="right">0.0002391</td>
<td align="right">0.0085269</td>
<td align="right">0.0077341</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1236975</td>
<td align="left">R-HSA-1236975</td>
<td align="left">Antigen processing-Cross presentation</td>
<td align="left">10/238</td>
<td align="left">84/7629</td>
<td align="right">0.0002628</td>
<td align="right">0.0086187</td>
<td align="right">0.0078174</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-975634</td>
<td align="left">R-HSA-975634</td>
<td align="left">Retinoid metabolism and transport</td>
<td align="left">6/238</td>
<td align="left">30/7629</td>
<td align="right">0.0002739</td>
<td align="right">0.0086187</td>
<td align="right">0.0078174</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">R-HSA-210745</td>
<td align="left">R-HSA-210745</td>
<td align="left">Regulation of gene expression in beta cells</td>
<td align="left">4/238</td>
<td align="left">12/7629</td>
<td align="right">0.0003752</td>
<td align="right">0.0111530</td>
<td align="right">0.0101161</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1236974</td>
<td align="left">R-HSA-1236974</td>
<td align="left">ER-Phagosome pathway</td>
<td align="left">9/238</td>
<td align="left">74/7629</td>
<td align="right">0.0004537</td>
<td align="right">0.0127757</td>
<td align="right">0.0115880</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">R-HSA-6806667</td>
<td align="left">R-HSA-6806667</td>
<td align="left">Metabolism of fat-soluble vitamins</td>
<td align="left">6/238</td>
<td align="left">34/7629</td>
<td align="right">0.0005588</td>
<td align="right">0.0145316</td>
<td align="right">0.0131807</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-425366</td>
<td align="left">R-HSA-425366</td>
<td align="left">Transport of bile salts and organic acids, metal ions and amine compounds</td>
<td align="left">7/238</td>
<td align="left">47/7629</td>
<td align="right">0.0005704</td>
<td align="right">0.0145316</td>
<td align="right">0.0131807</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">R-HSA-3000480</td>
<td align="left">R-HSA-3000480</td>
<td align="left">Scavenging by Class A Receptors</td>
<td align="left">4/238</td>
<td align="left">14/7629</td>
<td align="right">0.0007223</td>
<td align="right">0.0171876</td>
<td align="right">0.0155897</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-174824</td>
<td align="left">R-HSA-174824</td>
<td align="left">Plasma lipoprotein assembly, remodeling, and clearance</td>
<td align="left">7/238</td>
<td align="left">49/7629</td>
<td align="right">0.0007389</td>
<td align="right">0.0171876</td>
<td align="right">0.0155897</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">R-HSA-352230</td>
<td align="left">R-HSA-352230</td>
<td align="left">Amino acid transport across the plasma membrane</td>
<td align="left">5/238</td>
<td align="left">25/7629</td>
<td align="right">0.0009025</td>
<td align="right">0.0197750</td>
<td align="right">0.0179366</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-5619115</td>
<td align="left">R-HSA-5619115</td>
<td align="left">Disorders of transmembrane transporters</td>
<td align="left">12/238</td>
<td align="left">134/7629</td>
<td align="right">0.0009350</td>
<td align="right">0.0197750</td>
<td align="right">0.0179366</td>
<td align="right">12</td>
</tr>
<tr class="even">
<td align="left">R-HSA-156584</td>
<td align="left">R-HSA-156584</td>
<td align="left">Cytosolic sulfonation of small molecules</td>
<td align="left">4/238</td>
<td align="left">15/7629</td>
<td align="right">0.0009610</td>
<td align="right">0.0197750</td>
<td align="right">0.0179366</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-2980736</td>
<td align="left">R-HSA-2980736</td>
<td align="left">Peptide hormone metabolism</td>
<td align="left">7/238</td>
<td align="left">52/7629</td>
<td align="right">0.0010630</td>
<td align="right">0.0210636</td>
<td align="right">0.0191054</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1660662</td>
<td align="left">R-HSA-1660662</td>
<td align="left">Glycosphingolipid metabolism</td>
<td align="left">6/238</td>
<td align="left">39/7629</td>
<td align="right">0.0011901</td>
<td align="right">0.0227399</td>
<td align="right">0.0206259</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-5619102</td>
<td align="left">R-HSA-5619102</td>
<td align="left">SLC transporter disorders</td>
<td align="left">8/238</td>
<td align="left">70/7629</td>
<td align="right">0.0014177</td>
<td align="right">0.0261543</td>
<td align="right">0.0237229</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">R-HSA-211945</td>
<td align="left">R-HSA-211945</td>
<td align="left">Phase I - Functionalization of compounds</td>
<td align="left">7/238</td>
<td align="left">57/7629</td>
<td align="right">0.0018410</td>
<td align="right">0.0328307</td>
<td align="right">0.0297786</td>
<td align="right">7</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-156580</td>
<td align="left">R-HSA-156580</td>
<td align="left">Phase II - Conjugation of compounds</td>
<td align="left">7/238</td>
<td align="left">60/7629</td>
<td align="right">0.0024848</td>
<td align="right">0.0428834</td>
<td align="right">0.0388967</td>
<td align="right">7</td>
</tr>
</tbody>
</table>
<pre class="r"><code>viewPathway(&quot;Interferon alpha/beta signaling&quot;, organism = &quot;human&quot;,
            foldChange = setNames(top$logFC, 
                                  top$entrezid)[!duplicated(top$entrezid)])</code></pre>
<pre><code>Loading required package: graphite</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns
&#39;select()&#39; returned 1:many mapping between keys and columns
&#39;select()&#39; returned 1:many mapping between keys and columns
&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
use `guide = &quot;none&quot;` instead.</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Functional class scoring analysis of Reactome Pathways.</p>
<pre class="r"><code>rct &lt;- gsePathway(setNames(top$LR, top$entrezid)[!duplicated(top$entrezid)],
                  scoreType = &quot;pos&quot;)</code></pre>
<pre><code>preparing geneSet collections...</code></pre>
<pre><code>GSEA analysis...</code></pre>
<pre><code>Warning in fgseaMultilevel(...): For some pathways, in reality P-values are less
than 1e-10. You can set the `eps` argument to zero for better estimation.</code></pre>
<pre><code>leading edge analysis...</code></pre>
<pre><code>done...</code></pre>
<pre class="r"><code>rct %&gt;% data.frame %&gt;%
  dplyr::filter(p.adjust &lt; 0.05) %&gt;%
  dplyr::select(-core_enrichment) -&gt; rct
rct %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="6%" />
<col width="6%" />
<col width="37%" />
<col width="3%" />
<col width="7%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="2%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="right">setSize</th>
<th align="right">enrichmentScore</th>
<th align="right">NES</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalues</th>
<th align="right">rank</th>
<th align="left">leading_edge</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">R-HSA-909733</td>
<td align="left">R-HSA-909733</td>
<td align="left">Interferon alpha/beta signaling</td>
<td align="right">54</td>
<td align="right">0.9292407</td>
<td align="right">1.609161</td>
<td align="right">0.0000000</td>
<td align="right">0.0000001</td>
<td align="right">0.0000001</td>
<td align="right">639</td>
<td align="left">tags=61%, list=5%, signal=59%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-913531</td>
<td align="left">R-HSA-913531</td>
<td align="left">Interferon Signaling</td>
<td align="right">159</td>
<td align="right">0.8562859</td>
<td align="right">1.521806</td>
<td align="right">0.0000000</td>
<td align="right">0.0000001</td>
<td align="right">0.0000001</td>
<td align="right">993</td>
<td align="left">tags=33%, list=7%, signal=31%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1169410</td>
<td align="left">R-HSA-1169410</td>
<td align="left">Antiviral mechanism by IFN-stimulated genes</td>
<td align="right">74</td>
<td align="right">0.8560654</td>
<td align="right">1.490239</td>
<td align="right">0.0000000</td>
<td align="right">0.0000008</td>
<td align="right">0.0000008</td>
<td align="right">253</td>
<td align="left">tags=18%, list=2%, signal=17%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-877300</td>
<td align="left">R-HSA-877300</td>
<td align="left">Interferon gamma signaling</td>
<td align="right">70</td>
<td align="right">0.8647611</td>
<td align="right">1.505336</td>
<td align="right">0.0000000</td>
<td align="right">0.0000017</td>
<td align="right">0.0000015</td>
<td align="right">993</td>
<td align="left">tags=40%, list=7%, signal=37%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1236974</td>
<td align="left">R-HSA-1236974</td>
<td align="left">ER-Phagosome pathway</td>
<td align="right">74</td>
<td align="right">0.8139049</td>
<td align="right">1.416846</td>
<td align="right">0.0000010</td>
<td align="right">0.0002225</td>
<td align="right">0.0002065</td>
<td align="right">752</td>
<td align="left">tags=18%, list=5%, signal=17%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-425407</td>
<td align="left">R-HSA-425407</td>
<td align="left">SLC-mediated transmembrane transport</td>
<td align="right">154</td>
<td align="right">0.7471372</td>
<td align="right">1.327018</td>
<td align="right">0.0000010</td>
<td align="right">0.0002225</td>
<td align="right">0.0002065</td>
<td align="right">1119</td>
<td align="left">tags=29%, list=8%, signal=27%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1236975</td>
<td align="left">R-HSA-1236975</td>
<td align="left">Antigen processing-Cross presentation</td>
<td align="right">84</td>
<td align="right">0.7940579</td>
<td align="right">1.389399</td>
<td align="right">0.0000018</td>
<td align="right">0.0003185</td>
<td align="right">0.0002956</td>
<td align="right">503</td>
<td align="left">tags=15%, list=4%, signal=15%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-983169</td>
<td align="left">R-HSA-983169</td>
<td align="left">Class I MHC mediated antigen processing &amp; presentation</td>
<td align="right">326</td>
<td align="right">0.6860647</td>
<td align="right">1.233201</td>
<td align="right">0.0000019</td>
<td align="right">0.0003185</td>
<td align="right">0.0002956</td>
<td align="right">783</td>
<td align="left">tags=10%, list=6%, signal=9%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1169408</td>
<td align="left">R-HSA-1169408</td>
<td align="left">ISG15 antiviral mechanism</td>
<td align="right">66</td>
<td align="right">0.8179769</td>
<td align="right">1.420812</td>
<td align="right">0.0000033</td>
<td align="right">0.0004735</td>
<td align="right">0.0004395</td>
<td align="right">535</td>
<td align="left">tags=15%, list=4%, signal=15%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-1236977</td>
<td align="left">R-HSA-1236977</td>
<td align="left">Endosomal/Vacuolar pathway</td>
<td align="right">10</td>
<td align="right">0.9767098</td>
<td align="right">1.598107</td>
<td align="right">0.0000105</td>
<td align="right">0.0013752</td>
<td align="right">0.0012765</td>
<td align="right">196</td>
<td align="left">tags=70%, list=1%, signal=69%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-8963898</td>
<td align="left">R-HSA-8963898</td>
<td align="left">Plasma lipoprotein assembly</td>
<td align="right">15</td>
<td align="right">0.9457502</td>
<td align="right">1.579325</td>
<td align="right">0.0000295</td>
<td align="right">0.0035119</td>
<td align="right">0.0032596</td>
<td align="right">119</td>
<td align="left">tags=40%, list=1%, signal=40%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-912694</td>
<td align="left">R-HSA-912694</td>
<td align="left">Regulation of IFNA signaling</td>
<td align="right">12</td>
<td align="right">0.9497032</td>
<td align="right">1.565370</td>
<td align="right">0.0000497</td>
<td align="right">0.0054266</td>
<td align="right">0.0050369</td>
<td align="right">586</td>
<td align="left">tags=42%, list=4%, signal=40%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-983170</td>
<td align="left">R-HSA-983170</td>
<td align="left">Antigen Presentation: Folding, assembly and peptide loading of class I MHC</td>
<td align="right">24</td>
<td align="right">0.8880091</td>
<td align="right">1.512129</td>
<td align="right">0.0000931</td>
<td align="right">0.0093853</td>
<td align="right">0.0087112</td>
<td align="right">752</td>
<td align="left">tags=42%, list=5%, signal=39%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-5619115</td>
<td align="left">R-HSA-5619115</td>
<td align="left">Disorders of transmembrane transporters</td>
<td align="right">134</td>
<td align="right">0.7155870</td>
<td align="right">1.265659</td>
<td align="right">0.0002075</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">535</td>
<td align="left">tags=15%, list=4%, signal=14%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-1059683</td>
<td align="left">R-HSA-1059683</td>
<td align="left">Interleukin-6 signaling</td>
<td align="right">10</td>
<td align="right">0.9527017</td>
<td align="right">1.558824</td>
<td align="right">0.0002620</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">12</td>
<td align="left">tags=10%, list=0%, signal=10%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-425393</td>
<td align="left">R-HSA-425393</td>
<td align="left">Transport of inorganic cations/anions and amino acids/oligopeptides</td>
<td align="right">71</td>
<td align="right">0.7627086</td>
<td align="right">1.327263</td>
<td align="right">0.0002620</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">2250</td>
<td align="left">tags=46%, list=16%, signal=39%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-936440</td>
<td align="left">R-HSA-936440</td>
<td align="left">Negative regulators of DDX58/IFIH1 signaling</td>
<td align="right">31</td>
<td align="right">0.8384476</td>
<td align="right">1.441980</td>
<td align="right">0.0002666</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">211</td>
<td align="left">tags=13%, list=2%, signal=13%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-8939902</td>
<td align="left">R-HSA-8939902</td>
<td align="left">Regulation of RUNX2 expression and activity</td>
<td align="right">64</td>
<td align="right">0.7658348</td>
<td align="right">1.328857</td>
<td align="right">0.0002711</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">818</td>
<td align="left">tags=12%, list=6%, signal=12%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-6798695</td>
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="right">373</td>
<td align="right">0.6470410</td>
<td align="right">1.165319</td>
<td align="right">0.0002803</td>
<td align="right">0.0193374</td>
<td align="right">0.0179486</td>
<td align="right">3104</td>
<td align="left">tags=34%, list=23%, signal=27%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-75815</td>
<td align="left">R-HSA-75815</td>
<td align="left">Ubiquitin-dependent degradation of Cyclin D</td>
<td align="right">48</td>
<td align="right">0.7727639</td>
<td align="right">1.335344</td>
<td align="right">0.0007464</td>
<td align="right">0.0471714</td>
<td align="right">0.0437835</td>
<td align="right">373</td>
<td align="left">tags=10%, list=3%, signal=10%</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-5083625</td>
<td align="left">R-HSA-5083625</td>
<td align="left">Defective GALNT3 causes familial hyperphosphatemic tumoral calcinosis (HFTC)</td>
<td align="right">15</td>
<td align="right">0.8961816</td>
<td align="right">1.496550</td>
<td align="right">0.0007556</td>
<td align="right">0.0471714</td>
<td align="right">0.0437835</td>
<td align="right">542</td>
<td align="left">tags=27%, list=4%, signal=26%</td>
</tr>
<tr class="even">
<td align="left">R-HSA-8963899</td>
<td align="left">R-HSA-8963899</td>
<td align="left">Plasma lipoprotein remodeling</td>
<td align="right">19</td>
<td align="right">0.8838558</td>
<td align="right">1.491993</td>
<td align="right">0.0008014</td>
<td align="right">0.0477562</td>
<td align="right">0.0443262</td>
<td align="right">1066</td>
<td align="left">tags=53%, list=8%, signal=49%</td>
</tr>
</tbody>
</table>
<pre class="r"><code>viewPathway(&quot;Interferon gamma signaling&quot;, 
            organism = &quot;human&quot;,
            foldChange = setNames(top$logFC, 
                                  top$entrezid)[!duplicated(top$entrezid)],
            max.overlaps = 100, guide = &quot;none&quot;)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns
&#39;select()&#39; returned 1:1 mapping between keys and columns
&#39;select()&#39; returned 1:1 mapping between keys and columns
&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre><code>Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
use `guide = &quot;none&quot;` instead.</code></pre>
<pre><code>Warning: ggrepel: 15 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /> ## Camera analysis with various gene sets</p>
<p>The <code>camera</code> function performs a competitive test to assess whether the genes in a given set are highly ranked in terms of differential expression relative to genes that are not in the set. We have tested several collections of gene sets from the Broad Institute’s Molecular Signatures Database <a href="http://software.broadinstitute.org/gsea/msigdb/index.jsp">MSigDB</a>.</p>
<p>Build gene set indexes.</p>
<pre><code>[1] &quot;Loading MSigDB Gene Sets ... &quot;
[1] &quot;Loaded gene sets for the collection h ...&quot;
[1] &quot;Indexed the collection h ...&quot;
[1] &quot;Created annotation for the collection h ...&quot;
[1] &quot;Loaded gene sets for the collection c2 ...&quot;
[1] &quot;Indexed the collection c2 ...&quot;
[1] &quot;Created annotation for the collection c2 ...&quot;
[1] &quot;Loaded gene sets for the collection c5 ...&quot;
[1] &quot;Indexed the collection c5 ...&quot;
[1] &quot;Created annotation for the collection c5 ...&quot;
[1] &quot;Building KEGG pathways annotation object ... &quot;</code></pre>
<p>The GO gene sets consist of genes annotated by the same GO terms.</p>
<table>
<colgroup>
<col width="67%" />
<col width="6%" />
<col width="9%" />
<col width="6%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NGenes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_TYPE_I_INTERFERON</td>
<td align="right">51</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_INTERFERON_GAMMA_MEDIATED_SIGNALING_PATHWAY</td>
<td align="right">52</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">GO_CELLULAR_RESPONSE_TO_INTERFERON_GAMMA</td>
<td align="right">73</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_INTERFERON_GAMMA</td>
<td align="right">89</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">GO_MHC_CLASS_I_PROTEIN_COMPLEX</td>
<td align="right">9</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_PEPTIDE_ANTIGEN</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">GO_MHC_PROTEIN_COMPLEX</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_ANTIGEN</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">GO_PEPTIDE_ANTIGEN_BINDING</td>
<td align="right">20</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_DEFENSE_RESPONSE_TO_VIRUS</td>
<td align="right">113</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="odd">
<td align="left">GO_LUMENAL_SIDE_OF_MEMBRANE</td>
<td align="right">23</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0.00e+00</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_CELL_KILLING</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">1.60e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ACID_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">80</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">2.80e-06</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY</td>
<td align="right">8</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">3.80e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_ORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">99</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">8.40e-06</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_INNATE_IMMUNE_RESPONSE</td>
<td align="right">32</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">9.60e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">9.60e-06</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY</td>
<td align="right">11</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">9.60e-06</td>
</tr>
<tr class="odd">
<td align="left">GO_SECONDARY_ACTIVE_TRANSMEMBRANE_TRANSPORTER_ACTIVITY</td>
<td align="right">124</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">9.60e-06</td>
</tr>
<tr class="even">
<td align="left">GO_DIGESTION</td>
<td align="right">67</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">1.28e-05</td>
</tr>
</tbody>
</table>
<p>The Hallmark gene sets are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NGenes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HALLMARK_INTERFERON_ALPHA_RESPONSE</td>
<td align="right">86</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_INTERFERON_GAMMA_RESPONSE</td>
<td align="right">163</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_E2F_TARGETS</td>
<td align="right">188</td>
<td align="left">Down</td>
<td align="right">0.0000000</td>
<td align="right">0.0000002</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_MYC_TARGETS_V1</td>
<td align="right">191</td>
<td align="left">Down</td>
<td align="right">0.0000000</td>
<td align="right">0.0000003</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_INFLAMMATORY_RESPONSE</td>
<td align="right">125</td>
<td align="left">Up</td>
<td align="right">0.0000007</td>
<td align="right">0.0000072</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_IL6_JAK_STAT3_SIGNALING</td>
<td align="right">53</td>
<td align="left">Up</td>
<td align="right">0.0000035</td>
<td align="right">0.0000294</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_COMPLEMENT</td>
<td align="right">151</td>
<td align="left">Up</td>
<td align="right">0.0000093</td>
<td align="right">0.0000666</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_G2M_CHECKPOINT</td>
<td align="right">191</td>
<td align="left">Down</td>
<td align="right">0.0000120</td>
<td align="right">0.0000749</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_MITOTIC_SPINDLE</td>
<td align="right">196</td>
<td align="left">Down</td>
<td align="right">0.0000353</td>
<td align="right">0.0001961</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_TNFA_SIGNALING_VIA_NFKB</td>
<td align="right">163</td>
<td align="left">Up</td>
<td align="right">0.0000822</td>
<td align="right">0.0004108</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_KRAS_SIGNALING_UP</td>
<td align="right">147</td>
<td align="left">Up</td>
<td align="right">0.0000911</td>
<td align="right">0.0004139</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_ALLOGRAFT_REJECTION</td>
<td align="right">115</td>
<td align="left">Up</td>
<td align="right">0.0003384</td>
<td align="right">0.0014099</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_COAGULATION</td>
<td align="right">92</td>
<td align="left">Up</td>
<td align="right">0.0012766</td>
<td align="right">0.0049098</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_XENOBIOTIC_METABOLISM</td>
<td align="right">144</td>
<td align="left">Up</td>
<td align="right">0.0030194</td>
<td align="right">0.0107836</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_IL2_STAT5_SIGNALING</td>
<td align="right">154</td>
<td align="left">Up</td>
<td align="right">0.0084250</td>
<td align="right">0.0280835</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_DNA_REPAIR</td>
<td align="right">142</td>
<td align="left">Down</td>
<td align="right">0.0543543</td>
<td align="right">0.1698571</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_MYC_TARGETS_V2</td>
<td align="right">56</td>
<td align="left">Down</td>
<td align="right">0.0593275</td>
<td align="right">0.1744927</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_ADIPOGENESIS</td>
<td align="right">179</td>
<td align="left">Up</td>
<td align="right">0.0801847</td>
<td align="right">0.2227352</td>
</tr>
<tr class="odd">
<td align="left">HALLMARK_ANDROGEN_RESPONSE</td>
<td align="right">95</td>
<td align="left">Up</td>
<td align="right">0.1039429</td>
<td align="right">0.2735340</td>
</tr>
<tr class="even">
<td align="left">HALLMARK_P53_PATHWAY</td>
<td align="right">185</td>
<td align="left">Up</td>
<td align="right">0.1122435</td>
<td align="right">0.2806087</td>
</tr>
</tbody>
</table>
<p>Barcode plots show the enrichment of gene sets among up or down-regulated genes. The following barcode plots show the enrichment of the top 4 hallmark gene sets among the genes differentially expressed between CMV positive and CMV negative samples.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-31-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
<p>The curated gene sets are compiled from online pathway databases, publications in PubMed, and knowledge of domain experts.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NGenes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MOSERLE_IFNA_RESPONSE</td>
<td align="right">29</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">BROWNE_INTERFERON_RESPONSIVE_GENES</td>
<td align="right">60</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">HECKER_IFNB1_TARGETS</td>
<td align="right">64</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">SANA_RESPONSE_TO_IFNG_UP</td>
<td align="right">58</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">DAUER_STAT3_TARGETS_DN</td>
<td align="right">49</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">BOSCO_INTERFERON_INDUCED_ANTIVIRAL_MODULE</td>
<td align="right">65</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">FARMER_BREAST_CANCER_CLUSTER_1</td>
<td align="right">22</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">REACTOME_INTERFERON_ALPHA_BETA_SIGNALING</td>
<td align="right">47</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">EINAV_INTERFERON_SIGNATURE_IN_CANCER</td>
<td align="right">26</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">BOWIE_RESPONSE_TO_TAMOXIFEN</td>
<td align="right">17</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">RADAEVA_RESPONSE_TO_IFNA1_UP</td>
<td align="right">45</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ZHANG_INTERFERON_RESPONSE</td>
<td align="right">21</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">BENNETT_SYSTEMIC_LUPUS_ERYTHEMATOSUS</td>
<td align="right">25</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">BOWIE_RESPONSE_TO_EXTRACELLULAR_MATRIX</td>
<td align="right">16</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">SANA_TNF_SIGNALING_UP</td>
<td align="right">64</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">DER_IFN_ALPHA_RESPONSE_UP</td>
<td align="right">66</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">SEITZ_NEOPLASTIC_TRANSFORMATION_BY_8P_DELETION_UP</td>
<td align="right">59</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">UROSEVIC_RESPONSE_TO_IMIQUIMOD</td>
<td align="right">16</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ROETH_TERT_TARGETS_UP</td>
<td align="right">14</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">KRASNOSELSKAYA_ILF3_TARGETS_UP</td>
<td align="right">31</td>
<td align="left">Up</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>The following barcode plots show the enrichment of the top 4 curated gene sets among the genes differentially expressed between CMV positive and CMV negative samples.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-33-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
<p>The KEGG gene sets encompass all of the pathways defined in the <a href="https://www.genome.jp/kegg/pathway.html">Kegg pathway database</a>.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NGenes</th>
<th align="left">Direction</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Mineral absorption</td>
<td align="right">40</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Type I diabetes mellitus</td>
<td align="right">16</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">Autoimmune thyroid disease</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000001</td>
</tr>
<tr class="even">
<td align="left">Allograft rejection</td>
<td align="right">12</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000001</td>
</tr>
<tr class="odd">
<td align="left">Graft-versus-host disease</td>
<td align="right">13</td>
<td align="left">Up</td>
<td align="right">0.0000000</td>
<td align="right">0.0000002</td>
</tr>
<tr class="even">
<td align="left">Antigen processing and presentation</td>
<td align="right">46</td>
<td align="left">Up</td>
<td align="right">0.0000008</td>
<td align="right">0.0000374</td>
</tr>
<tr class="odd">
<td align="left">Protein digestion and absorption</td>
<td align="right">52</td>
<td align="left">Up</td>
<td align="right">0.0000010</td>
<td align="right">0.0000432</td>
</tr>
<tr class="even">
<td align="left">Fat digestion and absorption</td>
<td align="right">21</td>
<td align="left">Up</td>
<td align="right">0.0000040</td>
<td align="right">0.0001431</td>
</tr>
<tr class="odd">
<td align="left">Spliceosome</td>
<td align="right">126</td>
<td align="left">Down</td>
<td align="right">0.0000044</td>
<td align="right">0.0001431</td>
</tr>
<tr class="even">
<td align="left">Vitamin digestion and absorption</td>
<td align="right">17</td>
<td align="left">Up</td>
<td align="right">0.0000079</td>
<td align="right">0.0002297</td>
</tr>
<tr class="odd">
<td align="left">Cell adhesion molecules (CAMs)</td>
<td align="right">78</td>
<td align="left">Up</td>
<td align="right">0.0000251</td>
<td align="right">0.0006635</td>
</tr>
<tr class="even">
<td align="left">Cytokine-cytokine receptor interaction</td>
<td align="right">99</td>
<td align="left">Up</td>
<td align="right">0.0000288</td>
<td align="right">0.0006990</td>
</tr>
<tr class="odd">
<td align="left">Bile secretion</td>
<td align="right">36</td>
<td align="left">Up</td>
<td align="right">0.0001339</td>
<td align="right">0.0027887</td>
</tr>
<tr class="even">
<td align="left">Proximal tubule bicarbonate reclamation</td>
<td align="right">15</td>
<td align="left">Up</td>
<td align="right">0.0001342</td>
<td align="right">0.0027887</td>
</tr>
<tr class="odd">
<td align="left">Hepatitis C</td>
<td align="right">104</td>
<td align="left">Up</td>
<td align="right">0.0001677</td>
<td align="right">0.0032535</td>
</tr>
<tr class="even">
<td align="left">Influenza A</td>
<td align="right">127</td>
<td align="left">Up</td>
<td align="right">0.0002455</td>
<td align="right">0.0044650</td>
</tr>
<tr class="odd">
<td align="left">Lysosome</td>
<td align="right">112</td>
<td align="left">Up</td>
<td align="right">0.0003188</td>
<td align="right">0.0054573</td>
</tr>
<tr class="even">
<td align="left">RNA polymerase</td>
<td align="right">30</td>
<td align="left">Down</td>
<td align="right">0.0003424</td>
<td align="right">0.0055352</td>
</tr>
<tr class="odd">
<td align="left">Sphingolipid metabolism</td>
<td align="right">43</td>
<td align="left">Up</td>
<td align="right">0.0004647</td>
<td align="right">0.0071173</td>
</tr>
<tr class="even">
<td align="left">Mucin type O-Glycan biosynthesis</td>
<td align="right">22</td>
<td align="left">Up</td>
<td align="right">0.0006637</td>
<td align="right">0.0095882</td>
</tr>
</tbody>
</table>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-35-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
</div>
</div>
<div id="differential-expression-analysis-of-only-brain-development-genes" class="section level1">
<h1>Differential expression analysis of ONLY brain development genes</h1>
<p>Test only the specialised brain development gene set.</p>
<table>
<thead>
<tr class="header">
<th align="left">BRAIN_DEV</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AC139768.1</td>
</tr>
<tr class="even">
<td align="left">ADGRG1</td>
</tr>
<tr class="odd">
<td align="left">AFF2</td>
</tr>
<tr class="even">
<td align="left">ALK</td>
</tr>
<tr class="odd">
<td align="left">ALX1</td>
</tr>
<tr class="even">
<td align="left">BPTF</td>
</tr>
<tr class="odd">
<td align="left">CDK5R1</td>
</tr>
<tr class="even">
<td align="left">CEP290</td>
</tr>
<tr class="odd">
<td align="left">CLN5</td>
</tr>
<tr class="even">
<td align="left">CNTN4</td>
</tr>
<tr class="odd">
<td align="left">CTNS</td>
</tr>
<tr class="even">
<td align="left">DLX2</td>
</tr>
<tr class="odd">
<td align="left">DMBX1</td>
</tr>
<tr class="even">
<td align="left">DSCAML1</td>
</tr>
<tr class="odd">
<td align="left">ECE2</td>
</tr>
<tr class="even">
<td align="left">EGR2</td>
</tr>
<tr class="odd">
<td align="left">FOXG1</td>
</tr>
<tr class="even">
<td align="left">FOXP2</td>
</tr>
<tr class="odd">
<td align="left">GLI2</td>
</tr>
<tr class="even">
<td align="left">HESX1</td>
</tr>
<tr class="odd">
<td align="left">LHX6</td>
</tr>
<tr class="even">
<td align="left">MAP1S</td>
</tr>
<tr class="odd">
<td align="left">MDGA1</td>
</tr>
<tr class="even">
<td align="left">MYO16</td>
</tr>
<tr class="odd">
<td align="left">NCOA6</td>
</tr>
<tr class="even">
<td align="left">NDUFS4</td>
</tr>
<tr class="odd">
<td align="left">NF1</td>
</tr>
<tr class="even">
<td align="left">NKX2-2</td>
</tr>
<tr class="odd">
<td align="left">NNAT</td>
</tr>
<tr class="even">
<td align="left">OTX2</td>
</tr>
<tr class="odd">
<td align="left">PBX1</td>
</tr>
<tr class="even">
<td align="left">PBX3</td>
</tr>
<tr class="odd">
<td align="left">PBX4</td>
</tr>
<tr class="even">
<td align="left">PCDH18</td>
</tr>
<tr class="odd">
<td align="left">PHGDH</td>
</tr>
<tr class="even">
<td align="left">PITPNM1</td>
</tr>
<tr class="odd">
<td align="left">PPT1</td>
</tr>
<tr class="even">
<td align="left">ROBO2</td>
</tr>
<tr class="odd">
<td align="left">SHH</td>
</tr>
<tr class="even">
<td align="left">SHROOM2</td>
</tr>
<tr class="odd">
<td align="left">SHROOM4</td>
</tr>
<tr class="even">
<td align="left">SIX3</td>
</tr>
<tr class="odd">
<td align="left">SLIT1</td>
</tr>
<tr class="even">
<td align="left">SMARCA1</td>
</tr>
<tr class="odd">
<td align="left">TBR1</td>
</tr>
<tr class="even">
<td align="left">UBE3A</td>
</tr>
<tr class="odd">
<td align="left">UNC5C</td>
</tr>
<tr class="even">
<td align="left">UTP3</td>
</tr>
<tr class="odd">
<td align="left">VCX3A</td>
</tr>
<tr class="even">
<td align="left">ZIC1</td>
</tr>
<tr class="odd">
<td align="left">ZIC2</td>
</tr>
</tbody>
</table>
<pre><code>       CMV_status
Down            1
NotSig         33
Up              0</code></pre>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="14%" />
<col width="7%" />
<col width="8%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">gene_id</th>
<th align="left">symbol</th>
<th align="right">entrezid</th>
<th align="right">length</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">LR</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000041515</td>
<td align="left">ENSG00000041515</td>
<td align="left">MYO16</td>
<td align="right">23026</td>
<td align="right">6683.5671</td>
<td align="right">-1.0856429</td>
<td align="right">1.1739934</td>
<td align="right">12.4438219</td>
<td align="right">0.0004194</td>
<td align="right">0.0142588</td>
</tr>
<tr class="even">
<td align="left">ENSG00000053438</td>
<td align="left">ENSG00000053438</td>
<td align="left">NNAT</td>
<td align="right">4826</td>
<td align="right">861.0895</td>
<td align="right">-0.7163499</td>
<td align="right">5.2118804</td>
<td align="right">7.0023215</td>
<td align="right">0.0081404</td>
<td align="right">0.1383869</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000114062</td>
<td align="left">ENSG00000114062</td>
<td align="left">UBE3A</td>
<td align="right">7337</td>
<td align="right">1608.4766</td>
<td align="right">0.4186465</td>
<td align="right">6.1637782</td>
<td align="right">5.4700808</td>
<td align="right">0.0193447</td>
<td align="right">0.2192403</td>
</tr>
<tr class="even">
<td align="left">ENSG00000138083</td>
<td align="left">ENSG00000138083</td>
<td align="left">SIX3</td>
<td align="right">6496</td>
<td align="right">2197.7000</td>
<td align="right">0.6504864</td>
<td align="right">0.5033609</td>
<td align="right">3.1992469</td>
<td align="right">0.0736722</td>
<td align="right">0.6262136</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000102038</td>
<td align="left">ENSG00000102038</td>
<td align="left">SMARCA1</td>
<td align="right">6594</td>
<td align="right">4267.7809</td>
<td align="right">-0.2146878</td>
<td align="right">7.0315400</td>
<td align="right">1.7173596</td>
<td align="right">0.1900333</td>
<td align="right">0.8480540</td>
</tr>
<tr class="even">
<td align="left">ENSG00000164690</td>
<td align="left">ENSG00000164690</td>
<td align="left">SHH</td>
<td align="right">6469</td>
<td align="right">2943.7495</td>
<td align="right">0.3260005</td>
<td align="right">3.0862403</td>
<td align="right">1.6438684</td>
<td align="right">0.1997955</td>
<td align="right">0.8480540</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198707</td>
<td align="left">ENSG00000198707</td>
<td align="left">CEP290</td>
<td align="right">80184</td>
<td align="right">8016.6558</td>
<td align="right">-0.2724468</td>
<td align="right">6.1421266</td>
<td align="right">1.5759259</td>
<td align="right">0.2093485</td>
<td align="right">0.8480540</td>
</tr>
<tr class="even">
<td align="left">ENSG00000144619</td>
<td align="left">ENSG00000144619</td>
<td align="left">CNTN4</td>
<td align="right">152330</td>
<td align="right">2544.5782</td>
<td align="right">0.4015377</td>
<td align="right">2.3393533</td>
<td align="right">1.4955268</td>
<td align="right">0.2213609</td>
<td align="right">0.8480540</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000145194</td>
<td align="left">ENSG00000145194</td>
<td align="left">ECE2</td>
<td align="right">9718</td>
<td align="right">527.0225</td>
<td align="right">0.4001688</td>
<td align="right">0.9565955</td>
<td align="right">1.3700872</td>
<td align="right">0.2417969</td>
<td align="right">0.8480540</td>
</tr>
<tr class="even">
<td align="left">ENSG00000102805</td>
<td align="left">ENSG00000102805</td>
<td align="left">CLN5</td>
<td align="right">1203</td>
<td align="right">2528.9001</td>
<td align="right">-0.2611666</td>
<td align="right">3.1638053</td>
<td align="right">1.3265067</td>
<td align="right">0.2494276</td>
<td align="right">0.8480540</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000132467</td>
<td align="left">ENSG00000132467</td>
<td align="left">UTP3</td>
<td align="right">57050</td>
<td align="right">1982.5395</td>
<td align="right">0.1517575</td>
<td align="right">4.4173967</td>
<td align="right">0.9997956</td>
<td align="right">0.3173600</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000131238</td>
<td align="left">ENSG00000131238</td>
<td align="left">PPT1</td>
<td align="right">5538</td>
<td align="right">1157.4758</td>
<td align="right">0.9384843</td>
<td align="right">0.7635551</td>
<td align="right">0.8517706</td>
<td align="right">0.3560519</td>
<td align="right">0.9117897</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000176749</td>
<td align="left">ENSG00000176749</td>
<td align="left">CDK5R1</td>
<td align="right">8851</td>
<td align="right">3275.1560</td>
<td align="right">0.2394790</td>
<td align="right">3.4726362</td>
<td align="right">0.7958190</td>
<td align="right">0.3723464</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000196712</td>
<td align="left">ENSG00000196712</td>
<td align="left">NF1</td>
<td align="right">4763</td>
<td align="right">6477.5288</td>
<td align="right">0.1415020</td>
<td align="right">7.4871013</td>
<td align="right">0.5563889</td>
<td align="right">0.4557189</td>
<td align="right">0.9117897</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000171634</td>
<td align="left">ENSG00000171634</td>
<td align="left">BPTF</td>
<td align="right">2186</td>
<td align="right">4509.7218</td>
<td align="right">0.1200236</td>
<td align="right">8.4523279</td>
<td align="right">0.5204967</td>
<td align="right">0.4706299</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000122877</td>
<td align="left">ENSG00000122877</td>
<td align="left">EGR2</td>
<td align="right">1959</td>
<td align="right">2331.7450</td>
<td align="right">0.3158335</td>
<td align="right">0.9425032</td>
<td align="right">0.4877723</td>
<td align="right">0.4849227</td>
<td align="right">0.9117897</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000128573</td>
<td align="left">ENSG00000128573</td>
<td align="left">FOXP2</td>
<td align="right">93986</td>
<td align="right">3596.5898</td>
<td align="right">-0.1432209</td>
<td align="right">5.5491752</td>
<td align="right">0.4844851</td>
<td align="right">0.4863977</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000167081</td>
<td align="left">ENSG00000167081</td>
<td align="left">PBX3</td>
<td align="right">5090</td>
<td align="right">2140.4376</td>
<td align="right">-0.1510205</td>
<td align="right">4.2710581</td>
<td align="right">0.4603749</td>
<td align="right">0.4974488</td>
<td align="right">0.9117897</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000205336</td>
<td align="left">ENSG00000205336</td>
<td align="left">ADGRG1</td>
<td align="right">9289</td>
<td align="right">1241.4825</td>
<td align="right">0.1724544</td>
<td align="right">3.9344214</td>
<td align="right">0.4204266</td>
<td align="right">0.5167243</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000146950</td>
<td align="left">ENSG00000146950</td>
<td align="left">SHROOM2</td>
<td align="right">357</td>
<td align="right">6185.4230</td>
<td align="right">-0.1043625</td>
<td align="right">5.4334478</td>
<td align="right">0.3477305</td>
<td align="right">0.5554007</td>
<td align="right">0.9117897</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000185630</td>
<td align="left">ENSG00000185630</td>
<td align="left">PBX1</td>
<td align="right">5087</td>
<td align="right">4342.9472</td>
<td align="right">-0.0948849</td>
<td align="right">8.2752791</td>
<td align="right">0.3342551</td>
<td align="right">0.5631642</td>
<td align="right">0.9117897</td>
</tr>
<tr class="even">
<td align="left">ENSG00000155966</td>
<td align="left">ENSG00000155966</td>
<td align="left">AFF2</td>
<td align="right">2334</td>
<td align="right">5311.3446</td>
<td align="right">0.2241587</td>
<td align="right">1.6561692</td>
<td align="right">0.2528811</td>
<td align="right">0.6150537</td>
<td align="right">0.9392828</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000185008</td>
<td align="left">ENSG00000185008</td>
<td align="left">ROBO2</td>
<td align="right">6092</td>
<td align="right">4746.8103</td>
<td align="right">0.1081623</td>
<td align="right">3.5967231</td>
<td align="right">0.2248126</td>
<td align="right">0.6353972</td>
<td align="right">0.9392828</td>
</tr>
<tr class="even">
<td align="left">ENSG00000105717</td>
<td align="left">ENSG00000105717</td>
<td align="left">PBX4</td>
<td align="right">80714</td>
<td align="right">836.1420</td>
<td align="right">0.1225477</td>
<td align="right">0.7072889</td>
<td align="right">0.1515219</td>
<td align="right">0.6970852</td>
<td align="right">0.9695678</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000158352</td>
<td align="left">ENSG00000158352</td>
<td align="left">SHROOM4</td>
<td align="right">57477</td>
<td align="right">8059.7769</td>
<td align="right">-0.0596539</td>
<td align="right">4.9507195</td>
<td align="right">0.1221758</td>
<td align="right">0.7266865</td>
<td align="right">0.9695678</td>
</tr>
<tr class="even">
<td align="left">ENSG00000074047</td>
<td align="left">ENSG00000074047</td>
<td align="left">GLI2</td>
<td align="right">2736</td>
<td align="right">4587.5252</td>
<td align="right">-0.1896013</td>
<td align="right">2.4724415</td>
<td align="right">0.1088701</td>
<td align="right">0.7414342</td>
<td align="right">0.9695678</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000040531</td>
<td align="left">ENSG00000040531</td>
<td align="left">CTNS</td>
<td align="right">1497</td>
<td align="right">1907.7159</td>
<td align="right">-0.0898046</td>
<td align="right">1.9196361</td>
<td align="right">0.0420346</td>
<td align="right">0.8375539</td>
<td align="right">0.9728754</td>
</tr>
<tr class="even">
<td align="left">ENSG00000164258</td>
<td align="left">ENSG00000164258</td>
<td align="left">NDUFS4</td>
<td align="right">4724</td>
<td align="right">489.0870</td>
<td align="right">0.0240181</td>
<td align="right">4.3076250</td>
<td align="right">0.0205422</td>
<td align="right">0.8860332</td>
<td align="right">0.9728754</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000189184</td>
<td align="left">ENSG00000189184</td>
<td align="left">PCDH18</td>
<td align="right">54510</td>
<td align="right">4560.4925</td>
<td align="right">0.0572759</td>
<td align="right">0.5020654</td>
<td align="right">0.0162954</td>
<td align="right">0.8984231</td>
<td align="right">0.9728754</td>
</tr>
<tr class="even">
<td align="left">ENSG00000110697</td>
<td align="left">ENSG00000110697</td>
<td align="left">PITPNM1</td>
<td align="right">9600</td>
<td align="right">2443.7057</td>
<td align="right">0.0300676</td>
<td align="right">4.9210566</td>
<td align="right">0.0159811</td>
<td align="right">0.8994023</td>
<td align="right">0.9728754</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000182168</td>
<td align="left">ENSG00000182168</td>
<td align="left">UNC5C</td>
<td align="right">8633</td>
<td align="right">4360.0385</td>
<td align="right">0.0375956</td>
<td align="right">0.4507718</td>
<td align="right">0.0055116</td>
<td align="right">0.9408193</td>
<td align="right">0.9728754</td>
</tr>
<tr class="even">
<td align="left">ENSG00000092621</td>
<td align="left">ENSG00000092621</td>
<td align="left">PHGDH</td>
<td align="right">26227</td>
<td align="right">580.5296</td>
<td align="right">0.0432141</td>
<td align="right">2.2656611</td>
<td align="right">0.0053947</td>
<td align="right">0.9414490</td>
<td align="right">0.9728754</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198646</td>
<td align="left">ENSG00000198646</td>
<td align="left">NCOA6</td>
<td align="right">23054</td>
<td align="right">6662.2644</td>
<td align="right">-0.0089225</td>
<td align="right">6.8887138</td>
<td align="right">0.0048881</td>
<td align="right">0.9442614</td>
<td align="right">0.9728754</td>
</tr>
<tr class="even">
<td align="left">ENSG00000130479</td>
<td align="left">ENSG00000130479</td>
<td align="left">MAP1S</td>
<td align="right">55201</td>
<td align="right">2163.7981</td>
<td align="right">0.0067983</td>
<td align="right">4.3150671</td>
<td align="right">0.0011140</td>
<td align="right">0.9733739</td>
<td align="right">0.9733739</td>
</tr>
</tbody>
</table>
<p>The following plots show the expression of the top 9 genes from the brain development set as ranked by their differential expression with regard to CMV positive and CMV negative status.</p>
<p><img src="figure/salmon-RUV-all.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Although the effective library sizes were lower than hoped, the data is generally of good quality. We found at total of 1 differentially expressed genes at FDR &lt; 0.05. The significant genes were enriched for GO terms associated with interferon response and similar. Further gene set testing results indicate an upregulation of interferon response genes in the CMV positive samples, relative to the CMV negative samples, which is consistent with the top genes from the DE analysis.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] graphite_1.36.0             org.Hs.eg.db_3.12.0        
 [3] RColorBrewer_1.1-2          ReactomePA_1.34.0          
 [5] tximport_1.18.0             RUVSeq_1.24.0              
 [7] EDASeq_2.24.0               ShortRead_1.48.0           
 [9] GenomicAlignments_1.26.0    SummarizedExperiment_1.20.0
[11] MatrixGenerics_1.2.1        matrixStats_0.59.0         
[13] Rsamtools_2.6.0             Biostrings_2.58.0          
[15] XVector_0.30.0              BiocParallel_1.24.1        
[17] EGSEA_1.18.1                pathview_1.30.1            
[19] topGO_2.42.0                SparseM_1.78               
[21] GO.db_3.12.1                graph_1.68.0               
[23] gage_2.40.1                 patchwork_1.1.1            
[25] NMF_0.23.0                  cluster_2.1.0              
[27] rngtools_1.5                pkgmaker_0.32.2            
[29] registry_0.5-1              edgeR_3.32.1               
[31] limma_3.46.0                EnsDb.Hsapiens.v86_2.99.0  
[33] ensembldb_2.14.0            AnnotationFilter_1.14.0    
[35] GenomicFeatures_1.42.1      AnnotationDbi_1.52.0       
[37] Biobase_2.50.0              GenomicRanges_1.42.0       
[39] GenomeInfoDb_1.26.7         IRanges_2.24.1             
[41] S4Vectors_0.28.1            BiocGenerics_0.36.1        
[43] forcats_0.5.1               stringr_1.4.0              
[45] dplyr_1.0.4                 purrr_0.3.4                
[47] readr_1.4.0                 tidyr_1.1.2                
[49] tibble_3.1.2                ggplot2_3.3.5              
[51] tidyverse_1.3.0             here_1.0.1                 
[53] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3           rtracklayer_1.50.0       Glimma_2.0.0            
  [4] R.methodsS3_1.8.1        bit64_4.0.5              knitr_1.31              
  [7] R.utils_2.10.1           aroma.light_3.20.0       multcomp_1.4-16         
 [10] DelayedArray_0.16.3      data.table_1.13.6        hwriter_1.3.2           
 [13] KEGGREST_1.30.1          RCurl_1.98-1.3           doParallel_1.0.16       
 [16] generics_0.1.0           metap_1.4                org.Mm.eg.db_3.12.0     
 [19] cowplot_1.1.1            TH.data_1.0-10           RSQLite_2.2.5           
 [22] shadowtext_0.0.7         enrichplot_1.10.2        bit_4.0.4               
 [25] mutoss_0.1-12            xml2_1.3.2               lubridate_1.7.9.2       
 [28] httpuv_1.5.5             assertthat_0.2.1         viridis_0.5.1           
 [31] xfun_0.23                hms_1.0.0                evaluate_0.14           
 [34] promises_1.2.0.1         fansi_0.5.0              progress_1.2.2          
 [37] caTools_1.18.1           dbplyr_2.1.0             readxl_1.3.1            
 [40] igraph_1.2.6             Rgraphviz_2.34.0         DBI_1.1.1               
 [43] geneplotter_1.68.0       tmvnsim_1.0-2            htmlwidgets_1.5.3       
 [46] ellipsis_0.3.2           backports_1.2.1          annotate_1.68.0         
 [49] PADOG_1.32.0             gbRd_0.4-11              gridBase_0.4-7          
 [52] biomaRt_2.46.3           HTMLUtils_0.1.7          vctrs_0.3.8             
 [55] cachem_1.0.4             withr_2.4.2              ggforce_0.3.2           
 [58] globaltest_5.44.0        checkmate_2.0.0          prettyunits_1.1.1       
 [61] mnormt_2.0.2             DOSE_3.16.0              lazyeval_0.2.2          
 [64] crayon_1.4.1             genefilter_1.72.1        labeling_0.4.2          
 [67] pkgconfig_2.0.3          tweenr_1.0.1             nlme_3.1-152            
 [70] ProtGenerics_1.22.0      GSA_1.03.1               rlang_0.4.11            
 [73] lifecycle_1.0.0          sandwich_3.0-0           BiocFileCache_1.14.0    
 [76] mathjaxr_1.2-0           modelr_0.1.8             polyclip_1.10-0         
 [79] cellranger_1.1.0         rprojroot_2.0.2          GSVA_1.38.2             
 [82] Matrix_1.3-2             zoo_1.8-8                reprex_1.0.0            
 [85] whisker_0.4              png_0.1-7                viridisLite_0.4.0       
 [88] bitops_1.0-7             R.oo_1.24.0              KernSmooth_2.23-18      
 [91] blob_1.2.1               R2HTML_2.3.2             doRNG_1.8.2             
 [94] qvalue_2.22.0            jpeg_0.1-8.1             reactome.db_1.74.0      
 [97] scales_1.1.1             memoise_2.0.0            GSEABase_1.52.1         
[100] magrittr_2.0.1           plyr_1.8.6               safe_3.30.0             
[103] gplots_3.1.1             zlibbioc_1.36.0          scatterpie_0.1.5        
[106] compiler_4.0.2           plotrix_3.8-1            KEGGgraph_1.50.0        
[109] DESeq2_1.30.1            cli_3.0.0                EGSEAdata_1.18.0        
[112] MASS_7.3-53.1            tidyselect_1.1.0         stringi_1.5.3           
[115] highr_0.8                GOSemSim_2.16.1          yaml_2.2.1              
[118] askpass_1.1              locfit_1.5-9.4           ggrepel_0.9.1           
[121] latticeExtra_0.6-29      grid_4.0.2               fastmatch_1.1-0         
[124] tools_4.0.2              rstudioapi_0.13          foreach_1.5.1           
[127] git2r_0.28.0             gridExtra_2.3            farver_2.1.0            
[130] ggraph_2.0.4             BiocManager_1.30.10      rvcheck_0.1.8           
[133] digest_0.6.27            Rcpp_1.0.6               broom_0.7.4             
[136] later_1.1.0.1            httr_1.4.2               Rdpack_2.1              
[139] colorspace_2.0-2         rvest_0.3.6              XML_3.99-0.5            
[142] fs_1.5.0                 splines_4.0.2            sn_1.6-2                
[145] graphlayouts_0.7.1       multtest_2.46.0          plotly_4.9.3            
[148] xtable_1.8-4             jsonlite_1.7.2           tidygraph_1.2.0         
[151] R6_2.5.0                 TFisher_0.2.0            KEGGdzPathwaysGEO_1.28.0
[154] pillar_1.6.1             htmltools_0.5.1.1        hgu133plus2.db_3.2.3    
[157] glue_1.4.2               fastmap_1.1.0            DT_0.17                 
[160] codetools_0.2-18         fgsea_1.16.0             mvtnorm_1.1-1           
[163] utf8_1.2.1               lattice_0.20-41          numDeriv_2016.8-1.1     
[166] hgu133a.db_3.2.3         curl_4.3                 gtools_3.8.2            
[169] openssl_1.4.3            survival_3.2-7           rmarkdown_2.6           
[172] org.Rn.eg.db_3.12.0      munsell_0.5.0            DO.db_2.9               
[175] GenomeInfoDbData_1.2.4   iterators_1.0.13         haven_2.3.1             
[178] reshape2_1.4.4           gtable_0.3.0             rbibutils_2.0           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
