<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Transcript Usage (DTU) Analysis: DRIMseq</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential Transcript Usage (DTU) Analysis: DRIMseq</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree40bcb2a486c430df24ae726cf36e84e578b188c7targetblank40bcb2aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/40bcb2a486c430df24ae726cf36e84e578b188c7" target="_blank">40bcb2a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatree40bcb2a486c430df24ae726cf36e84e578b188c7targetblank40bcb2aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/40bcb2a486c430df24ae726cf36e84e578b188c7" target="_blank">40bcb2a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    analysis/old_analysis/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/rds/
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruv-all-gsea-c2.csv
    Ignored:    output/salmon-ruv-all-gsea-c5.csv
    Ignored:    output/salmon-ruv-all.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/salmon-DRIMseq.Rmd</code>) and HTML (<code>docs/salmon-DRIMseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/40bcb2a486c430df24ae726cf36e84e578b188c7/analysis/salmon-DRIMseq.Rmd" target="_blank">40bcb2a</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(“analysis/salmon-DRIMseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/7ca0e36a09f6dd079d1d9713e1d2b830384e9edd/docs/salmon-DRIMseq.html" target="_blank">7ca0e36</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/63282589d25b165c63f636b0405ded24e5a200bb/analysis/salmon-DRIMseq.Rmd" target="_blank">6328258</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
<td>
wflow_publish(c(“analysis/salmon-BANDITS.Rmd”, “analysis/salmon-DRIMseq.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/631fbebf116d344279db6e3c5a1b82a0d8792bcf/docs/salmon-DRIMseq.html" target="_blank">631fbeb</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/ed601692a43bf148e82949815a8d24a6e438eab1/analysis/salmon-DRIMseq.Rmd" target="_blank">ed60169</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-14
</td>
<td>
wflow_publish(“analysis/salmon-DRIMseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/salmon-DRIMseq.html" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
</tbody>
</table>
<p>Only retain paired samples with clinical information for donstream analysis.</p>
<pre class="r"><code>int &lt;- intersect(names(files), targets$id)
targets &lt;- targets[match(int, targets$id),]
files &lt;- files[match(int, names(files))]

files</code></pre>
<pre><code>                                                                                                                                                                                           CMV30 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV31 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV8 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV9 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV26 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV14 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV15 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV20 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV21 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV1 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV2 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV3 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                            CMV4 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV10 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV11 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV19 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV35 
&quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV51 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV52 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV53 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV54 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV56 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV57 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV58 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV60 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; 
                                                                                                                                                                                           CMV61 
 &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/lisa.hui/amnio-cell-free-RNA/data/gene-transcriptome-analysis/quants/200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000/quant.sf&quot; </code></pre>
<p>Read in transcript count data and remove transcripts with no counts.</p>
<pre class="r"><code>txi &lt;- tximport(files, type = &quot;salmon&quot;, txOut = TRUE)</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 </code></pre>
<pre class="r"><code>cts &lt;- txi$counts
cts &lt;- cts[rowSums(cts) &gt; 0,]

dim(cts)</code></pre>
<pre><code>[1] 155749     26</code></pre>
<p>Plot effective library size per sample.</p>
<pre class="r"><code>colSums(cts) %&gt;%
  data.frame %&gt;%
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  dplyr::rename(&quot;libsize&quot; = &quot;.&quot;) %&gt;%
ggplot(aes(x = sample, y = libsize)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))</code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/4868a17b4ee59740dda04fbacccbe73e85dacf1d/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-6-1.png" target="_blank">4868a17</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Associate transcripts with gene IDs for gene-level summarization. Filter out genes without Entrez IDs, genes with only one transcript isoform and transcripts without any counts across all samples.</p>
<pre class="r"><code>txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene
txdf &lt;- select(txdb, keys(txdb, keytype = &quot;TXNAME&quot;), 
               &quot;GENEID&quot;, &quot;TXNAME&quot;, &quot;SYMBOL&quot;) %&gt;%
  drop_na(GENEID)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>txdf %&gt;% dplyr::inner_join(select(org.Hs.eg.db, 
                                  keys = unique(txdf$GENEID), 
                           columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
                           keytype=&quot;ENTREZID&quot;), 
                           by = c(&quot;GENEID&quot; = &quot;ENTREZID&quot;)) %&gt;%
  dplyr::distinct() %&gt;%
  dplyr::filter((duplicated(GENEID) | 
                  duplicated(GENEID, fromLast = TRUE)) &amp;
                  TXNAME %in% rownames(cts)) -&gt; txdf</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>head(txdf, n = 10) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">TXNAME</th>
<th align="left">GENEID</th>
<th align="left">SYMBOL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000623808.3</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000445118.7</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000669922.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000670780.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000667414.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000666741.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000623070.4</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000659124.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">ENST00000658846.1</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">ENST00000608189.5</td>
<td align="left">643837</td>
<td align="left">LINC01128</td>
</tr>
</tbody>
</table>
<p>Match transcript count data to transcript annotation table.</p>
<pre class="r"><code>cts &lt;- cts[match(txdf$TXNAME, rownames(cts)),]

head(cts) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">CMV30</th>
<th align="right">CMV31</th>
<th align="right">CMV8</th>
<th align="right">CMV9</th>
<th align="right">CMV26</th>
<th align="right">CMV14</th>
<th align="right">CMV15</th>
<th align="right">CMV20</th>
<th align="right">CMV21</th>
<th align="right">CMV1</th>
<th align="right">CMV2</th>
<th align="right">CMV3</th>
<th align="right">CMV4</th>
<th align="right">CMV10</th>
<th align="right">CMV11</th>
<th align="right">CMV19</th>
<th align="right">CMV35</th>
<th align="right">CMV51</th>
<th align="right">CMV52</th>
<th align="right">CMV53</th>
<th align="right">CMV54</th>
<th align="right">CMV56</th>
<th align="right">CMV57</th>
<th align="right">CMV58</th>
<th align="right">CMV60</th>
<th align="right">CMV61</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000623808.3</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">2.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">2.840</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.113</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000445118.7</td>
<td align="right">81.080</td>
<td align="right">49.139</td>
<td align="right">51.049</td>
<td align="right">35.215</td>
<td align="right">40.649</td>
<td align="right">71.627</td>
<td align="right">112.693</td>
<td align="right">72.270</td>
<td align="right">88.424</td>
<td align="right">23.261</td>
<td align="right">36.969</td>
<td align="right">3.356</td>
<td align="right">25.536</td>
<td align="right">88.967</td>
<td align="right">15.132</td>
<td align="right">3.924</td>
<td align="right">51.927</td>
<td align="right">39.892</td>
<td align="right">80.193</td>
<td align="right">45.610</td>
<td align="right">48.656</td>
<td align="right">85.781</td>
<td align="right">95.585</td>
<td align="right">155.849</td>
<td align="right">89.77</td>
<td align="right">52.703</td>
</tr>
<tr class="odd">
<td align="left">ENST00000669922.1</td>
<td align="right">7.952</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">37.732</td>
<td align="right">7.738</td>
<td align="right">0.000</td>
<td align="right">6.574</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.538</td>
<td align="right">0.000</td>
<td align="right">18.000</td>
<td align="right">0.000</td>
<td align="right">11.000</td>
<td align="right">3.808</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000670780.1</td>
<td align="right">15.481</td>
<td align="right">22.471</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">24.094</td>
<td align="right">14.058</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">4.001</td>
<td align="right">0.000</td>
<td align="right">15.947</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">5.814</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">16.986</td>
<td align="right">11.345</td>
<td align="right">0.000</td>
<td align="right">1.836</td>
<td align="right">0.000</td>
<td align="right">6.21</td>
<td align="right">20.954</td>
</tr>
<tr class="odd">
<td align="left">ENST00000667414.1</td>
<td align="right">0.000</td>
<td align="right">1.155</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.196</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">1.049</td>
<td align="right">1.436</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ENST00000666741.1</td>
<td align="right">5.858</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">37.606</td>
<td align="right">7.820</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">21.911</td>
<td align="right">16.865</td>
<td align="right">0.000</td>
<td align="right">49.674</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">39.220</td>
<td align="right">25.097</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.00</td>
<td align="right">0.000</td>
</tr>
</tbody>
</table>
</div>
<div id="drimseq-analysis" class="section level1">
<h1>DRIMseq analysis</h1>
<p>Filter transcripts using <em>BANDITS</em> strategy.</p>
<pre class="r"><code>transcripts_to_keep &lt;- filter_transcripts(gene_to_transcript = txdf[, c(2,1)],
                                          transcript_counts = cts, 
                                          min_transcript_proportion = 0.01,
                                          min_transcript_counts = min(table(targets$CMV_status)), 
                                          min_gene_counts = ncol(cts))</code></pre>
<pre><code>After filtering, 50.63% of transcripts are kept</code></pre>
<pre class="r"><code>length(transcripts_to_keep)</code></pre>
<pre><code>[1] 65831</code></pre>
<p>Keep only transcripts that pass filtering criteria.</p>
<pre class="r"><code>keep &lt;- match(transcripts_to_keep, rownames(cts))

cts &lt;- cts[keep,]
txdf &lt;- txdf[keep,]

cts %&gt;% data.frame %&gt;%
  rownames_to_column(var = &quot;feature_id&quot;) %&gt;% 
  mutate(gene_id = txdf$GENEID) %&gt;%
  relocate(gene_id, .before = feature_id) -&gt; cts

d &lt;- dmDSdata(counts = cts, 
              samples = targets %&gt;% dplyr::select(id, CMV_status) %&gt;%
                dplyr::rename(sample_id = id,
                              condition = CMV_status) %&gt;% data.frame)
d</code></pre>
<pre><code>An object of class dmDSdata 
with 16340 genes and 26 samples
* data accessors: counts(), samples()</code></pre>
<p>Setup the design matrix.</p>
<pre class="r"><code>design_full &lt;- model.matrix(~ condition, 
                            data = DRIMSeq::samples(d))
colnames(design_full)</code></pre>
<pre><code>[1] &quot;(Intercept)&quot;  &quot;conditionneg&quot;</code></pre>
<p>Perform DTU analysis.</p>
<pre class="r"><code>out &lt;- here(&quot;data/rds/dmResults.rds&quot;)

if(!file.exists(out)){  
set.seed(1)
  d &lt;- dmPrecision(d, design = design_full)
  d &lt;- dmFit(d, design = design_full)
  d &lt;- dmTest(d, coef = &quot;conditionneg&quot;)
  
  saveRDS(d, file = out)
  
} else{
  d &lt;- readRDS(out)
  
}</code></pre>
<p>Extract the results by gene.</p>
<pre class="r"><code>res &lt;- DRIMSeq::results(d)
head(res) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">gene_id</th>
<th align="right">lr</th>
<th align="right">df</th>
<th align="right">pvalue</th>
<th align="right">adj_pvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">0.0479528</td>
<td align="right">1</td>
<td align="right">0.8266647</td>
<td align="right">0.9769147</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="right">0.1425669</td>
<td align="right">1</td>
<td align="right">0.7057423</td>
<td align="right">0.9511446</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="right">0.5008255</td>
<td align="right">2</td>
<td align="right">0.7784794</td>
<td align="right">0.9677716</td>
</tr>
<tr class="even">
<td align="left">1000</td>
<td align="right">1.6295483</td>
<td align="right">3</td>
<td align="right">0.6527086</td>
<td align="right">0.9401655</td>
</tr>
<tr class="odd">
<td align="left">10000</td>
<td align="right">3.6500405</td>
<td align="right">7</td>
<td align="right">0.8190873</td>
<td align="right">0.9753751</td>
</tr>
<tr class="even">
<td align="left">10001</td>
<td align="right">6.5342813</td>
<td align="right">3</td>
<td align="right">0.0883203</td>
<td align="right">0.6590604</td>
</tr>
</tbody>
</table>
<p>Extract the results by transcript.</p>
<pre class="r"><code>res.txp &lt;- DRIMSeq::results(d, level=&quot;feature&quot;)
head(res.txp) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">gene_id</th>
<th align="left">feature_id</th>
<th align="right">lr</th>
<th align="right">df</th>
<th align="right">pvalue</th>
<th align="right">adj_pvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">ENST00000598345.1</td>
<td align="right">0.0479528</td>
<td align="right">1</td>
<td align="right">0.8266647</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">ENST00000599109.5</td>
<td align="right">0.0479528</td>
<td align="right">1</td>
<td align="right">0.8266647</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">ENST00000286479.4</td>
<td align="right">0.1425669</td>
<td align="right">1</td>
<td align="right">0.7057423</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">ENST00000520116.1</td>
<td align="right">0.1425669</td>
<td align="right">1</td>
<td align="right">0.7057423</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">100</td>
<td align="left">ENST00000372874.9</td>
<td align="right">0.1383652</td>
<td align="right">1</td>
<td align="right">0.7099116</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">100</td>
<td align="left">ENST00000537820.1</td>
<td align="right">0.2421833</td>
<td align="right">1</td>
<td align="right">0.6226337</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>Convert any NAs in the pvalue column into 1’s to avoid problems in downstream analysis.</p>
<pre class="r"><code>no.na &lt;- function(x) ifelse(is.na(x), 1, x)
res$pvalue &lt;- no.na(res$pvalue)
res.txp$pvalue &lt;- no.na(res.txp$pvalue)</code></pre>
<p>Identify genes with statistically significant DTU.</p>
<pre class="r"><code>idx &lt;- which(res$adj_pvalue &lt; 0.05)
res[idx,] %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">gene_id</th>
<th align="right">lr</th>
<th align="right">df</th>
<th align="right">pvalue</th>
<th align="right">adj_pvalue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">716</td>
<td align="left">101060226</td>
<td align="right">30.04312</td>
<td align="right">5</td>
<td align="right">1.45e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">796</td>
<td align="left">10184</td>
<td align="right">32.33891</td>
<td align="right">4</td>
<td align="right">1.60e-06</td>
<td align="right">0.0057248</td>
</tr>
<tr class="odd">
<td align="left">1104</td>
<td align="left">102546294</td>
<td align="right">25.33861</td>
<td align="right">4</td>
<td align="right">4.30e-05</td>
<td align="right">0.0252655</td>
</tr>
<tr class="even">
<td align="left">2007</td>
<td align="left">11188</td>
<td align="right">32.25751</td>
<td align="right">4</td>
<td align="right">1.70e-06</td>
<td align="right">0.0057248</td>
</tr>
<tr class="odd">
<td align="left">3555</td>
<td align="left">160518</td>
<td align="right">21.39404</td>
<td align="right">3</td>
<td align="right">8.72e-05</td>
<td align="right">0.0420753</td>
</tr>
<tr class="even">
<td align="left">3725</td>
<td align="left">168850</td>
<td align="right">23.60048</td>
<td align="right">4</td>
<td align="right">9.60e-05</td>
<td align="right">0.0432552</td>
</tr>
<tr class="odd">
<td align="left">4424</td>
<td align="left">222068</td>
<td align="right">23.92006</td>
<td align="right">3</td>
<td align="right">2.60e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">5570</td>
<td align="left">26040</td>
<td align="right">15.31150</td>
<td align="right">1</td>
<td align="right">9.12e-05</td>
<td align="right">0.0424742</td>
</tr>
<tr class="odd">
<td align="left">5710</td>
<td align="left">26354</td>
<td align="right">20.63181</td>
<td align="right">2</td>
<td align="right">3.31e-05</td>
<td align="right">0.0224361</td>
</tr>
<tr class="even">
<td align="left">5760</td>
<td align="left">266655</td>
<td align="right">21.32351</td>
<td align="right">2</td>
<td align="right">2.34e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="odd">
<td align="left">5834</td>
<td align="left">27067</td>
<td align="right">35.65270</td>
<td align="right">8</td>
<td align="right">2.03e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">5878</td>
<td align="left">27151</td>
<td align="right">47.37643</td>
<td align="right">7</td>
<td align="right">0.00e+00</td>
<td align="right">0.0003184</td>
</tr>
<tr class="odd">
<td align="left">6290</td>
<td align="left">285492</td>
<td align="right">17.65382</td>
<td align="right">1</td>
<td align="right">2.65e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">7724</td>
<td align="left">399687</td>
<td align="right">24.83946</td>
<td align="right">3</td>
<td align="right">1.67e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="odd">
<td align="left">7885</td>
<td align="left">404203</td>
<td align="right">18.90374</td>
<td align="right">2</td>
<td align="right">7.85e-05</td>
<td align="right">0.0408181</td>
</tr>
<tr class="even">
<td align="left">8403</td>
<td align="left">4779</td>
<td align="right">36.49140</td>
<td align="right">8</td>
<td align="right">1.43e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="odd">
<td align="left">9108</td>
<td align="left">51657</td>
<td align="right">30.27356</td>
<td align="right">6</td>
<td align="right">3.49e-05</td>
<td align="right">0.0224361</td>
</tr>
<tr class="even">
<td align="left">9622</td>
<td align="left">54801</td>
<td align="right">16.46457</td>
<td align="right">1</td>
<td align="right">4.96e-05</td>
<td align="right">0.0279067</td>
</tr>
<tr class="odd">
<td align="left">9855</td>
<td align="left">55075</td>
<td align="right">27.19514</td>
<td align="right">4</td>
<td align="right">1.82e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">10067</td>
<td align="left">55323</td>
<td align="right">22.37352</td>
<td align="right">3</td>
<td align="right">5.45e-05</td>
<td align="right">0.0294775</td>
</tr>
<tr class="odd">
<td align="left">12223</td>
<td align="left">653145</td>
<td align="right">18.78825</td>
<td align="right">2</td>
<td align="right">8.32e-05</td>
<td align="right">0.0416428</td>
</tr>
<tr class="even">
<td align="left">13104</td>
<td align="left">7528</td>
<td align="right">16.79804</td>
<td align="right">1</td>
<td align="right">4.16e-05</td>
<td align="right">0.0252655</td>
</tr>
<tr class="odd">
<td align="left">13144</td>
<td align="left">7587</td>
<td align="right">24.61990</td>
<td align="right">3</td>
<td align="right">1.85e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">13553</td>
<td align="left">79719</td>
<td align="right">26.63948</td>
<td align="right">4</td>
<td align="right">2.35e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="odd">
<td align="left">13650</td>
<td align="left">79848</td>
<td align="right">27.16367</td>
<td align="right">3</td>
<td align="right">5.40e-06</td>
<td align="right">0.0147013</td>
</tr>
<tr class="even">
<td align="left">13801</td>
<td align="left">80127</td>
<td align="right">42.34932</td>
<td align="right">4</td>
<td align="right">0.00e+00</td>
<td align="right">0.0001908</td>
</tr>
<tr class="odd">
<td align="left">15178</td>
<td align="left">8906</td>
<td align="right">32.43836</td>
<td align="right">7</td>
<td align="right">3.37e-05</td>
<td align="right">0.0224361</td>
</tr>
<tr class="even">
<td align="left">16031</td>
<td align="left">9647</td>
<td align="right">22.00320</td>
<td align="right">2</td>
<td align="right">1.67e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="odd">
<td align="left">16111</td>
<td align="left">9731</td>
<td align="right">23.84608</td>
<td align="right">3</td>
<td align="right">2.69e-05</td>
<td align="right">0.0201915</td>
</tr>
<tr class="even">
<td align="left">16313</td>
<td align="left">9961</td>
<td align="right">25.53827</td>
<td align="right">3</td>
<td align="right">1.19e-05</td>
<td align="right">0.0201915</td>
</tr>
</tbody>
</table>
<pre class="r"><code>plotProportions(d, res$gene_id[idx][1], &quot;condition&quot;)</code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/7ca0e36a09f6dd079d1d9713e1d2b830384e9edd/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-17-1.png" target="_blank">7ca0e36</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="adjust-p-values-with-stager" class="section level1">
<h1>Adjust p-values with stageR</h1>
<p>A typical analysis of differential transcript usage would involve asking first: “which genes contain any evidence of DTU?”, and secondly, “which transcripts in the genes that contain some evidence may be participating in the DTU?” A gene may pass the first stage without exhibiting enough evidence to identify one or more transcripts that are participating in the DTU. The stageR package is designed to allow for such two-stage testing procedures, where the first stage is called a screening stage and the second stage a confirmation stage.</p>
<p>We first construct a vector of p-values for the screening stage.</p>
<pre class="r"><code>pScreen &lt;- setNames(res$pvalue, res$gene_id)</code></pre>
<p>We construct a one column matrix of the confirmation p-values:</p>
<pre class="r"><code>pConfirmation &lt;- matrix(res.txp$pvalue, ncol = 1)
strp &lt;- function(x) substr(x, 1, 15)
rownames(pConfirmation) &lt;- strp(res.txp$feature_id)</code></pre>
<p>We arrange a two column data.frame with the transcript and gene identifiers.</p>
<pre class="r"><code>tx2gene &lt;- res.txp[,c(&quot;feature_id&quot;, &quot;gene_id&quot;)]
tx2gene[,1] &lt;- strp(tx2gene[,1])</code></pre>
<p>The following functions then perform the stageR analysis. We must specify an alpha, which will be the overall false discovery rate target for the analysis, defined below. Unlike typical adjusted p-values or q-values, we cannot choose an arbitrary threshold later: after specifying alpha=0.05, we need to use 5% as the target in downstream steps.</p>
<pre class="r"><code>stageRObj &lt;- stageRTx(pScreen = pScreen, 
                      pConfirmation = pConfirmation,
                      pScreenAdjusted = FALSE, 
                      tx2gene = tx2gene)
stageRObj &lt;- stageWiseAdjustment(stageRObj, method = &quot;dtu&quot;, alpha = 0.05) 
suppressWarnings({
  drim.padj &lt;- getAdjustedPValues(stageRObj, 
                                  order=FALSE,
                                  onlySignificantGenes=TRUE)
})</code></pre>
<pre><code>The returned adjusted p-values are based on a stage-wise testing approach and are only valid for the provided target OFDR level of 5%. If a different target OFDR level is of interest,the entire adjustment should be re-run. </code></pre>
<pre class="r"><code>head(drim.padj) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">geneID</th>
<th align="left">txID</th>
<th align="right">gene</th>
<th align="right">transcript</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">101060226</td>
<td align="left">ENST00000621744</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">ENST00000651566</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">101060226</td>
<td align="left">ENST00000611931</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">ENST00000612881</td>
<td align="right">0.0244175</td>
<td align="right">0.0034786</td>
</tr>
<tr class="odd">
<td align="left">101060226</td>
<td align="left">ENST00000369227</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">ENST00000426482</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>select(org.Hs.eg.db, 
       keys = drim.padj$geneID, 
       columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
       keytype = &quot;ENTREZID&quot;) %&gt;%
  dplyr::distinct() %&gt;% 
  dplyr::inner_join(drim.padj, by = c(&quot;ENTREZID&quot; = &quot;geneID&quot;)) %&gt;%
  dplyr::arrange(gene, transcript) -&gt; top_gene</code></pre>
<pre><code>&#39;select()&#39; returned many:1 mapping between keys and columns</code></pre>
<pre class="r"><code>top_gene %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ENTREZID</th>
<th align="left">SYMBOL</th>
<th align="left">txID</th>
<th align="right">gene</th>
<th align="right">transcript</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000464394</td>
<td align="right">0.0002307</td>
<td align="right">0.0002456</td>
</tr>
<tr class="even">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000463558</td>
<td align="right">0.0002307</td>
<td align="right">0.1343327</td>
</tr>
<tr class="odd">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000394009</td>
<td align="right">0.0002307</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000489323</td>
<td align="right">0.0002307</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">80127</td>
<td align="left">BBOF1</td>
<td align="left">ENST00000492026</td>
<td align="right">0.0002307</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000443236</td>
<td align="right">0.0003851</td>
<td align="right">0.0048779</td>
</tr>
<tr class="odd">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000651564</td>
<td align="right">0.0003851</td>
<td align="right">0.0839855</td>
</tr>
<tr class="even">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000594249</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000388925</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000600235</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000596224</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000602132</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27151</td>
<td align="left">CPAMD8</td>
<td align="left">ENST00000291440</td>
<td align="right">0.0003851</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">10184</td>
<td align="left">LHFPL2</td>
<td align="left">ENST00000380345</td>
<td align="right">0.0069229</td>
<td align="right">0.2299604</td>
</tr>
<tr class="odd">
<td align="left">10184</td>
<td align="left">LHFPL2</td>
<td align="left">ENST00000515007</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">10184</td>
<td align="left">LHFPL2</td>
<td align="left">ENST00000503686</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">10184</td>
<td align="left">LHFPL2</td>
<td align="left">ENST00000512759</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">10184</td>
<td align="left">LHFPL2</td>
<td align="left">ENST00000510949</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">11188</td>
<td align="left">NISCH</td>
<td align="left">ENST00000479054</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">11188</td>
<td align="left">NISCH</td>
<td align="left">ENST00000345716</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">11188</td>
<td align="left">NISCH</td>
<td align="left">ENST00000488380</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">11188</td>
<td align="left">NISCH</td>
<td align="left">ENST00000420808</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">11188</td>
<td align="left">NISCH</td>
<td align="left">ENST00000489895</td>
<td align="right">0.0069229</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">79848</td>
<td align="left">CSPP1</td>
<td align="left">ENST00000519668</td>
<td align="right">0.0177782</td>
<td align="right">0.0002706</td>
</tr>
<tr class="odd">
<td align="left">79848</td>
<td align="left">CSPP1</td>
<td align="left">ENST00000521919</td>
<td align="right">0.0177782</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">79848</td>
<td align="left">CSPP1</td>
<td align="left">ENST00000262210</td>
<td align="right">0.0177782</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">79848</td>
<td align="left">CSPP1</td>
<td align="left">ENST00000519163</td>
<td align="right">0.0177782</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">285492</td>
<td align="left">LINC00955</td>
<td align="left">ENST00000514422</td>
<td align="right">0.0244175</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">285492</td>
<td align="left">LINC00955</td>
<td align="left">ENST00000502775</td>
<td align="right">0.0244175</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000361085</td>
<td align="right">0.0244175</td>
<td align="right">0.0024997</td>
</tr>
<tr class="odd">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000351773</td>
<td align="right">0.0244175</td>
<td align="right">0.0032046</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000612881</td>
<td align="right">0.0244175</td>
<td align="right">0.0034786</td>
</tr>
<tr class="odd">
<td align="left">266655</td>
<td align="left">BRD3OS</td>
<td align="left">ENST00000550853</td>
<td align="right">0.0244175</td>
<td align="right">0.0053333</td>
</tr>
<tr class="even">
<td align="left">9647</td>
<td align="left">PPM1F</td>
<td align="left">ENST00000263212</td>
<td align="right">0.0244175</td>
<td align="right">0.0064470</td>
</tr>
<tr class="odd">
<td align="left">79719</td>
<td align="left">AAGAB</td>
<td align="left">ENST00000558725</td>
<td align="right">0.0244175</td>
<td align="right">0.0237074</td>
</tr>
<tr class="even">
<td align="left">222068</td>
<td align="left">TMED4</td>
<td align="left">ENST00000477639</td>
<td align="right">0.0244175</td>
<td align="right">0.0434705</td>
</tr>
<tr class="odd">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000357402</td>
<td align="right">0.0244175</td>
<td align="right">0.0513760</td>
</tr>
<tr class="even">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000395353</td>
<td align="right">0.0244175</td>
<td align="right">0.0742427</td>
</tr>
<tr class="odd">
<td align="left">9731</td>
<td align="left">CEP104</td>
<td align="left">ENST00000494653</td>
<td align="right">0.0244175</td>
<td align="right">0.0840414</td>
</tr>
<tr class="even">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000583378</td>
<td align="right">0.0244175</td>
<td align="right">0.0980587</td>
</tr>
<tr class="odd">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000524300</td>
<td align="right">0.0244175</td>
<td align="right">0.2241812</td>
</tr>
<tr class="even">
<td align="left">399687</td>
<td align="left">MYO18A</td>
<td align="left">ENST00000628822</td>
<td align="right">0.0244175</td>
<td align="right">0.3494535</td>
</tr>
<tr class="odd">
<td align="left">9647</td>
<td align="left">PPM1F</td>
<td align="left">ENST00000445205</td>
<td align="right">0.0244175</td>
<td align="right">0.6337945</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000621744</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000651566</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000611931</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000369227</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">101060226</td>
<td align="left">NBPF19</td>
<td align="left">ENST00000426482</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">222068</td>
<td align="left">TMED4</td>
<td align="left">ENST00000457408</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">222068</td>
<td align="left">TMED4</td>
<td align="left">ENST00000289577</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">222068</td>
<td align="left">TMED4</td>
<td align="left">ENST00000481238</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">266655</td>
<td align="left">BRD3OS</td>
<td align="left">ENST00000603928</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">266655</td>
<td align="left">BRD3OS</td>
<td align="left">ENST00000432807</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000522695</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000523558</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000521210</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000523533</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000355780</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000519961</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000517542</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">27067</td>
<td align="left">STAU2</td>
<td align="left">ENST00000521447</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">399687</td>
<td align="left">MYO18A</td>
<td align="left">ENST00000527372</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">399687</td>
<td align="left">MYO18A</td>
<td align="left">ENST00000531253</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">399687</td>
<td align="left">MYO18A</td>
<td align="left">ENST00000530254</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000579889</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000362042</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000357480</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000583060</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000361665</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000581441</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000577411</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">4779</td>
<td align="left">NFE2L1</td>
<td align="left">ENST00000582155</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">55075</td>
<td align="left">UACA</td>
<td align="left">ENST00000322954</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">55075</td>
<td align="left">UACA</td>
<td align="left">ENST00000379983</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">55075</td>
<td align="left">UACA</td>
<td align="left">ENST00000539319</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">55075</td>
<td align="left">UACA</td>
<td align="left">ENST00000558758</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">55075</td>
<td align="left">UACA</td>
<td align="left">ENST00000558308</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000638053</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">7587</td>
<td align="left">ZNF37A</td>
<td align="left">ENST00000640275</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">79719</td>
<td align="left">AAGAB</td>
<td align="left">ENST00000261880</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">79719</td>
<td align="left">AAGAB</td>
<td align="left">ENST00000561452</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">79719</td>
<td align="left">AAGAB</td>
<td align="left">ENST00000542650</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">79719</td>
<td align="left">AAGAB</td>
<td align="left">ENST00000560362</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">9647</td>
<td align="left">PPM1F</td>
<td align="left">ENST00000407142</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">9731</td>
<td align="left">CEP104</td>
<td align="left">ENST00000378230</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">9731</td>
<td align="left">CEP104</td>
<td align="left">ENST00000495701</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">9731</td>
<td align="left">CEP104</td>
<td align="left">ENST00000378223</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000563915</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">9961</td>
<td align="left">MVP</td>
<td align="left">ENST00000566859</td>
<td align="right">0.0244175</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">26354</td>
<td align="left">GNL3</td>
<td align="left">ENST00000394799</td>
<td align="right">0.0271319</td>
<td align="right">0.0034324</td>
</tr>
<tr class="odd">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000535852</td>
<td align="right">0.0271319</td>
<td align="right">0.0053290</td>
</tr>
<tr class="even">
<td align="left">26354</td>
<td align="left">GNL3</td>
<td align="left">ENST00000468146</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">26354</td>
<td align="left">GNL3</td>
<td align="left">ENST00000418458</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000248600</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000340062</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000359697</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000438695</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000431581</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000454618</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">51657</td>
<td align="left">STYXL1</td>
<td align="left">ENST00000460184</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000308724</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000397120</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000460049</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000465445</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000557391</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000554477</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">8906</td>
<td align="left">AP1G2</td>
<td align="left">ENST00000556277</td>
<td align="right">0.0271319</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">7528</td>
<td align="left">YY1</td>
<td align="left">ENST00000262238</td>
<td align="right">0.0305535</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">7528</td>
<td align="left">YY1</td>
<td align="left">ENST00000554804</td>
<td align="right">0.0305535</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">102546294</td>
<td align="left">LOC102546294</td>
<td align="left">ENST00000661004</td>
<td align="right">0.0305535</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">102546294</td>
<td align="left">LOC102546294</td>
<td align="left">ENST00000669063</td>
<td align="right">0.0305535</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">102546294</td>
<td align="left">LOC102546294</td>
<td align="left">ENST00000655946</td>
<td align="right">0.0305535</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">102546294</td>
<td align="left">LOC102546294</td>
<td align="left">ENST00000661718</td>
<td align="right">0.0305535</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">102546294</td>
<td align="left">LOC102546294</td>
<td align="left">ENST00000669375</td>
<td align="right">0.0305535</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">54801</td>
<td align="left">HAUS6</td>
<td align="left">ENST00000380502</td>
<td align="right">0.0337475</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">54801</td>
<td align="left">HAUS6</td>
<td align="left">ENST00000380496</td>
<td align="right">0.0337475</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">55323</td>
<td align="left">LARP6</td>
<td align="left">ENST00000559316</td>
<td align="right">0.0356469</td>
<td align="right">0.0653846</td>
</tr>
<tr class="even">
<td align="left">55323</td>
<td align="left">LARP6</td>
<td align="left">ENST00000299213</td>
<td align="right">0.0356469</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">55323</td>
<td align="left">LARP6</td>
<td align="left">ENST00000560052</td>
<td align="right">0.0356469</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">55323</td>
<td align="left">LARP6</td>
<td align="left">ENST00000344870</td>
<td align="right">0.0356469</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">404203</td>
<td align="left">SPINK6</td>
<td align="left">ENST00000325630</td>
<td align="right">0.0493611</td>
<td align="right">0.3664403</td>
</tr>
<tr class="even">
<td align="left">404203</td>
<td align="left">SPINK6</td>
<td align="left">ENST00000514389</td>
<td align="right">0.0493611</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">404203</td>
<td align="left">SPINK6</td>
<td align="left">ENST00000621437</td>
<td align="right">0.0493611</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>dtu_genes &lt;- unique(top_gene$ENTREZID) 
p &lt;- vector(&quot;list&quot;, length(dtu_genes))

for(i in 1:length(p)){
  p[[i]] &lt;- plotProportions(d, gene_id = dtu_genes[i], 
                            group_variable = &quot;condition&quot;,
                            plot_type = &quot;boxplot1&quot;) +
    ggtitle(unique(top_gene$SYMBOL[top_gene$ENTREZID == dtu_genes[i]])) +
    theme(axis.text = element_text(size = 8))
}

wrap_plots(p, ncol = 1) + plot_layout(guides = &quot;collect&quot;) </code></pre>
<p><img src="figure/salmon-DRIMseq.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/7ca0e36a09f6dd079d1d9713e1d2b830384e9edd/docs/figure/salmon-DRIMseq.Rmd/unnamed-chunk-23-1.png" target="_blank">7ca0e36</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-testing" class="section level1">
<h1>Gene set testing</h1>
<pre class="r"><code>gst &lt;- goana(dtu_genes, 
             universe = d@results_gene$gene_id)
topGO(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="57%" />
<col width="3%" />
<col width="4%" />
<col width="2%" />
<col width="9%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Term</th>
<th align="left">Ont</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0048027" class="uri">GO:0048027</a></td>
<td align="left">mRNA 5’-UTR binding</td>
<td align="left">MF</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="right">0.0007731</td>
<td align="right">0.0154615</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0008138" class="uri">GO:0008138</a></td>
<td align="left">protein tyrosine/serine/threonine phosphatase activity</td>
<td align="left">MF</td>
<td align="right">37</td>
<td align="right">2</td>
<td align="right">0.0015673</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0062030" class="uri">GO:0062030</a></td>
<td align="left">negative regulation of stress granule assembly</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0015914</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1903028" class="uri">GO:1903028</a></td>
<td align="left">positive regulation of opsonization</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0015914</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1903353" class="uri">GO:1903353</a></td>
<td align="left">regulation of nucleus organization</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0015914</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0034696" class="uri">GO:0034696</a></td>
<td align="left">response to prostaglandin F</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0015914</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0016596" class="uri">GO:0016596</a></td>
<td align="left">thienylcyclohexylpiperidine binding</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0015914</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0052548" class="uri">GO:0052548</a></td>
<td align="left">regulation of endopeptidase activity</td>
<td align="left">BP</td>
<td align="right">325</td>
<td align="right">4</td>
<td align="right">0.0016259</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1902893" class="uri">GO:1902893</a></td>
<td align="left">regulation of pri-miRNA transcription by RNA polymerase II</td>
<td align="left">BP</td>
<td align="right">40</td>
<td align="right">2</td>
<td align="right">0.0018302</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0061614" class="uri">GO:0061614</a></td>
<td align="left">pri-miRNA transcription by RNA polymerase II</td>
<td align="left">BP</td>
<td align="right">42</td>
<td align="right">2</td>
<td align="right">0.0020163</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0052547" class="uri">GO:0052547</a></td>
<td align="left">regulation of peptidase activity</td>
<td align="left">BP</td>
<td align="right">349</td>
<td align="right">4</td>
<td align="right">0.0021095</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0030162" class="uri">GO:0030162</a></td>
<td align="left">regulation of proteolysis</td>
<td align="left">BP</td>
<td align="right">599</td>
<td align="right">5</td>
<td align="right">0.0022597</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0043086" class="uri">GO:0043086</a></td>
<td align="left">negative regulation of catalytic activity</td>
<td align="left">BP</td>
<td align="right">644</td>
<td align="right">5</td>
<td align="right">0.0030951</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0090164" class="uri">GO:0090164</a></td>
<td align="left">asymmetric Golgi ribbon formation</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0031803</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042883" class="uri">GO:0042883</a></td>
<td align="left">cysteine transport</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0031803</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0001691" class="uri">GO:0001691</a></td>
<td align="left">pseudophosphatase activity</td>
<td align="left">MF</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0031803</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1903027" class="uri">GO:1903027</a></td>
<td align="left">regulation of opsonization</td>
<td align="left">BP</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0.0031803</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0004867" class="uri">GO:0004867</a></td>
<td align="left">serine-type endopeptidase inhibitor activity</td>
<td align="left">MF</td>
<td align="right">62</td>
<td align="right">2</td>
<td align="right">0.0043427</td>
<td align="right">0.0297779</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0032268" class="uri">GO:0032268</a></td>
<td align="left">regulation of cellular protein metabolic process</td>
<td align="left">BP</td>
<td align="right">2181</td>
<td align="right">9</td>
<td align="right">0.0048760</td>
<td align="right">0.0297779</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051246" class="uri">GO:0051246</a></td>
<td align="left">regulation of protein metabolic process</td>
<td align="left">BP</td>
<td align="right">2299</td>
<td align="right">9</td>
<td align="right">0.0069384</td>
<td align="right">0.0297779</td>
</tr>
</tbody>
</table>
<pre class="r"><code>gst &lt;- kegga(dtu_genes, 
             universe = d@results_gene$gene_id)
topKEGG(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="16%" />
<col width="59%" />
<col width="4%" />
<col width="3%" />
<col width="11%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Pathway</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">path:hsa03008</td>
<td align="left">Ribosome biogenesis in eukaryotes</td>
<td align="right">72</td>
<td align="right">1</td>
<td align="right">0.1085507</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa04142</td>
<td align="left">Lysosome</td>
<td align="right">122</td>
<td align="right">1</td>
<td align="right">0.1771678</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05170</td>
<td align="left">Human immunodeficiency virus 1 infection</td>
<td align="right">178</td>
<td align="right">1</td>
<td align="right">0.2479892</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa05168</td>
<td align="left">Herpes simplex virus 1 infection</td>
<td align="right">446</td>
<td align="right">1</td>
<td align="right">0.5132932</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00780</td>
<td align="left">Biotin metabolism</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00290</td>
<td align="left">Valine, leucine and isoleucine biosynthesis</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00232</td>
<td align="left">Caffeine metabolism</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00471</td>
<td align="left">D-Glutamine and D-glutamate metabolism</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00785</td>
<td align="left">Lipoic acid metabolism</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00524</td>
<td align="left">Neomycin, kanamycin and gentamicin biosynthesis</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00400</td>
<td align="left">Phenylalanine, tyrosine and tryptophan biosynthesis</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00440</td>
<td align="left">Phosphonate and phosphinate metabolism</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04122</td>
<td align="left">Sulfur relay system</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00750</td>
<td align="left">Vitamin B6 metabolism</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00740</td>
<td align="left">Riboflavin metabolism</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00072</td>
<td align="left">Synthesis and degradation of ketone bodies</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00430</td>
<td align="left">Taurine and hypotaurine metabolism</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00130</td>
<td align="left">Ubiquinone and other terpenoid-quinone biosynthesis</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00920</td>
<td align="left">Sulfur metabolism</td>
<td align="right">10</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">path:hsa00533</td>
<td align="left">Glycosaminoglycan biosynthesis - keratan sulfate</td>
<td align="right">11</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] stageR_1.12.0                           
 [2] SummarizedExperiment_1.20.0             
 [3] MatrixGenerics_1.2.1                    
 [4] matrixStats_0.59.0                      
 [5] BANDITS_1.6.0                           
 [6] DRIMSeq_1.18.0                          
 [7] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
 [8] GenomicFeatures_1.42.1                  
 [9] GenomicRanges_1.42.0                    
[10] GenomeInfoDb_1.26.7                     
[11] org.Hs.eg.db_3.12.0                     
[12] AnnotationDbi_1.52.0                    
[13] IRanges_2.24.1                          
[14] S4Vectors_0.28.1                        
[15] Biobase_2.50.0                          
[16] BiocGenerics_0.36.1                     
[17] tximport_1.18.0                         
[18] patchwork_1.1.1                         
[19] edgeR_3.32.1                            
[20] limma_3.46.0                            
[21] forcats_0.5.1                           
[22] stringr_1.4.0                           
[23] dplyr_1.0.4                             
[24] purrr_0.3.4                             
[25] readr_1.4.0                             
[26] tidyr_1.1.2                             
[27] tibble_3.1.2                            
[28] ggplot2_3.3.5                           
[29] tidyverse_1.3.0                         
[30] here_1.0.1                              
[31] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2         ellipsis_0.3.2           rprojroot_2.0.2         
  [4] XVector_0.30.0           fs_1.5.0                 rstudioapi_0.13         
  [7] farver_2.1.0             bit64_4.0.5              fansi_0.5.0             
 [10] lubridate_1.7.9.2        xml2_1.3.2               codetools_0.2-18        
 [13] doParallel_1.0.16        cachem_1.0.4             knitr_1.31              
 [16] jsonlite_1.7.2           Rsamtools_2.6.0          broom_0.7.4             
 [19] GO.db_3.12.1             dbplyr_2.1.0             compiler_4.0.2          
 [22] httr_1.4.2               backports_1.2.1          assertthat_0.2.1        
 [25] Matrix_1.3-2             fastmap_1.1.0            cli_3.0.0               
 [28] later_1.1.0.1            htmltools_0.5.1.1        prettyunits_1.1.1       
 [31] tools_4.0.2              gtable_0.3.0             glue_1.4.2              
 [34] GenomeInfoDbData_1.2.4   reshape2_1.4.4           doRNG_1.8.2             
 [37] rappdirs_0.3.3           Rcpp_1.0.6               cellranger_1.1.0        
 [40] vctrs_0.3.8              Biostrings_2.58.0        rtracklayer_1.50.0      
 [43] iterators_1.0.13         xfun_0.23                rvest_0.3.6             
 [46] lifecycle_1.0.0          rngtools_1.5             XML_3.99-0.5            
 [49] MASS_7.3-53.1            zlibbioc_1.36.0          scales_1.1.1            
 [52] hms_1.0.0                promises_1.2.0.1         yaml_2.2.1              
 [55] curl_4.3                 memoise_2.0.0            biomaRt_2.46.3          
 [58] stringi_1.5.3            RSQLite_2.2.5            highr_0.8               
 [61] foreach_1.5.1            BiocParallel_1.24.1      rlang_0.4.11            
 [64] pkgconfig_2.0.3          bitops_1.0-7             evaluate_0.14           
 [67] lattice_0.20-41          labeling_0.4.2           GenomicAlignments_1.26.0
 [70] bit_4.0.4                tidyselect_1.1.0         plyr_1.8.6              
 [73] magrittr_2.0.1           R6_2.5.0                 generics_0.1.0          
 [76] DelayedArray_0.16.3      DBI_1.1.1                pillar_1.6.1            
 [79] haven_2.3.1              whisker_0.4              withr_2.4.2             
 [82] RCurl_1.98-1.3           modelr_0.1.8             crayon_1.4.1            
 [85] utf8_1.2.1               BiocFileCache_1.14.0     rmarkdown_2.6           
 [88] progress_1.2.2           locfit_1.5-9.4           grid_4.0.2              
 [91] readxl_1.3.1             data.table_1.13.6        blob_1.2.1              
 [94] git2r_0.28.0             reprex_1.0.0             digest_0.6.27           
 [97] httpuv_1.5.5             openssl_1.4.3            munsell_0.5.0           
[100] askpass_1.1             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
