<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Alternative splicing analysis</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Alternative splicing analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreea9955fecedf1cbec5b44a7f142d12a9482b0163ctargetblanka9955fea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/a9955fecedf1cbec5b44a7f142d12a9482b0163c" target="_blank">a9955fe</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreea9955fecedf1cbec5b44a7f142d12a9482b0163ctargetblanka9955fea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/a9955fecedf1cbec5b44a7f142d12a9482b0163c" target="_blank">a9955fe</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/commandlog.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/multiqc_data/
    Ignored:    data/multiqc_report.html
    Ignored:    data/samples.csv
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/exclude-CMV11/
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData
    Ignored:    output/salmon-limma-voom-c5Cam.csv
    Ignored:    output/salmon-limma-voom.Rmd/
    Ignored:    output/salmon-limma-voom.csv
    Ignored:    output/salmon-ruvseq-edger.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-all-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-all.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-limma-voom-no_us_ab.csv
    Ignored:    output/star-fc-ruv-all-gsea-c2.csv
    Ignored:    output/star-fc-ruv-all-gsea-c5.csv
    Ignored:    output/star-fc-ruv-all.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Ignored:    output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Ignored:    output/star-fc-ruv-no_us_ab.csv

Untracked files:
    Untracked:  .bpipe/
    Untracked:  analysis/STAR-diffSplice.Rmd
    Untracked:  analysis/salmon-DRIMseq.Rmd
    Untracked:  analysis/saturn-dtu.Rmd
    Untracked:  renv.lock
    Untracked:  satuRn_0.99.7.tar.gz

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load sample information.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">CMV_status</th>
<th align="left">pair</th>
<th align="left">sex</th>
<th align="left">GA_at_amnio</th>
<th align="left">indication</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CMV2</td>
<td align="left">neg</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV1</td>
<td align="left">pos</td>
<td align="left">M1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV4</td>
<td align="left">pos</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV3</td>
<td align="left">neg</td>
<td align="left">M2</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV10</td>
<td align="left">neg</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">20</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV11</td>
<td align="left">pos</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">19</td>
<td align="left">us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV19</td>
<td align="left">pos</td>
<td align="left">NC2</td>
<td align="left">F</td>
<td align="left">18</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV35</td>
<td align="left">neg</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV30</td>
<td align="left">pos</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV31</td>
<td align="left">neg</td>
<td align="left">L1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV8</td>
<td align="left">neg</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV9</td>
<td align="left">pos</td>
<td align="left">L2</td>
<td align="left">F</td>
<td align="left">23</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV26</td>
<td align="left">pos</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV56</td>
<td align="left">neg</td>
<td align="left">L3</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV14</td>
<td align="left">neg</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV15</td>
<td align="left">pos</td>
<td align="left">L4</td>
<td align="left">F</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV20</td>
<td align="left">pos</td>
<td align="left">L5</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV51</td>
<td align="left">neg</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV57</td>
<td align="left">pos</td>
<td align="left">L6</td>
<td align="left">M</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV58</td>
<td align="left">pos</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV60</td>
<td align="left">neg</td>
<td align="left">L7</td>
<td align="left">M</td>
<td align="left">20</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV52</td>
<td align="left">pos</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV61</td>
<td align="left">neg</td>
<td align="left">L8</td>
<td align="left">M</td>
<td align="left">22</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV54</td>
<td align="left">neg</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
<tr class="odd">
<td align="left">CMV53</td>
<td align="left">pos</td>
<td align="left">L9</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">us_ab</td>
</tr>
<tr class="even">
<td align="left">CMV21</td>
<td align="left">neg</td>
<td align="left">NC1</td>
<td align="left">F</td>
<td align="left">21</td>
<td align="left">no_us_ab</td>
</tr>
</tbody>
</table>
<pre class="r"><code>int &lt;- intersect(names(bams), targets$id)
targets &lt;- targets[match(int, targets$id),]
bams &lt;- bams[match(int, names(bams))]
names(bams)</code></pre>
<pre><code> [1] &quot;CMV30&quot; &quot;CMV31&quot; &quot;CMV8&quot;  &quot;CMV9&quot;  &quot;CMV26&quot; &quot;CMV14&quot; &quot;CMV15&quot; &quot;CMV20&quot; &quot;CMV21&quot;
[10] &quot;CMV1&quot;  &quot;CMV2&quot;  &quot;CMV3&quot;  &quot;CMV4&quot;  &quot;CMV10&quot; &quot;CMV11&quot; &quot;CMV19&quot; &quot;CMV35&quot; &quot;CMV51&quot;
[19] &quot;CMV52&quot; &quot;CMV53&quot; &quot;CMV54&quot; &quot;CMV56&quot; &quot;CMV57&quot; &quot;CMV58&quot; &quot;CMV60&quot; &quot;CMV61&quot;</code></pre>
<pre class="r"><code>exon_counts &lt;- Rsubread::featureCounts(bams, 
                        annot.ext = &quot;/oshlack_lab/shared/genomes_mcri/hg38/gtf/gencode.v34.annotation.gtf&quot;,
                        isGTFAnnotationFile = TRUE,
                        GTF.featureType = &quot;exon&quot;,
                        GTF.attrType = &quot;gene_id&quot;,
                        useMetaFeatures = FALSE,
                        isPairedEnd = TRUE,
                        strandSpecific = 0,
                        allowMultiOverlap=TRUE,
                        nthreads = 5)</code></pre>
<pre><code>
        ==========     _____ _    _ ____  _____  ______          _____  
        =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ 
          =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
            ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |
              ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
        ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
       Rsubread 2.4.2

//========================== featureCounts setting ===========================\\
||                                                                            ||
||             Input files : 26 BAM files                                     ||
||                                                                            ||
||                           190717_A00692_0021_TL1906128_CMV30_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906129_CMV31_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906130_CMV8_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906131_CMV9_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906132_CMV26_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906134_CMV14_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906135_CMV15_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906136_CMV20_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906137_CMV21_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906138_CMV1_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906139_CMV2_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906140_CMV3_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906141_CMV4_MAN-201907 ... ||
||                           190717_A00692_0021_TL1906142_CMV10_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906143_CMV11_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906145_CMV19_MAN-20190 ... ||
||                           190717_A00692_0021_TL1906146_CMV35_MAN-20190 ... ||
||                           200612_A00692_0107_ML205836_CMV51_MAN-202006 ... ||
||                           200612_A00692_0107_ML205837_CMV52_MAN-202006 ... ||
||                           200612_A00692_0107_ML205838_CMV53_MAN-202006 ... ||
||                           200612_A00692_0107_ML205839_CMV54_MAN-202006 ... ||
||                           200612_A00692_0107_ML205840_CMV56_MAN-202006 ... ||
||                           200612_A00692_0107_ML205841_CMV57_MAN-202006 ... ||
||                           200612_A00692_0107_ML205842_CMV58_MAN-202006 ... ||
||                           200612_A00692_0107_ML205843_CMV60_MAN-202006 ... ||
||                           200612_A00692_0107_ML205844_CMV61_MAN-202006 ... ||
||                                                                            ||
||              Paired-end : yes                                              ||
||        Count read pairs : yes                                              ||
||              Annotation : gencode.v34.annotation.gtf (GTF)                 ||
||      Dir for temp files : .                                                ||
||                 Threads : 5                                                ||
||                   Level : feature level                                    ||
||      Multimapping reads : counted                                          ||
|| Multi-overlapping reads : counted                                          ||
||   Min overlapping bases : 1                                                ||
||                                                                            ||
\\============================================================================//

//================================= Running ==================================\\
||                                                                            ||
|| Load annotation file gencode.v34.annotation.gtf ...                        ||
||    Features : 1378888                                                      ||
||    Meta-features : 60669                                                   ||
||    Chromosomes/contigs : 25                                                ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906128_CMV30_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 46772403                                             ||
||    Successfully assigned alignments : 22442386 (48.0%)                     ||
||    Running time : 2.58 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906129_CMV31_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 52451946                                             ||
||    Successfully assigned alignments : 20618496 (39.3%)                     ||
||    Running time : 3.53 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906130_CMV8_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 52085219                                             ||
||    Successfully assigned alignments : 20564541 (39.5%)                     ||
||    Running time : 3.84 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906131_CMV9_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 58025539                                             ||
||    Successfully assigned alignments : 21672514 (37.3%)                     ||
||    Running time : 5.21 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906132_CMV26_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 48727354                                             ||
||    Successfully assigned alignments : 18828412 (38.6%)                     ||
||    Running time : 3.00 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906134_CMV14_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 60890000                                             ||
||    Successfully assigned alignments : 26104430 (42.9%)                     ||
||    Running time : 5.56 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906135_CMV15_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 52984669                                             ||
||    Successfully assigned alignments : 21260783 (40.1%)                     ||
||    Running time : 3.13 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906136_CMV20_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 48823416                                             ||
||    Successfully assigned alignments : 20417211 (41.8%)                     ||
||    Running time : 3.13 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906137_CMV21_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 45962903                                             ||
||    Successfully assigned alignments : 19745700 (43.0%)                     ||
||    Running time : 3.78 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906138_CMV1_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 53919799                                             ||
||    Successfully assigned alignments : 18850779 (35.0%)                     ||
||    Running time : 4.41 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906139_CMV2_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 49145958                                             ||
||    Successfully assigned alignments : 18216745 (37.1%)                     ||
||    Running time : 3.94 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906140_CMV3_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 48645023                                             ||
||    Successfully assigned alignments : 17665801 (36.3%)                     ||
||    Running time : 3.68 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906141_CMV4_MAN-20190712_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 57767938                                             ||
||    Successfully assigned alignments : 18852042 (32.6%)                     ||
||    Running time : 5.10 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906142_CMV10_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 51477025                                             ||
||    Successfully assigned alignments : 20794454 (40.4%)                     ||
||    Running time : 3.60 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906143_CMV11_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 39217048                                             ||
||    Successfully assigned alignments : 14083574 (35.9%)                     ||
||    Running time : 2.69 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906145_CMV19_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 63443808                                             ||
||    Successfully assigned alignments : 21596535 (34.0%)                     ||
||    Running time : 6.82 minutes                                             ||
||                                                                            ||
|| Process BAM file 190717_A00692_0021_TL1906146_CMV35_MAN-20190712_Ovati ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 50333928                                             ||
||    Successfully assigned alignments : 16625521 (33.0%)                     ||
||    Running time : 5.58 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205836_CMV51_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 48290664                                             ||
||    Successfully assigned alignments : 20374700 (42.2%)                     ||
||    Running time : 3.70 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205837_CMV52_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 39380458                                             ||
||    Successfully assigned alignments : 16550771 (42.0%)                     ||
||    Running time : 2.09 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205838_CMV53_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 38675992                                             ||
||    Successfully assigned alignments : 16954055 (43.8%)                     ||
||    Running time : 2.13 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205839_CMV54_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 42142786                                             ||
||    Successfully assigned alignments : 19502355 (46.3%)                     ||
||    Running time : 2.14 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205840_CMV56_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 48003631                                             ||
||    Successfully assigned alignments : 22807881 (47.5%)                     ||
||    Running time : 2.73 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205841_CMV57_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 45860078                                             ||
||    Successfully assigned alignments : 21859214 (47.7%)                     ||
||    Running time : 3.53 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205842_CMV58_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 46648421                                             ||
||    Successfully assigned alignments : 24499391 (52.5%)                     ||
||    Running time : 2.80 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205843_CMV60_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 38413315                                             ||
||    Successfully assigned alignments : 17682135 (46.0%)                     ||
||    Running time : 1.75 minutes                                             ||
||                                                                            ||
|| Process BAM file 200612_A00692_0107_ML205844_CMV61_MAN-20200611_Ovatio ... ||
||    Paired-end reads are included.                                          ||
||    Total alignments : 36227746                                             ||
||    Successfully assigned alignments : 14489362 (40.0%)                     ||
||    Running time : 2.35 minutes                                             ||
||                                                                            ||
|| Write the final count table.                                               ||
|| Write the read assignment summary.                                         ||
||                                                                            ||
\\============================================================================//</code></pre>
<pre class="r"><code>ects &lt;- exon_counts$counts
rownames(ects) &lt;- exon_counts$annotation %&gt;% 
  unite(&quot;ID&quot;, GeneID, Chr, Start, End) %&gt;% 
  pull(ID)
colnames(ects) &lt;- names(bams)

yall &lt;- DGEList(counts = exon_counts$counts, genes = exon_counts$annotation,
                samples = targets, group = targets$CMV_status)</code></pre>
<pre><code>Warning: Count matrix has duplicated rownames</code></pre>
<pre class="r"><code>keep &lt;- filterByExpr(yall)

y &lt;- yall[keep, , keep.lib.sizes=FALSE]
y &lt;- calcNormFactors(y)
y$samples</code></pre>
<pre><code>                                                                                                 group
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   neg
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   neg
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   neg
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   neg
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   pos
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   neg
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    pos
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    neg
                                                                                                  lib.size
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 169930332
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 135486891
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  137730875
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  141242553
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 123795339
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 195188086
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 137942988
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 132256377
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 152798099
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  123502104
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  133073012
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   97186314
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  122082238
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 134284214
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  83903936
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 130730299
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam 107224630
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  143750797
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  121513044
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  132494424
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  162229539
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  192680012
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  175387069
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  194383099
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  145039558
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  103541814
                                                                                                 norm.factors
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.0623467
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.0653499
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0236374
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.9405121
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.0250761
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    0.9797457
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.1584306
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.1107525
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.0022780
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.9274646
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.8960790
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0246169
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.8680287
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    0.9217408
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    0.7443903
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    1.0090947
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    0.9024024
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0562678
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0404825
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.9943397
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.9907885
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     0.9756185
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.1148193
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.1660306
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0978425
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam     1.0177611
                                                                                                    id
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV30
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV31
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV8
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV9
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV26
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV14
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV15
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV20
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV21
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV1
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV2
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV3
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   CMV4
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV10
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV11
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV19
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam CMV35
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV51
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV52
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV53
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV54
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV56
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV57
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV58
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV60
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  CMV61
                                                                                                 CMV_status
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        neg
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        neg
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        neg
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        neg
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        pos
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam        neg
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         pos
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam         neg
                                                                                                 pair
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L1
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L1
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L2
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L2
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L3
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L4
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L4
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L5
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  NC1
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M1
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M1
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M2
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M2
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  NC2
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  NC1
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam  NC2
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   L5
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L6
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L8
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L9
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L9
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L3
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L6
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L7
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L7
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    L8
                                                                                                 sex
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   M
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   F
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   M
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    F
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    M
                                                                                                 GA_at_amnio
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           23
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           23
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          22
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          22
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           20
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           22
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          20
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          19
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          18
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam          21
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           22
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           22
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           21
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           20
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           20
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam           22
                                                                                                 indication
190717_A00692_0021_TL1906128_CMV30_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906129_CMV31_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906130_CMV8_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906131_CMV9_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906132_CMV26_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906134_CMV14_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906135_CMV15_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906136_CMV20_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906137_CMV21_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906138_CMV1_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906139_CMV2_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906140_CMV3_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906141_CMV4_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
190717_A00692_0021_TL1906142_CMV10_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam      us_ab
190717_A00692_0021_TL1906143_CMV11_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam      us_ab
190717_A00692_0021_TL1906145_CMV19_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
190717_A00692_0021_TL1906146_CMV35_MAN-20190712_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam   no_us_ab
200612_A00692_0107_ML205836_CMV51_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205837_CMV52_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205838_CMV53_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam       us_ab
200612_A00692_0107_ML205839_CMV54_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205840_CMV56_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205841_CMV57_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205842_CMV58_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205843_CMV60_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab
200612_A00692_0107_ML205844_CMV61_MAN-20200611_OvationSoloRNA_L000Aligned.sortedByCoord.out.bam    no_us_ab</code></pre>
<pre class="r"><code>pal &lt;- scales::hue_pal()(2)
plotMDS(y, gene.selection = &quot;common&quot;, labels = names(bams),
        col = pal[factor(y$samples$CMV_status)])</code></pre>
<p><img src="figure/STAR-diffSplice.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>design &lt;- model.matrix(~ y$samples$CMV_status + y$samples$pair)
design</code></pre>
<pre><code>   (Intercept) y$samples$CMV_statuspos y$samples$pairL2 y$samples$pairL3
1            1                       1                0                0
2            1                       0                0                0
3            1                       0                1                0
4            1                       1                1                0
5            1                       1                0                1
6            1                       0                0                0
7            1                       1                0                0
8            1                       1                0                0
9            1                       0                0                0
10           1                       1                0                0
11           1                       0                0                0
12           1                       0                0                0
13           1                       1                0                0
14           1                       0                0                0
15           1                       1                0                0
16           1                       1                0                0
17           1                       0                0                0
18           1                       0                0                0
19           1                       1                0                0
20           1                       1                0                0
21           1                       0                0                0
22           1                       0                0                1
23           1                       1                0                0
24           1                       1                0                0
25           1                       0                0                0
26           1                       0                0                0
   y$samples$pairL4 y$samples$pairL5 y$samples$pairL6 y$samples$pairL7
1                 0                0                0                0
2                 0                0                0                0
3                 0                0                0                0
4                 0                0                0                0
5                 0                0                0                0
6                 1                0                0                0
7                 1                0                0                0
8                 0                1                0                0
9                 0                0                0                0
10                0                0                0                0
11                0                0                0                0
12                0                0                0                0
13                0                0                0                0
14                0                0                0                0
15                0                0                0                0
16                0                0                0                0
17                0                1                0                0
18                0                0                1                0
19                0                0                0                0
20                0                0                0                0
21                0                0                0                0
22                0                0                0                0
23                0                0                1                0
24                0                0                0                1
25                0                0                0                1
26                0                0                0                0
   y$samples$pairL8 y$samples$pairL9 y$samples$pairM1 y$samples$pairM2
1                 0                0                0                0
2                 0                0                0                0
3                 0                0                0                0
4                 0                0                0                0
5                 0                0                0                0
6                 0                0                0                0
7                 0                0                0                0
8                 0                0                0                0
9                 0                0                0                0
10                0                0                1                0
11                0                0                1                0
12                0                0                0                1
13                0                0                0                1
14                0                0                0                0
15                0                0                0                0
16                0                0                0                0
17                0                0                0                0
18                0                0                0                0
19                1                0                0                0
20                0                1                0                0
21                0                1                0                0
22                0                0                0                0
23                0                0                0                0
24                0                0                0                0
25                0                0                0                0
26                1                0                0                0
   y$samples$pairNC1 y$samples$pairNC2
1                  0                 0
2                  0                 0
3                  0                 0
4                  0                 0
5                  0                 0
6                  0                 0
7                  0                 0
8                  0                 0
9                  1                 0
10                 0                 0
11                 0                 0
12                 0                 0
13                 0                 0
14                 0                 1
15                 1                 0
16                 0                 1
17                 0                 0
18                 0                 0
19                 0                 0
20                 0                 0
21                 0                 0
22                 0                 0
23                 0                 0
24                 0                 0
25                 0                 0
26                 0                 0
attr(,&quot;assign&quot;)
 [1] 0 1 2 2 2 2 2 2 2 2 2 2 2 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$`y$samples$CMV_status`
[1] &quot;contr.treatment&quot;

attr(,&quot;contrasts&quot;)$`y$samples$pair`
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code>y &lt;- estimateDisp(y, design, robust=TRUE)
y$common.dispersion</code></pre>
<pre><code>[1] 0.1307782</code></pre>
<pre class="r"><code>plotBCV(y)</code></pre>
<p><img src="figure/STAR-diffSplice.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- glmQLFit(y, design, robust=TRUE)
plotQLDisp(fit)</code></pre>
<p><img src="figure/STAR-diffSplice.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>qlf &lt;- glmQLFTest(fit, coef=2)
topTags(qlf)</code></pre>
<pre><code>Coefficient:  y$samples$CMV_statuspos 
                                   GeneID  Chr     Start       End Strand
ENSG00000115415.20.544 ENSG00000115415.20 chr2 191009027 191009107      -
ENSG00000115415.20.160 ENSG00000115415.20 chr2 191009876 191009913      -
ENSG00000115415.20.231 ENSG00000115415.20 chr2 190969312 190970717      -
ENSG00000115415.20.256 ENSG00000115415.20 chr2 190969312 190970717      -
ENSG00000115415.20.281 ENSG00000115415.20 chr2 190969318 190970717      -
ENSG00000115415.20.153 ENSG00000115415.20 chr2 190969267 190970717      -
ENSG00000115415.20.159 ENSG00000115415.20 chr2 190969268 190970717      -
ENSG00000115415.20.169 ENSG00000115415.20 chr2 190969268 190970717      -
ENSG00000115415.20.176 ENSG00000115415.20 chr2 190969268 190970717      -
ENSG00000115415.20.200 ENSG00000115415.20 chr2 190969268 190970717      -
                       Length    logFC    logCPM        F       PValue
ENSG00000115415.20.544     81 1.326963 0.7024811 88.85237 1.792811e-10
ENSG00000115415.20.160     38 1.355379 0.4419156 80.16397 5.664399e-10
ENSG00000115415.20.231   1406 1.240790 2.7914698 79.08308 7.226015e-10
ENSG00000115415.20.256   1406 1.240790 2.7914698 79.08308 7.226015e-10
ENSG00000115415.20.281   1400 1.240523 2.7903807 78.95763 7.352242e-10
ENSG00000115415.20.153   1451 1.240723 2.7980852 78.56146 7.826197e-10
ENSG00000115415.20.159   1450 1.240723 2.7980852 78.56146 7.826197e-10
ENSG00000115415.20.169   1450 1.240723 2.7980852 78.56146 7.826197e-10
ENSG00000115415.20.176   1450 1.240723 2.7980852 78.56146 7.826197e-10
ENSG00000115415.20.200   1450 1.240723 2.7980852 78.56146 7.826197e-10
                                FDR
ENSG00000115415.20.544 1.795422e-05
ENSG00000115415.20.160 1.795422e-05
ENSG00000115415.20.231 1.795422e-05
ENSG00000115415.20.256 1.795422e-05
ENSG00000115415.20.281 1.795422e-05
ENSG00000115415.20.153 1.795422e-05
ENSG00000115415.20.159 1.795422e-05
ENSG00000115415.20.169 1.795422e-05
ENSG00000115415.20.176 1.795422e-05
ENSG00000115415.20.200 1.795422e-05</code></pre>
<pre class="r"><code>is.de &lt;- decideTests(qlf, p.value=0.05)
summary(is.de)</code></pre>
<pre><code>       y$samples$CMV_statuspos
Down                        36
NotSig                  718906
Up                        1731</code></pre>
<pre class="r"><code>sp &lt;- diffSpliceDGE(fit, coef=2, geneid=&quot;GeneID&quot;, exonid=&quot;Start&quot;)</code></pre>
<pre><code>Total number of exons:  720673 
Total number of genes:  18438 
Number of genes with 1 exon:  3654 
Mean number of exons in a gene:  39 
Max number of exons in a gene:  1894 </code></pre>
<pre class="r"><code>topSpliceDGE(sp, test=&quot;Simes&quot;, n=20)</code></pre>
<pre><code>                    GeneID   Chr Strand NExons      P.Value        FDR
470548  ENSG00000112782.19  chr6      -     53 8.915490e-07 0.01318066
135362  ENSG00000084676.15  chr2      +    145 2.849882e-05 0.21066327
173856   ENSG00000257207.5  chr2      -      8 5.740143e-05 0.28287424
173470   ENSG00000284337.1  chr2      +     15 1.063021e-04 0.34867118
956978  ENSG00000114062.21 chr15      -    223 1.317021e-04 0.34867118
1079487  ENSG00000264772.6 chr17      +     12 1.415062e-04 0.34867118
381722  ENSG00000180104.16  chr5      +     58 1.876956e-04 0.39641306
1024663  ENSG00000227827.3 chr16      +     17 2.272028e-04 0.41987084
418214   ENSG00000283782.2  chr5      +    117 2.956724e-04 0.48569127
1024950  ENSG00000254681.6 chr16      -     37 3.790926e-04 0.51294589
648881  ENSG00000106785.15  chr9      -     30 3.816562e-04 0.51294589
793883  ENSG00000196655.12 chr11      +     52 5.478322e-04 0.67492926
723261   ENSG00000283496.1 chr10      +      7 1.264088e-03 1.00000000
33339    ENSG00000198815.9  chr1      -     78 1.265580e-03 1.00000000
1276384 ENSG00000124201.15 chr20      -     61 1.752259e-03 1.00000000
704140  ENSG00000119922.10 chr10      +      8 1.895289e-03 1.00000000
753787   ENSG00000288534.1 chr11      +     73 2.117461e-03 1.00000000
295440   ENSG00000114107.9  chr3      -    117 3.201029e-03 1.00000000
1348208 ENSG00000196632.10  chrX      -     47 3.775203e-03 1.00000000
725932  ENSG00000177542.11 chr11      -     63 3.997188e-03 1.00000000</code></pre>
<pre class="r"><code>top_gene &lt;- topSpliceDGE(sp, test=&quot;gene&quot;, n=Inf)
top_gene$ENSEMBL &lt;- strsplit2(top_gene$GeneID, &quot;.&quot;, fixed = TRUE)[,1]
select(org.Hs.eg.db, keys = top_gene$ENSEMBL, columns = c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;),
       keytype = &quot;ENSEMBL&quot;) %&gt;% 
  dplyr::inner_join(top_gene, by = c(&quot;ENSEMBL&quot; = &quot;ENSEMBL&quot;)) -&gt; top_gene</code></pre>
<pre><code>&#39;select()&#39; returned many:many mapping between keys and columns</code></pre>
<pre class="r"><code>head(top_gene)</code></pre>
<pre><code>          ENSEMBL SYMBOL ENTREZID             GeneID   Chr Strand NExons
1 ENSG00000114062  UBE3A     7337 ENSG00000114062.21 chr15      -    223
2 ENSG00000103544 VPS35L    57020 ENSG00000103544.15 chr16      +    251
3 ENSG00000123106 CCDC91    55297 ENSG00000123106.10 chr12      +    141
4 ENSG00000062485     CS     1431 ENSG00000062485.19 chr12      -    176
5 ENSG00000172456   FGGY    55277 ENSG00000172456.18  chr1      +    199
6 ENSG00000284770   TBCE     6905  ENSG00000284770.2  chr1      +    598
    gene.F       P.Value           FDR
1 6.329515 2.365894e-129 3.497738e-125
2 3.028532  1.230479e-44  9.095699e-41
3 3.496014  8.491205e-34  4.184466e-30
4 2.907758  1.939050e-29  7.166729e-26
5 2.570646  1.888695e-25  5.584492e-22
6 1.760111  1.095195e-24  2.698562e-21</code></pre>
<pre class="r"><code>head(top_gene) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ENSEMBL</th>
<th align="left">SYMBOL</th>
<th align="left">ENTREZID</th>
<th align="left">GeneID</th>
<th align="left">Chr</th>
<th align="left">Strand</th>
<th align="right">NExons</th>
<th align="right">gene.F</th>
<th align="right">P.Value</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000114062</td>
<td align="left">UBE3A</td>
<td align="left">7337</td>
<td align="left">ENSG00000114062.21</td>
<td align="left">chr15</td>
<td align="left">-</td>
<td align="right">223</td>
<td align="right">6.329515</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000103544</td>
<td align="left">VPS35L</td>
<td align="left">57020</td>
<td align="left">ENSG00000103544.15</td>
<td align="left">chr16</td>
<td align="left">+</td>
<td align="right">251</td>
<td align="right">3.028532</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000123106</td>
<td align="left">CCDC91</td>
<td align="left">55297</td>
<td align="left">ENSG00000123106.10</td>
<td align="left">chr12</td>
<td align="left">+</td>
<td align="right">141</td>
<td align="right">3.496014</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000062485</td>
<td align="left">CS</td>
<td align="left">1431</td>
<td align="left">ENSG00000062485.19</td>
<td align="left">chr12</td>
<td align="left">-</td>
<td align="right">176</td>
<td align="right">2.907758</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000172456</td>
<td align="left">FGGY</td>
<td align="left">55277</td>
<td align="left">ENSG00000172456.18</td>
<td align="left">chr1</td>
<td align="left">+</td>
<td align="right">199</td>
<td align="right">2.570646</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">ENSG00000284770</td>
<td align="left">TBCE</td>
<td align="left">6905</td>
<td align="left">ENSG00000284770.2</td>
<td align="left">chr1</td>
<td align="left">+</td>
<td align="right">598</td>
<td align="right">1.760111</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>par(mfrow=c(6,2))
sapply(1:12, function(i){
  plotSpliceDGE(sp, geneid = top_gene$GeneID[i], genecolname = &quot;GeneID&quot;)
  title(sub = top_gene$SYMBOL[i])
})</code></pre>
<p><img src="figure/STAR-diffSplice.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL

[[5]]
NULL

[[6]]
NULL

[[7]]
NULL

[[8]]
NULL

[[9]]
NULL

[[10]]
NULL

[[11]]
NULL

[[12]]
NULL</code></pre>
<pre class="r"><code>gst &lt;- goana(top_gene$ENTREZID[1:50], 
             universe = top_gene$ENTREZID)
topGO(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="61%" />
<col width="3%" />
<col width="3%" />
<col width="2%" />
<col width="9%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Term</th>
<th align="left">Ont</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0006282" class="uri">GO:0006282</a></td>
<td align="left">regulation of DNA repair</td>
<td align="left">BP</td>
<td align="right">119</td>
<td align="right">4</td>
<td align="right">0.0006666</td>
<td align="right">0.0133315</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1905809" class="uri">GO:1905809</a></td>
<td align="left">negative regulation of synapse organization</td>
<td align="left">BP</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">0.0011275</td>
<td align="right">0.0214222</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0045739" class="uri">GO:0045739</a></td>
<td align="left">positive regulation of DNA repair</td>
<td align="left">BP</td>
<td align="right">65</td>
<td align="right">3</td>
<td align="right">0.0013389</td>
<td align="right">0.0241002</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006281" class="uri">GO:0006281</a></td>
<td align="left">DNA repair</td>
<td align="left">BP</td>
<td align="right">505</td>
<td align="right">7</td>
<td align="right">0.0013600</td>
<td align="right">0.0241002</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0016578" class="uri">GO:0016578</a></td>
<td align="left">histone deubiquitination</td>
<td align="left">BP</td>
<td align="right">22</td>
<td align="right">2</td>
<td align="right">0.0024437</td>
<td align="right">0.0390986</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0043015" class="uri">GO:0043015</a></td>
<td align="left">gamma-tubulin binding</td>
<td align="left">MF</td>
<td align="right">25</td>
<td align="right">2</td>
<td align="right">0.0031533</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0097510" class="uri">GO:0097510</a></td>
<td align="left">base-excision repair, AP site formation via deaminated base removal</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0004108" class="uri">GO:0004108</a></td>
<td align="left">citrate (Si)-synthase activity</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0036440" class="uri">GO:0036440</a></td>
<td align="left">citrate synthase activity</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0019150" class="uri">GO:0019150</a></td>
<td align="left">D-ribulokinase activity</td>
<td align="left">MF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0061795" class="uri">GO:0061795</a></td>
<td align="left">Golgi lumen acidification</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1905528" class="uri">GO:1905528</a></td>
<td align="left">positive regulation of Golgi lumen acidification</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1905526" class="uri">GO:1905526</a></td>
<td align="left">regulation of Golgi lumen acidification</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:1903302" class="uri">GO:1903302</a></td>
<td align="left">regulation of pyruvate kinase activity</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0035037" class="uri">GO:0035037</a></td>
<td align="left">sperm entry</td>
<td align="left">BP</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.0033602</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:2001022" class="uri">GO:2001022</a></td>
<td align="left">positive regulation of response to DNA damage stimulus</td>
<td align="left">BP</td>
<td align="right">92</td>
<td align="right">3</td>
<td align="right">0.0036139</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006259" class="uri">GO:0006259</a></td>
<td align="left">DNA metabolic process</td>
<td align="left">BP</td>
<td align="right">791</td>
<td align="right">8</td>
<td align="right">0.0043915</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:2001020" class="uri">GO:2001020</a></td>
<td align="left">regulation of response to DNA damage stimulus</td>
<td align="left">BP</td>
<td align="right">202</td>
<td align="right">4</td>
<td align="right">0.0046177</td>
<td align="right">0.0473001</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0090303" class="uri">GO:0090303</a></td>
<td align="left">positive regulation of wound healing</td>
<td align="left">BP</td>
<td align="right">33</td>
<td align="right">2</td>
<td align="right">0.0054561</td>
<td align="right">0.0473001</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006409" class="uri">GO:0006409</a></td>
<td align="left">tRNA export from nucleus</td>
<td align="left">BP</td>
<td align="right">33</td>
<td align="right">2</td>
<td align="right">0.0054561</td>
<td align="right">0.0473001</td>
</tr>
</tbody>
</table>
<pre class="r"><code>gst &lt;- kegga(top_gene$ENTREZID[1:50])
topKEGG(gst) %&gt;% 
  mutate(FDR = p.adjust(P.DE)) %&gt;% 
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="17%" />
<col width="50%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Pathway</th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">path:hsa03440</td>
<td align="left">Homologous recombination</td>
<td align="right">41</td>
<td align="right">2</td>
<td align="right">0.0044748</td>
<td align="right">0.0894965</td>
</tr>
<tr class="even">
<td align="left">path:hsa04120</td>
<td align="left">Ubiquitin mediated proteolysis</td>
<td align="right">140</td>
<td align="right">3</td>
<td align="right">0.0046317</td>
<td align="right">0.0894965</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05135</td>
<td align="left">Yersinia infection</td>
<td align="right">137</td>
<td align="right">2</td>
<td align="right">0.0441522</td>
<td align="right">0.7947390</td>
</tr>
<tr class="even">
<td align="left">path:hsa01210</td>
<td align="left">2-Oxocarboxylic acid metabolism</td>
<td align="right">19</td>
<td align="right">1</td>
<td align="right">0.0458911</td>
<td align="right">0.7947390</td>
</tr>
<tr class="odd">
<td align="left">path:hsa00020</td>
<td align="left">Citrate cycle (TCA cycle)</td>
<td align="right">30</td>
<td align="right">1</td>
<td align="right">0.0715376</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa00630</td>
<td align="left">Glyoxylate and dicarboxylate metabolism</td>
<td align="right">30</td>
<td align="right">1</td>
<td align="right">0.0715376</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04360</td>
<td align="left">Axon guidance</td>
<td align="right">182</td>
<td align="right">2</td>
<td align="right">0.0731063</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa03013</td>
<td align="left">RNA transport</td>
<td align="right">186</td>
<td align="right">2</td>
<td align="right">0.0759215</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa03410</td>
<td align="left">Base excision repair</td>
<td align="right">33</td>
<td align="right">1</td>
<td align="right">0.0784176</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05340</td>
<td align="left">Primary immunodeficiency</td>
<td align="right">38</td>
<td align="right">1</td>
<td align="right">0.0897768</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05170</td>
<td align="left">Human immunodeficiency virus 1 infection</td>
<td align="right">212</td>
<td align="right">2</td>
<td align="right">0.0950417</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa05417</td>
<td align="left">Lipid and atherosclerosis</td>
<td align="right">215</td>
<td align="right">2</td>
<td align="right">0.0973332</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05166</td>
<td align="left">Human T-cell leukemia virus 1 infection</td>
<td align="right">219</td>
<td align="right">2</td>
<td align="right">0.1004140</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa00860</td>
<td align="left">Porphyrin and chlorophyll metabolism</td>
<td align="right">43</td>
<td align="right">1</td>
<td align="right">0.1010029</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05163</td>
<td align="left">Human cytomegalovirus infection</td>
<td align="right">225</td>
<td align="right">2</td>
<td align="right">0.1050881</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa03420</td>
<td align="left">Nucleotide excision repair</td>
<td align="right">47</td>
<td align="right">1</td>
<td align="right">0.1098890</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa05131</td>
<td align="left">Shigellosis</td>
<td align="right">247</td>
<td align="right">2</td>
<td align="right">0.1227310</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa03460</td>
<td align="left">Fanconi anemia pathway</td>
<td align="right">54</td>
<td align="right">1</td>
<td align="right">0.1252391</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">path:hsa04370</td>
<td align="left">VEGF signaling pathway</td>
<td align="right">59</td>
<td align="right">1</td>
<td align="right">0.1360492</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">path:hsa04978</td>
<td align="left">Mineral absorption</td>
<td align="right">60</td>
<td align="right">1</td>
<td align="right">0.1381959</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] org.Hs.eg.db_3.12.0       tximport_1.18.0          
 [3] patchwork_1.1.1           NMF_0.23.0               
 [5] cluster_2.1.0             rngtools_1.5             
 [7] pkgmaker_0.32.2           registry_0.5-1           
 [9] edgeR_3.32.1              limma_3.46.0             
[11] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.14.0         
[13] AnnotationFilter_1.14.0   GenomicFeatures_1.42.1   
[15] AnnotationDbi_1.52.0      Biobase_2.50.0           
[17] GenomicRanges_1.42.0      GenomeInfoDb_1.26.7      
[19] IRanges_2.24.1            S4Vectors_0.28.1         
[21] BiocGenerics_0.36.1       forcats_0.5.1            
[23] stringr_1.4.0             dplyr_1.0.4              
[25] purrr_0.3.4               readr_1.4.0              
[27] tidyr_1.1.2               tibble_3.1.2             
[29] ggplot2_3.3.5             tidyverse_1.3.0          
[31] here_1.0.1                workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                backports_1.2.1            
  [3] BiocFileCache_1.14.0        plyr_1.8.6                 
  [5] lazyeval_0.2.2              splines_4.0.2              
  [7] BiocParallel_1.24.1         gridBase_0.4-7             
  [9] digest_0.6.27               foreach_1.5.1              
 [11] htmltools_0.5.1.1           GO.db_3.12.1               
 [13] fansi_0.5.0                 magrittr_2.0.1             
 [15] memoise_2.0.0               doParallel_1.0.16          
 [17] Biostrings_2.58.0           modelr_0.1.8               
 [19] matrixStats_0.59.0          askpass_1.1                
 [21] prettyunits_1.1.1           colorspace_2.0-2           
 [23] blob_1.2.1                  rvest_0.3.6                
 [25] rappdirs_0.3.3              haven_2.3.1                
 [27] xfun_0.23                   crayon_1.4.1               
 [29] RCurl_1.98-1.3              jsonlite_1.7.2             
 [31] iterators_1.0.13            glue_1.4.2                 
 [33] gtable_0.3.0                zlibbioc_1.36.0            
 [35] XVector_0.30.0              Rsubread_2.4.2             
 [37] DelayedArray_0.16.3         scales_1.1.1               
 [39] DBI_1.1.1                   Rcpp_1.0.6                 
 [41] xtable_1.8-4                progress_1.2.2             
 [43] bit_4.0.4                   httr_1.4.2                 
 [45] RColorBrewer_1.1-2          ellipsis_0.3.2             
 [47] pkgconfig_2.0.3             XML_3.99-0.5               
 [49] farver_2.1.0                dbplyr_2.1.0               
 [51] locfit_1.5-9.4              utf8_1.2.1                 
 [53] tidyselect_1.1.0            rlang_0.4.11               
 [55] reshape2_1.4.4              later_1.1.0.1              
 [57] munsell_0.5.0               cellranger_1.1.0           
 [59] tools_4.0.2                 cachem_1.0.4               
 [61] cli_3.0.0                   generics_0.1.0             
 [63] RSQLite_2.2.5               broom_0.7.4                
 [65] evaluate_0.14               fastmap_1.1.0              
 [67] yaml_2.2.1                  knitr_1.31                 
 [69] bit64_4.0.5                 fs_1.5.0                   
 [71] xml2_1.3.2                  biomaRt_2.46.3             
 [73] compiler_4.0.2              rstudioapi_0.13            
 [75] curl_4.3                    reprex_1.0.0               
 [77] statmod_1.4.35              stringi_1.5.3              
 [79] highr_0.8                   lattice_0.20-41            
 [81] ProtGenerics_1.22.0         Matrix_1.3-2               
 [83] vctrs_0.3.8                 pillar_1.6.1               
 [85] lifecycle_1.0.0             bitops_1.0-7               
 [87] httpuv_1.5.5                rtracklayer_1.50.0         
 [89] R6_2.5.0                    promises_1.2.0.1           
 [91] codetools_0.2-18            assertthat_0.2.1           
 [93] SummarizedExperiment_1.20.0 openssl_1.4.3              
 [95] rprojroot_2.0.2             withr_2.4.2                
 [97] GenomicAlignments_1.26.0    Rsamtools_2.6.0            
 [99] GenomeInfoDbData_1.2.4      hms_1.0.0                  
[101] grid_4.0.2                  rmarkdown_2.6              
[103] MatrixGenerics_1.2.1        git2r_0.28.0               
[105] lubridate_1.7.9.2          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
