<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Limma-voom paired analysis of ALL samples</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">amnio-cell-free-RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Limma-voom paired analysis of ALL samples</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>amnio-cell-free-RNA/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200224code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200224)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200224code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200224)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreeaf2af9775305a916d4b4cf16bf152a4d0845f4c7targetblankaf2af97a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/af2af9775305a916d4b4cf16bf152a4d0845f4c7" target="_blank">af2af97</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmosamniocellfreernatreeaf2af9775305a916d4b4cf16bf152a4d0845f4c7targetblankaf2af97a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/tree/af2af9775305a916d4b4cf16bf152a4d0845f4c7" target="_blank">af2af97</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .bpipe/
    Ignored:    analysis/obsolete_analysis/
    Ignored:    analysis/salmon-ruvseq-edger.nb.html
    Ignored:    code/.bpipe/
    Ignored:    code/.rnaseq-test.groovy.swp
    Ignored:    code/obsolete_analysis/
    Ignored:    data/.bpipe/
    Ignored:    data/190717_A00692_0021_AHLLHFDSXX/
    Ignored:    data/190729_A00692_0022_AHLLHFDSXX/
    Ignored:    data/190802_A00692_0023_AHLLHFDSXX/
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX.tar
    Ignored:    data/200612_A00692_0107_AHN3YCDMXX/
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX.tar
    Ignored:    data/200626_A00692_0111_AHNJH7DMXX/
    Ignored:    data/CMV-AF-database-corrected-oct-2020.csv
    Ignored:    data/CMV-AF-database-final-included-samples.csv
    Ignored:    data/GONE4.10.13.txt
    Ignored:    data/HK_exons.csv
    Ignored:    data/HK_exons.txt
    Ignored:    data/HK_genes.txt
    Ignored:    data/IPA molecule summary.xls
    Ignored:    data/IPA-molecule-summary.csv
    Ignored:    data/brain-development-geneset.txt
    Ignored:    data/deduped_rRNA_coverage.txt
    Ignored:    data/gene-transcriptome-analysis/
    Ignored:    data/hg38_rRNA.bed
    Ignored:    data/hg38_rRNA.saf
    Ignored:    data/ignore-overlap-mapping/
    Ignored:    data/ignore/
    Ignored:    data/joindata.csv
    Ignored:    data/metadata.csv
    Ignored:    data/rds/
    Ignored:    data/salmon-pilot-analysis/
    Ignored:    data/samples.csv
    Ignored:    data/star-genome-analysis/
    Ignored:    output/c2Ens.RData
    Ignored:    output/c5Ens.RData
    Ignored:    output/hEns.RData
    Ignored:    output/keggEns.RData

Untracked files:
    Untracked:  code/output.R
    Untracked:  code/rnaseq.salmon-quant.groovy
    Untracked:  code/rnaseq.star.count-exons.groovy
    Untracked:  code/rnaseq.star.groovy
    Untracked:  output/salmon-limma-voom-GO-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c2Cam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c5Cam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-c5Cam.csv
    Untracked:  output/salmon-limma-voom-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-hCam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom-keggCam-exclude-CMV11.csv
    Untracked:  output/salmon-limma-voom.csv
    Untracked:  output/salmon-ruvseq-edger.csv
    Untracked:  output/star-fc-limma-voom-all-gsea-c2.csv
    Untracked:  output/star-fc-limma-voom-all-gsea-c5.csv
    Untracked:  output/star-fc-limma-voom-all.csv
    Untracked:  output/star-fc-limma-voom-no_us_ab-gsea-c2.csv
    Untracked:  output/star-fc-limma-voom-no_us_ab-gsea-c5.csv
    Untracked:  output/star-fc-limma-voom-no_us_ab.csv
    Untracked:  output/star-fc-ruv-all-gsea-c2.csv
    Untracked:  output/star-fc-ruv-all-gsea-c5.csv
    Untracked:  output/star-fc-ruv-all.csv
    Untracked:  output/star-fc-ruv-no_us_ab-gsea-c2.csv
    Untracked:  output/star-fc-ruv-no_us_ab-gsea-c5.csv
    Untracked:  output/star-fc-ruv-no_us_ab.csv
    Untracked:  renv.lock

Unstaged changes:
    Modified:   .gitignore
    Deleted:    analysis/STAR-DEXSeq-Summed.Rmd
    Deleted:    analysis/salmon-BANDITS.Rmd
    Deleted:    analysis/salmon-DRIMseq.Rmd
    Deleted:    analysis/salmon-RUV-all.Rmd
    Deleted:    analysis/salmon-SatuRn.Rmd
    Deleted:    code/rnaseq-with-salmon-quant.groovy
    Deleted:    code/rnaseq.groovy
    Deleted:    code/salmon-quant-trimmed.groovy
    Deleted:    code/unmapped-pipe.groovy

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/STAR-FC-all.Rmd</code>) and HTML (<code>docs/STAR-FC-all.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/af2af9775305a916d4b4cf16bf152a4d0845f4c7/analysis/STAR-FC-all.Rmd" target="_blank">af2af97</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/STAR-FC-all.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/2b6479ab40b7e78b559ce7d3912860d1423a98ca/analysis/STAR-FC-all.Rmd" target="_blank">2b6479a</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-07-26
</td>
<td>
Move/remove olds files.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/76d6992c5237835cde8e6ef137f6d55ffbe64912/analysis/STAR-FC-all.Rmd" target="_blank">76d6992</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-12-04
</td>
<td>
Explore rRNA counts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/d96660bec01347bcec29e84c1c4313e95abc8d04/docs/STAR-FC-all.html" target="_blank">d96660b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/d32e63b606d16b69ae5c35eae7e7615e92ac90c3/analysis/STAR-FC-all.Rmd" target="_blank">d32e63b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
<td>
wflow_publish(c(“analysis/STAR-FC-all.Rmd”, “analysis/STAR-FC-exclude-US-ab.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/STAR-FC-all.html" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/787e0a195e4863f0eb47d672e5689a3683f6ab43/analysis/STAR-FC-all.Rmd" target="_blank">787e0a1</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
<td>
wflow_publish(c(“analysis/STAR-FC-all.Rmd”, “analysis/STAR-FC-exclude-US-ab.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/STAR-FC-all.html" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/2ee5593cf8f25f38c0fae29032e77240be940634/analysis/STAR-FC-all.Rmd" target="_blank">2ee5593</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/STAR-FC-all.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<!-- ```{r} -->
<!-- foo <- read.table(here("data/sorted/rRNA_counts.txt"), header = FALSE, -->
<!--                   stringsAsFactors = FALSE) -->
<!-- rownames(foo) <- sapply(1:nrow(foo), function(i)  -->
<!--   paste(foo[i,1:4], collapse = ".")) -->
<!-- foo <- foo[,13:72] -->
<!-- summary(colSums(foo)) -->
<!-- blah <- read.table(here("data/deduped/rRNA_counts_deduped.txt"), header = FALSE, -->
<!--                   stringsAsFactors = FALSE) -->
<!-- rownames(blah) <- sapply(1:nrow(blah), function(i)  -->
<!--   paste(blah[i,1:4], collapse = ".")) -->
<!-- blah <- blah[,13:72] -->
<!-- summary(colSums(blah)) -->
<!-- ``` -->
<p>The data showed some adapter contamination and sequence duplication issues. Adapters were removed using <code>Trimmomatic</code> and both paired and unpaired reads were retained. Only paired reads were initially mapped with <code>Star</code> in conjunction with <code>GRCh38</code> and <code>gencode_v34</code> to detect all junctions, across all samples. Paired and unpaired reads were then mapped to <code>GRCh38</code> separately using <code>Star</code>. Duplicates were removed from paired and unpaired mapped data using <code>Picard MarkDuplicates</code>. Reads were then counted across features from <code>gencode_v34</code> using <code>featureCounts</code>.</p>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Set up <code>DGElist</code> object for downstream analysis. Sum paired and unpaired counts prior to downstream analysis.</p>
<pre><code>An object of class &quot;DGEList&quot;
$counts
  CMV30 CMV31 CMV8 CMV9 CMV26 CMV27 CMV14 CMV15 CMV20 CMV21 CMV1 CMV2 CMV3 CMV4
1     0     0    0    0     2     2     0     1     0     1    1    0    1    0
2    58    95   58   59   113   101    60    48    79    71   54   63   39   46
3     1     0    0    0     0     0     0     0     0     0    0    0    0    0
4     0     0    0    0     0     0     0     0     0     0    1    0    1    0
5     0     0    0    0     0     0     0     0     0     0    0    0    0    0
  CMV10 CMV11 CMV18 CMV19 CMV35 Corriel NTC-2 CMV51 CMV52 CMV53 CMV54 CMV56
1     0     0     0     0     1       1     0     0     0     0     0     0
2    62    35    51    45    59      84     0    63    28    49    46    37
3     0     0     0     0     0       1     0     0     0     0     0     0
4     0     0     0     0     0       0     0     0     0     0     0     0
5     0     0     0     0     0       0     0     0     0     0     0     0
  CMV57 CMV58 CMV60 CMV61
1     0     2     1     2
2    59    82    44    36
3     0     0     0     0
4     0     0     1     0
5     0     0     0     0
60664 more rows ...

$samples
      group lib.size norm.factors
CMV30     1  4673630            1
CMV31     1  5232010            1
CMV8      1  3594801            1
CMV9      1  3425478            1
CMV26     1  4892776            1
25 more rows ...

$genes
             Geneid Length
1 ENSG00000223972.5   1735
2 ENSG00000227232.5   1351
3 ENSG00000278267.1     68
4 ENSG00000243485.5   1021
5 ENSG00000284332.1    138
60664 more rows ...</code></pre>
<p>Load sample information and file names.</p>
<pre><code># A tibble: 26 x 7
   id    CMV_status pair  sex   GA_at_amnio indication batch
   &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt;      &lt;fct&gt;
 1 CMV30 pos        L1    F     21          no_us_ab   1    
 2 CMV31 neg        L1    F     21          no_us_ab   1    
 3 CMV8  neg        L2    F     23          no_us_ab   1    
 4 CMV9  pos        L2    F     23          no_us_ab   1    
 5 CMV26 pos        L3    F     22          no_us_ab   1    
 6 CMV14 neg        L4    F     21          no_us_ab   1    
 7 CMV15 pos        L4    F     22          no_us_ab   1    
 8 CMV20 pos        L5    M     21          no_us_ab   1    
 9 CMV21 neg        NC1   F     21          no_us_ab   1    
10 CMV1  pos        M1    F     21          no_us_ab   1    
# … with 16 more rows</code></pre>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>Genes that do not have an adequate number of reads in any sample should be filtered out prior to downstream analyses. From a biological perspective, genes that are not expressed at a biologically meaningful level in any condition are not of interest. Statistically, we get a better estimate of the mean-variance relationship in the data and reduce the number of statistical tests that are performed during differential expression analyses.</p>
<p>Filter out lowly expressed genes and genes without Entrez IDs and calculate TMM normalization factors.</p>
<pre><code>An object of class &quot;DGEList&quot;
$counts
   CMV30 CMV31 CMV8 CMV9 CMV26 CMV14 CMV15 CMV20 CMV21 CMV1 CMV2 CMV3 CMV4
32    20    36   28   42    28    25    19    26    18   32   25   55   31
52    88    73   55   43    55    75    53    64    55   61   68   36   62
55     6    15   17   15    15     9    14    13    12    7   10   16   12
63   148   172  148  126   175   179   176   141   179  155  194  121  123
64    14    15   14   11    20    14     9    13    14   13   13   19   10
   CMV10 CMV11 CMV19 CMV35 CMV51 CMV52 CMV53 CMV54 CMV56 CMV57 CMV58 CMV60
32    15    11    22    24    25    19    23    20    18    23    24     8
52    69    11    35    47    59    49    42    37    38    49    65    49
55     7     2    11     9    19     9     6    12     7     7     7     5
63   156    44   107   164   144    71    71   137   136   122   131    88
64     9     3     6     9    14     3     9    12     5     7    12     5
   CMV61
32    20
52    49
55     6
63   108
64     3
12727 more rows ...

$samples
      group lib.size norm.factors
CMV30     1  4673630     1.017563
CMV31     1  5232010     1.052277
CMV8      1  3594801     1.052059
CMV9      1  3425478     1.026378
CMV26     1  4892776     1.068518
21 more rows ...

$genes
               Geneid Length         Ensembl       symbol  entrezid
32 ENSG00000230021.10   5495 ENSG00000230021 RP5-857K21.4 101928626
52 ENSG00000228794.10  15682 ENSG00000228794    LINC01128    643837
55  ENSG00000230368.2   1971 ENSG00000230368       FAM41C    284593
63 ENSG00000188976.11   5540 ENSG00000188976        NOC2L     26155
64 ENSG00000187961.14   3402 ENSG00000187961       KLHL17    339451
12727 more rows ...

$group
[1] &quot;pos&quot; &quot;neg&quot; &quot;neg&quot; &quot;pos&quot; &quot;pos&quot;
21 more elements ...</code></pre>
<p>Plotting the distribution log-CPM values shows that a majority of genes within each sample are either not expressed or lowly-expressed with log-CPM values that are small or negative.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-5-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Although in excess of 30 million reads were obtained per sample, we can see that after mapping, duplicate removal and quantification of gene expression the median library size is just under than 4 million reads. This suggests that we are likely to only be capturing the most abundant cfRNAs.</p>
<p>It is assumed that all samples should have a similar range and distribution of expression values. The raw data looks fairly uniform between samples, although TMM normalization further improves this.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-6-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-6-1.png" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of exploring the relationships between the samples and identifying structure in the data. The following series of MDS plots examines the first four principal components. The samples are coloured by various known features of the samples such as CMV Status and foetal sex. The MDS plots do not show the samples strongly clustering by any of the known features of the data, although there does seem to be some separation between the CMV positive and negative samples in the 1st and 2nd principal components. This indicates that there are possibly some differentially expressed genes between CMV positive and negative samples.</p>
<p>A weak batch effect is also evident in the 3rd principal component, when we examine the plots coloured by batch.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-7-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-7-1.png" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<p>Due to the variability in the data, the TMM normalised data was transformed using <code>voomWithQualityWeights</code>. This takes into account the differing library sizes and the mean variance relationship in the data as well as calculating sample-specific quality weights. Linear models were fit in <code>limma</code>, taking into account the <code>voom</code> weights. The CMV positive samples were compared to the CMV negative samples, taking into account the sample pairs. A summary of the number of differentially expressed genes is shown below.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-8-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-8-1.png" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>       pos - neg
Down          17
NotSig     12505
Up           210</code></pre>
<p>There were 17 down-regulated and 210 up-regulated genes between CMV positive and CMV negative samples at FDR &lt; 0.05.</p>
<p>These are the top 10 differentially expressed genes.</p>
<pre><code>Warning: The `path` argument of `write_csv()` is deprecated as of readr 1.4.0.
Please use the `file` argument instead.</code></pre>
<pre><code>                  Geneid Length         Ensembl symbol entrezid    logFC
45751 ENSG00000140853.15  12386 ENSG00000140853  NLRC5    84166 2.797155
8687  ENSG00000115415.20   9770 ENSG00000115415  STAT1     6772 1.199505
19329 ENSG00000206337.12  11058 ENSG00000206337   HCP5    10866 4.440444
15204 ENSG00000137628.17   6746 ENSG00000137628  DDX60    55601 1.590297
8282   ENSG00000115267.8   5094 ENSG00000115267  IFIH1    64135 1.069293
14201  ENSG00000138646.9   4764 ENSG00000138646  HERC5    51191 2.943603
2095  ENSG00000137965.11   2038 ENSG00000137965  IFI44    10561 1.954652
46903 ENSG00000132530.17   6615 ENSG00000132530   XAF1    54739 2.858189
27479 ENSG00000107201.10   4640 ENSG00000107201  DDX58    23586 1.110450
19309 ENSG00000234745.11   3051 ENSG00000234745  HLA-B     3106 3.138597
       AveExpr        t      P.Value    adj.P.Val        B
45751 2.163443 9.405942 4.679903e-08 0.0005193123 7.426915
8687  7.926887 8.204584 3.789399e-07 0.0012061657 6.855860
19329 1.845420 9.507407 8.157593e-08 0.0005193123 6.512669
15204 5.968763 7.878387 9.119445e-07 0.0020294388 6.033060
8282  5.614202 7.524000 9.868764e-07 0.0020294388 5.912610
14201 1.990993 8.409535 2.397649e-07 0.0010175624 5.881695
2095  4.394309 7.738868 1.138190e-06 0.0020294388 5.756492
46903 4.797937 7.488143 1.705713e-06 0.0024130150 5.359573
27479 5.639234 7.054829 2.755727e-06 0.0035085914 4.933196
19309 4.348936 7.058890 3.844793e-06 0.0041278571 4.573300</code></pre>
<p>The following plots show the expression of the top 12 ranked differentially expressed genes for CMV positive and CMV negative samples. Although there is significant variability within the groups and the log2 fold changes are not large, there are obvious differences in expression for the top ranked genes.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-10-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-11-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-11-1.png" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level1">
<h1>Gene set enrichment analysis (GSEA)</h1>
<p>Testing for enrichment of Gene Ontology (GO) categories among statistically significant differentially expressed genes.</p>
<pre><code>                                                  Term Ont    N DE         P.DE
GO:0009607                 response to biotic stimulus  BP  876 62 9.627568e-22
GO:0051607                   defense response to virus  BP  178 30 5.069312e-21
GO:0043207        response to external biotic stimulus  BP  849 60 5.634435e-21
GO:0051707                  response to other organism  BP  849 60 5.634435e-21
GO:0045087                      innate immune response  BP  516 47 8.260606e-21
GO:0009615                           response to virus  BP  231 33 1.026368e-20
GO:0071357      cellular response to type I interferon  BP   70 21 1.410360e-20
GO:0060337         type I interferon signaling pathway  BP   70 21 1.410360e-20
GO:0034340               response to type I interferon  BP   74 21 5.205041e-20
GO:0098542          defense response to other organism  BP  628 49 7.916235e-19
GO:0006955                             immune response  BP 1177 67 2.011052e-18
GO:0006952                            defense response  BP  913 57 2.286585e-17
GO:0009605               response to external stimulus  BP 1674 76 1.969940e-15
GO:0002376                       immune system process  BP 1887 80 1.245515e-14
GO:0044419  interspecies interaction between organisms  BP 1439 68 1.548258e-14
GO:0034341                response to interferon-gamma  BP  121 20 3.483005e-14
GO:0071346       cellular response to interferon-gamma  BP  109 19 5.693487e-14
GO:0002252                     immune effector process  BP  776 47 8.171695e-14
GO:0060333 interferon-gamma-mediated signaling pathway  BP   61 15 1.316373e-13
GO:0034097                        response to cytokine  BP  802 46 1.113023e-12</code></pre>
<p>GSEA helps us to interpret the results of a differential expression analysis. The <code>camera</code> function performs a competitive test to assess whether the genes in a given set are highly ranked in terms of differential expression relative to genes that are not in the set. We have tested several collections of gene sets from the Broad Institute’s Molecular Signatures Database <a href="http://software.broadinstitute.org/gsea/msigdb/index.jsp">MSigDB</a>.</p>
<p>Build gene set indexes.</p>
<pre><code>[1] &quot;Loading MSigDB Gene Sets ... &quot;
[1] &quot;Loaded gene sets for the collection h ...&quot;
[1] &quot;Indexed the collection h ...&quot;
[1] &quot;Created annotation for the collection h ...&quot;
[1] &quot;Loaded gene sets for the collection c2 ...&quot;
[1] &quot;Indexed the collection c2 ...&quot;
[1] &quot;Created annotation for the collection c2 ...&quot;
[1] &quot;Loaded gene sets for the collection c5 ...&quot;
[1] &quot;Indexed the collection c5 ...&quot;
[1] &quot;Created annotation for the collection c5 ...&quot;
[1] &quot;Building KEGG pathways annotation object ... &quot;</code></pre>
<p>The GO gene sets consist of genes annotated by the same GO terms.</p>
<pre><code>                                                                     NGenes
GO_RESPONSE_TO_TYPE_I_INTERFERON                                         48
GO_DEFENSE_RESPONSE_TO_VIRUS                                            111
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_PEPTIDE_ANTIGEN     11
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_ANTIGEN             13
GO_INTERFERON_GAMMA_MEDIATED_SIGNALING_PATHWAY                           43
GO_DIGESTION                                                             56
GO_RESPONSE_TO_INTERFERON_GAMMA                                          71
GO_CELLULAR_RESPONSE_TO_INTERFERON_GAMMA                                 59
GO_REGULATION_OF_CELL_KILLING                                            32
GO_REGULATION_OF_LEUKOCYTE_MEDIATED_CYTOTOXICITY                         29
GO_RESPONSE_TO_VIRUS                                                    166
GO_SOLUTE_PROTON_SYMPORTER_ACTIVITY                                      18
GO_CYTOKINE_MEDIATED_SIGNALING_PATHWAY                                  245
GO_NEGATIVE_REGULATION_OF_CELL_KILLING                                   12
GO_MHC_PROTEIN_COMPLEX                                                   11
GO_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                            12
GO_RESPONSE_TO_INTERFERON_ALPHA                                          18
GO_POSITIVE_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                    8
GO_REGULATION_OF_PLASMA_LIPOPROTEIN_PARTICLE_LEVELS                      29
GO_MHC_CLASS_I_PROTEIN_COMPLEX                                            7
                                                                     Direction
GO_RESPONSE_TO_TYPE_I_INTERFERON                                            Up
GO_DEFENSE_RESPONSE_TO_VIRUS                                                Up
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_PEPTIDE_ANTIGEN        Up
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_ANTIGEN                Up
GO_INTERFERON_GAMMA_MEDIATED_SIGNALING_PATHWAY                              Up
GO_DIGESTION                                                                Up
GO_RESPONSE_TO_INTERFERON_GAMMA                                             Up
GO_CELLULAR_RESPONSE_TO_INTERFERON_GAMMA                                    Up
GO_REGULATION_OF_CELL_KILLING                                               Up
GO_REGULATION_OF_LEUKOCYTE_MEDIATED_CYTOTOXICITY                            Up
GO_RESPONSE_TO_VIRUS                                                        Up
GO_SOLUTE_PROTON_SYMPORTER_ACTIVITY                                         Up
GO_CYTOKINE_MEDIATED_SIGNALING_PATHWAY                                      Up
GO_NEGATIVE_REGULATION_OF_CELL_KILLING                                      Up
GO_MHC_PROTEIN_COMPLEX                                                      Up
GO_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                               Up
GO_RESPONSE_TO_INTERFERON_ALPHA                                             Up
GO_POSITIVE_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                      Up
GO_REGULATION_OF_PLASMA_LIPOPROTEIN_PARTICLE_LEVELS                         Up
GO_MHC_CLASS_I_PROTEIN_COMPLEX                                              Up
                                                                           PValue
GO_RESPONSE_TO_TYPE_I_INTERFERON                                     5.918561e-26
GO_DEFENSE_RESPONSE_TO_VIRUS                                         5.511815e-14
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_PEPTIDE_ANTIGEN 4.049629e-13
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_ANTIGEN         4.365482e-13
GO_INTERFERON_GAMMA_MEDIATED_SIGNALING_PATHWAY                       6.081715e-13
GO_DIGESTION                                                         4.430786e-11
GO_RESPONSE_TO_INTERFERON_GAMMA                                      6.699823e-11
GO_CELLULAR_RESPONSE_TO_INTERFERON_GAMMA                             6.410894e-10
GO_REGULATION_OF_CELL_KILLING                                        1.371315e-08
GO_REGULATION_OF_LEUKOCYTE_MEDIATED_CYTOTOXICITY                     2.794535e-08
GO_RESPONSE_TO_VIRUS                                                 2.890393e-08
GO_SOLUTE_PROTON_SYMPORTER_ACTIVITY                                  5.791605e-08
GO_CYTOKINE_MEDIATED_SIGNALING_PATHWAY                               9.989903e-08
GO_NEGATIVE_REGULATION_OF_CELL_KILLING                               1.023995e-07
GO_MHC_PROTEIN_COMPLEX                                               2.027142e-07
GO_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                        2.805316e-07
GO_RESPONSE_TO_INTERFERON_ALPHA                                      3.328300e-07
GO_POSITIVE_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY               3.395518e-07
GO_REGULATION_OF_PLASMA_LIPOPROTEIN_PARTICLE_LEVELS                  3.468113e-07
GO_MHC_CLASS_I_PROTEIN_COMPLEX                                       3.535987e-07
                                                                              FDR
GO_RESPONSE_TO_TYPE_I_INTERFERON                                     3.646426e-22
GO_DEFENSE_RESPONSE_TO_VIRUS                                         1.697915e-10
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_PEPTIDE_ANTIGEN 6.723934e-10
GO_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_ENDOGENOUS_ANTIGEN         6.723934e-10
GO_INTERFERON_GAMMA_MEDIATED_SIGNALING_PATHWAY                       7.493889e-10
GO_DIGESTION                                                         4.549679e-08
GO_RESPONSE_TO_INTERFERON_GAMMA                                      5.896801e-08
GO_CELLULAR_RESPONSE_TO_INTERFERON_GAMMA                             4.937189e-07
GO_REGULATION_OF_CELL_KILLING                                        9.387410e-06
GO_REGULATION_OF_LEUKOCYTE_MEDIATED_CYTOTOXICITY                     1.618883e-05
GO_RESPONSE_TO_VIRUS                                                 1.618883e-05
GO_SOLUTE_PROTON_SYMPORTER_ACTIVITY                                  2.973507e-05
GO_CYTOKINE_MEDIATED_SIGNALING_PATHWAY                               4.506309e-05
GO_NEGATIVE_REGULATION_OF_CELL_KILLING                               4.506309e-05
GO_MHC_PROTEIN_COMPLEX                                               8.326149e-05
GO_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY                        1.080222e-04
GO_RESPONSE_TO_INTERFERON_ALPHA                                      1.089261e-04
GO_POSITIVE_REGULATION_OF_T_CELL_MEDIATED_CYTOTOXICITY               1.089261e-04
GO_REGULATION_OF_PLASMA_LIPOPROTEIN_PARTICLE_LEVELS                  1.089261e-04
GO_MHC_CLASS_I_PROTEIN_COMPLEX                                       1.089261e-04</code></pre>
<p>The Hallmark gene sets are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.</p>
<pre><code>                                   NGenes Direction       PValue          FDR
HALLMARK_INTERFERON_ALPHA_RESPONSE     81        Up 4.065265e-46 2.032632e-44
HALLMARK_INTERFERON_GAMMA_RESPONSE    147        Up 5.643640e-33 1.410910e-31
HALLMARK_E2F_TARGETS                  196      Down 5.883620e-08 9.806033e-07
HALLMARK_INFLAMMATORY_RESPONSE        116        Up 8.645708e-08 1.080713e-06
HALLMARK_MYC_TARGETS_V1               198      Down 1.853964e-07 1.853964e-06
HALLMARK_TNFA_SIGNALING_VIA_NFKB      162        Up 2.346065e-06 1.955054e-05
HALLMARK_KRAS_SIGNALING_UP            133        Up 3.327963e-06 2.377116e-05
HALLMARK_IL6_JAK_STAT3_SIGNALING       47        Up 4.590363e-05 2.868977e-04
HALLMARK_COMPLEMENT                   140        Up 1.187809e-04 6.598941e-04
HALLMARK_G2M_CHECKPOINT               194      Down 1.898061e-04 9.490303e-04
HALLMARK_ALLOGRAFT_REJECTION          102        Up 3.379307e-04 1.536049e-03
HALLMARK_MITOTIC_SPINDLE              192      Down 1.686661e-03 7.027753e-03
HALLMARK_MYC_TARGETS_V2                54      Down 9.271710e-03 3.566042e-02
HALLMARK_DNA_REPAIR                   145      Down 1.106814e-02 3.952908e-02
HALLMARK_XENOBIOTIC_METABOLISM        139        Up 2.211981e-02 7.373271e-02
HALLMARK_COAGULATION                   84        Up 2.375926e-02 7.424769e-02
HALLMARK_IL2_STAT5_SIGNALING          142        Up 3.101322e-02 9.121536e-02
HALLMARK_ADIPOGENESIS                 182        Up 4.278309e-02 1.188419e-01
HALLMARK_MYOGENESIS                   122      Down 5.072696e-02 1.334920e-01
HALLMARK_ANDROGEN_RESPONSE             96        Up 5.671960e-02 1.417990e-01</code></pre>
<p>Barcode plots show the enrichment of gene sets among up or down-regulated genes. The following barcode plots show the enrichment of the top 4 hallmark gene sets among the genes differentially expressed between CMV positive and CMV negative samples.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-16-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-16-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
<p>The curated gene sets are compiled from online pathway databases, publications in PubMed, and knowledge of domain experts.</p>
<pre><code>                                                  NGenes Direction       PValue
MOSERLE_IFNA_RESPONSE                                 27        Up 4.452715e-42
BROWNE_INTERFERON_RESPONSIVE_GENES                    55        Up 2.853789e-40
HECKER_IFNB1_TARGETS                                  59        Up 1.074420e-36
SANA_RESPONSE_TO_IFNG_UP                              48        Up 4.805722e-36
FARMER_BREAST_CANCER_CLUSTER_1                        15        Up 2.184050e-27
DAUER_STAT3_TARGETS_DN                                49        Up 8.940232e-27
BOSCO_INTERFERON_INDUCED_ANTIVIRAL_MODULE             59        Up 4.536072e-26
SANA_TNF_SIGNALING_UP                                 63        Up 1.091778e-24
EINAV_INTERFERON_SIGNATURE_IN_CANCER                  24        Up 5.226263e-22
REACTOME_INTERFERON_ALPHA_BETA_SIGNALING              44        Up 7.206877e-22
BOWIE_RESPONSE_TO_TAMOXIFEN                           16        Up 7.637944e-21
RADAEVA_RESPONSE_TO_IFNA1_UP                          45        Up 1.858830e-20
BENNETT_SYSTEMIC_LUPUS_ERYTHEMATOSUS                  24        Up 2.002087e-20
BOWIE_RESPONSE_TO_EXTRACELLULAR_MATRIX                16        Up 1.543851e-19
ZHANG_INTERFERON_RESPONSE                             18        Up 1.924959e-18
DER_IFN_ALPHA_RESPONSE_UP                             67        Up 9.637826e-18
SEITZ_NEOPLASTIC_TRANSFORMATION_BY_8P_DELETION_UP     56        Up 1.992852e-17
KRASNOSELSKAYA_ILF3_TARGETS_UP                        30        Up 4.035282e-17
TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP           129        Up 1.819293e-16
ROETH_TERT_TARGETS_UP                                 13        Up 5.131494e-16
                                                           FDR
MOSERLE_IFNA_RESPONSE                             1.665761e-38
BROWNE_INTERFERON_RESPONSIVE_GENES                5.338013e-37
HECKER_IFNB1_TARGETS                              1.339802e-33
SANA_RESPONSE_TO_IFNG_UP                          4.494551e-33
FARMER_BREAST_CANCER_CLUSTER_1                    1.634106e-24
DAUER_STAT3_TARGETS_DN                            5.574235e-24
BOSCO_INTERFERON_INDUCED_ANTIVIRAL_MODULE         2.424206e-23
SANA_TNF_SIGNALING_UP                             5.105427e-22
EINAV_INTERFERON_SIGNATURE_IN_CANCER              2.172383e-19
REACTOME_INTERFERON_ALPHA_BETA_SIGNALING          2.696093e-19
BOWIE_RESPONSE_TO_TAMOXIFEN                       2.597595e-18
RADAEVA_RESPONSE_TO_IFNA1_UP                      5.761389e-18
BENNETT_SYSTEMIC_LUPUS_ERYTHEMATOSUS              5.761389e-18
BOWIE_RESPONSE_TO_EXTRACELLULAR_MATRIX            4.125391e-17
ZHANG_INTERFERON_RESPONSE                         4.800848e-16
DER_IFN_ALPHA_RESPONSE_UP                         2.253444e-15
SEITZ_NEOPLASTIC_TRANSFORMATION_BY_8P_DELETION_UP 4.385447e-15
KRASNOSELSKAYA_ILF3_TARGETS_UP                    8.386661e-15
TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP        3.582092e-14
ROETH_TERT_TARGETS_UP                             9.598459e-14</code></pre>
<p>The following barcode plots show the enrichment of the top 4 curated gene sets among the genes differentially expressed between CMV positive and CMV negative samples.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-18-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-18-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
<p>The KEGG gene sets encompass all of the pathways defined in the <a href="https://www.genome.jp/kegg/pathway.html">Kegg pathway database</a>.</p>
<pre><code>                                       NGenes Direction       PValue
Mineral absorption                         36        Up 1.375770e-10
Fat digestion and absorption               21        Up 4.933923e-08
Type I diabetes mellitus                   12        Up 4.088334e-07
Antigen processing and presentation        39        Up 9.194603e-07
Graft-versus-host disease                   8        Up 1.542188e-06
Autoimmune thyroid disease                 10        Up 2.363013e-06
Cytokine-cytokine receptor interaction     83        Up 2.773616e-06
Allograft rejection                         9        Up 7.401576e-06
Ribosome                                  125      Down 1.150378e-05
Spliceosome                               128      Down 1.501836e-05
Vitamin digestion and absorption           19        Up 1.620295e-05
NOD-like receptor signaling pathway       117        Up 1.756741e-05
Bile secretion                             34        Up 2.845918e-05
Hepatitis C                                98        Up 4.236596e-05
Influenza A                               119        Up 1.399519e-04
Measles                                    89        Up 2.186454e-04
Maturity onset diabetes of the young        9        Up 3.538335e-04
Cell adhesion molecules (CAMs)             54        Up 3.720338e-04
Mismatch repair                            22      Down 4.258843e-04
Leishmaniasis                              39        Up 4.565431e-04
                                                FDR
Mineral absorption                     4.003492e-08
Fat digestion and absorption           7.178858e-06
Type I diabetes mellitus               3.965684e-05
Antigen processing and presentation    6.689074e-05
Graft-versus-host disease              8.975534e-05
Autoimmune thyroid disease             1.146062e-04
Cytokine-cytokine receptor interaction 1.153032e-04
Allograft rejection                    2.692323e-04
Ribosome                               3.719557e-04
Spliceosome                            4.260096e-04
Vitamin digestion and absorption       4.260096e-04
NOD-like receptor signaling pathway    4.260096e-04
Bile secretion                         6.370478e-04
Hepatitis C                            8.806067e-04
Influenza A                            2.715068e-03
Measles                                3.976612e-03
Maturity onset diabetes of the young   6.014546e-03
Cell adhesion molecules (CAMs)         6.014546e-03
Mismatch repair                        6.522754e-03
Leishmaniasis                          6.642703e-03</code></pre>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-20-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-20-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL</code></pre>
</div>
<div id="brain-development-genes" class="section level1">
<h1>Brain development genes</h1>
<p>Test only the specialised brain development gene set for differential expression between CMV positive and CMV negative samples. This reduces the multiple testing burden and can identify DEGs from a particular set of interest.</p>
<pre><code># A tibble: 51 x 1
   BRAIN_DEV 
   &lt;chr&gt;     
 1 AC139768.1
 2 ADGRG1    
 3 AFF2      
 4 ALK       
 5 ALX1      
 6 BPTF      
 7 CDK5R1    
 8 CEP290    
 9 CLN5      
10 CNTN4     
# … with 41 more rows</code></pre>
<pre><code>       pos - neg
Down           1
NotSig        26
Up             0</code></pre>
<pre><code>                  Geneid Length         Ensembl  symbol entrezid        logFC
54718 ENSG00000053438.11   1329 ENSG00000053438    NNAT     4826 -0.658397816
46775 ENSG00000040531.16   4949 ENSG00000040531    CTNS     1497  0.207087793
47853  ENSG00000176749.9   3948 ENSG00000176749  CDK5R1     8851  0.197638751
59499 ENSG00000102038.15   4363 ENSG00000102038 SMARCA1     6594 -0.169096017
28963 ENSG00000167081.18   4904 ENSG00000167081    PBX3     5090  0.162490669
7681  ENSG00000074047.21   7341 ENSG00000074047    GLI2     2736 -0.136586759
59824 ENSG00000155966.14  14241 ENSG00000155966    AFF2     2334 -0.114952294
13911  ENSG00000132467.4   2020 ENSG00000132467    UTP3    57050  0.103859874
3760  ENSG00000185630.19  21370 ENSG00000185630    PBX1     5087 -0.099918011
51917 ENSG00000130479.11   4933 ENSG00000130479   MAP1S    55201  0.089549092
9697  ENSG00000144619.15   9123 ENSG00000144619   CNTN4   152330  0.015179716
45778 ENSG00000205336.13   8659 ENSG00000205336  ADGRG1     9289  0.131955954
58386 ENSG00000158352.15  10333 ENSG00000158352 SHROOM4    57477  0.080489267
24536  ENSG00000164690.8   5234 ENSG00000164690     SHH     6469  0.069843390
39080 ENSG00000102805.16  22814 ENSG00000102805    CLN5     1203 -0.063771691
42132 ENSG00000114062.21  12808 ENSG00000114062   UBE3A     7337  0.050810995
10970 ENSG00000185008.17  16663 ENSG00000185008   ROBO2     6092 -0.036031512
33637 ENSG00000110697.13   6442 ENSG00000110697 PITPNM1     9600  0.089169404
2823  ENSG00000092621.12   9217 ENSG00000092621   PHGDH    26227  0.030791118
1293  ENSG00000131238.17   5088 ENSG00000131238    PPT1     5538  0.034873502
16275 ENSG00000164258.12   1174 ENSG00000164258  NDUFS4     4724  0.019761214
49155 ENSG00000171634.18  15119 ENSG00000171634    BPTF     2186  0.022251393
57747 ENSG00000146950.13   8218 ENSG00000146950 SHROOM2      357 -0.025131407
23659 ENSG00000128573.26  16334 ENSG00000128573   FOXP2    93986  0.040384825
47791 ENSG00000196712.18  27130 ENSG00000196712     NF1     4763  0.001432021
37089 ENSG00000198707.16  10442 ENSG00000198707  CEP290    80184 -0.009571616
54624 ENSG00000198646.14  11133 ENSG00000198646   NCOA6    23054 -0.006603637
       AveExpr           t      P.Value  adj.P.Val          B
54718 6.704590 -4.03247492 0.0008889658 0.02400208 -0.5600968
46775 2.783450  2.12484119 0.0481112739 0.64950220 -3.8586145
47853 3.817792  1.70114074 0.1074223665 0.81797536 -4.7434003
59499 7.397718 -1.61608178 0.1247714223 0.81797536 -5.0555150
28963 4.749400  1.42912478 0.1713709361 0.81797536 -5.1513516
7681  2.125700 -0.92352083 0.3682863535 0.85268245 -5.1713022
59824 1.620371 -0.51390733 0.6137403741 0.95281611 -5.2117635
13911 4.929279  1.33379242 0.1993605308 0.81797536 -5.2983565
3760  8.662670 -1.28478125 0.2156034668 0.81797536 -5.4118594
51917 4.779260  1.16607815 0.2592187731 0.81797536 -5.5049297
9697  2.365792  0.10097458 0.9207177180 0.97006631 -5.5409852
45778 4.413092  0.99341767 0.3346552258 0.85268245 -5.6086049
58386 5.385050  1.13258409 0.2726584540 0.81797536 -5.6225705
24536 3.394605  0.51655473 0.6119286850 0.95281611 -5.6786584
39080 3.405381 -0.73383954 0.4727578109 0.91174721 -5.7417837
42132 7.708283  0.90263101 0.3789699778 0.85268245 -5.8709223
10970 3.886193 -0.24404785 0.8101607935 0.95281611 -5.9016656
33637 5.297035  0.75961566 0.4580529238 0.91174721 -5.9082439
2823  3.987221  0.27427429 0.7870829789 0.95281611 -5.9084511
1293  4.185204  0.38936485 0.7017004855 0.95281611 -5.9459601
16275 4.834618  0.24191244 0.8116581647 0.95281611 -6.0689442
49155 9.098445  0.33664492 0.7403851592 0.95281611 -6.1661418
57747 5.863156 -0.40069222 0.6934930242 0.95281611 -6.2123748
23659 5.960003  0.28640956 0.7780801393 0.95281611 -6.2473473
47791 7.809457  0.01968990 0.9845133510 0.98451335 -6.2771041
37089 6.668566 -0.08389176 0.9341379292 0.97006631 -6.2818256
54624 7.381450 -0.13412714 0.8948314579 0.97006631 -6.2842430</code></pre>
<p>The following plots show the expression of the top 9 genes from the brain development set as ranked by their differential expression with regard to CMV positive and CMV negative status.</p>
<p><img src="figure/STAR-FC-all.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/d96660bec01347bcec29e84c1c4313e95abc8d04/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-24-1.png" target="_blank">d96660b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-09
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/10dcedf7a44e759df3760eaf3a109c4c33ebefeb/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-24-1.png" target="_blank">10dcedf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-11-06
</td>
</tr>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/amnio-cell-free-rna/blob/a91102dcce4c98868c8ad20f025d55dfb71d9ec4/docs/figure/STAR-FC-all.Rmd/unnamed-chunk-24-1.png" target="_blank">a91102d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2020-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Although the effective library sizes were low, the data is generally of good quality. We found at total of 1 differentially expressed genes at FDR &lt; 0.05. The significant genes were enriched for GO terms associated with interferon response and similar. Further gene set testing results indicate an upregulation of interferon response genes in the CMV positive samples, relative to the CMV negative samples, which is consistent with the top genes from the DE analysis.</p>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2        EGSEA_1.18.1             
 [3] pathview_1.30.1           topGO_2.42.0             
 [5] SparseM_1.78              GO.db_3.12.1             
 [7] graph_1.68.0              gage_2.40.1              
 [9] patchwork_1.1.1           NMF_0.23.0               
[11] cluster_2.1.0             rngtools_1.5             
[13] pkgmaker_0.32.2           registry_0.5-1           
[15] edgeR_3.32.1              limma_3.46.0             
[17] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.14.0         
[19] AnnotationFilter_1.14.0   GenomicFeatures_1.42.1   
[21] AnnotationDbi_1.52.0      Biobase_2.50.0           
[23] GenomicRanges_1.42.0      GenomeInfoDb_1.26.7      
[25] IRanges_2.24.1            S4Vectors_0.28.1         
[27] BiocGenerics_0.36.1       forcats_0.5.1            
[29] stringr_1.4.0             dplyr_1.0.4              
[31] purrr_0.3.4               readr_1.4.0              
[33] tidyr_1.1.2               tibble_3.1.2             
[35] ggplot2_3.3.5             tidyverse_1.3.0          
[37] here_1.0.1                workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3              rtracklayer_1.50.0         
  [3] Glimma_2.0.0                bit64_4.0.5                
  [5] knitr_1.31                  multcomp_1.4-16            
  [7] DelayedArray_0.16.3         data.table_1.13.6          
  [9] hwriter_1.3.2               KEGGREST_1.30.1            
 [11] RCurl_1.98-1.3              doParallel_1.0.16          
 [13] generics_0.1.0              metap_1.4                  
 [15] org.Mm.eg.db_3.12.0         TH.data_1.0-10             
 [17] RSQLite_2.2.5               bit_4.0.4                  
 [19] mutoss_0.1-12               xml2_1.3.2                 
 [21] lubridate_1.7.9.2           httpuv_1.5.5               
 [23] SummarizedExperiment_1.20.0 assertthat_0.2.1           
 [25] xfun_0.23                   hms_1.0.0                  
 [27] evaluate_0.14               promises_1.2.0.1           
 [29] fansi_0.5.0                 progress_1.2.2             
 [31] caTools_1.18.1              dbplyr_2.1.0               
 [33] readxl_1.3.1                Rgraphviz_2.34.0           
 [35] DBI_1.1.1                   geneplotter_1.68.0         
 [37] tmvnsim_1.0-2               htmlwidgets_1.5.3          
 [39] ellipsis_0.3.2              backports_1.2.1            
 [41] annotate_1.68.0             PADOG_1.32.0               
 [43] gbRd_0.4-11                 gridBase_0.4-7             
 [45] biomaRt_2.46.3              MatrixGenerics_1.2.1       
 [47] HTMLUtils_0.1.7             vctrs_0.3.8                
 [49] cachem_1.0.4                withr_2.4.2                
 [51] globaltest_5.44.0           GenomicAlignments_1.26.0   
 [53] prettyunits_1.1.1           mnormt_2.0.2               
 [55] lazyeval_0.2.2              crayon_1.4.1               
 [57] genefilter_1.72.1           labeling_0.4.2             
 [59] pkgconfig_2.0.3             nlme_3.1-152               
 [61] ProtGenerics_1.22.0         GSA_1.03.1                 
 [63] rlang_0.4.11                lifecycle_1.0.0            
 [65] sandwich_3.0-0              BiocFileCache_1.14.0       
 [67] mathjaxr_1.2-0              modelr_0.1.8               
 [69] cellranger_1.1.0            rprojroot_2.0.2            
 [71] GSVA_1.38.2                 matrixStats_0.59.0         
 [73] Matrix_1.3-2                zoo_1.8-9                  
 [75] reprex_1.0.0                whisker_0.4                
 [77] png_0.1-7                   viridisLite_0.4.0          
 [79] bitops_1.0-7                KernSmooth_2.23-18         
 [81] Biostrings_2.58.0           blob_1.2.1                 
 [83] R2HTML_2.3.2                doRNG_1.8.2                
 [85] scales_1.1.1                memoise_2.0.0.9000         
 [87] GSEABase_1.52.1             magrittr_2.0.1             
 [89] plyr_1.8.6                  safe_3.30.0                
 [91] gplots_3.1.1                zlibbioc_1.36.0            
 [93] compiler_4.0.2              plotrix_3.8-1              
 [95] KEGGgraph_1.50.0            DESeq2_1.30.1              
 [97] Rsamtools_2.6.0             cli_3.0.0                  
 [99] XVector_0.30.0              EGSEAdata_1.18.0           
[101] MASS_7.3-53.1               tidyselect_1.1.0           
[103] stringi_1.5.3               highr_0.8                  
[105] yaml_2.2.1                  askpass_1.1                
[107] locfit_1.5-9.4              grid_4.0.2                 
[109] tools_4.0.2                 rstudioapi_0.13            
[111] foreach_1.5.1               git2r_0.28.0               
[113] farver_2.1.0                digest_0.6.27              
[115] Rcpp_1.0.6                  broom_0.7.4                
[117] later_1.1.0.1               org.Hs.eg.db_3.12.0        
[119] httr_1.4.2                  Rdpack_2.1                 
[121] colorspace_2.0-2            rvest_0.3.6                
[123] XML_3.99-0.5                fs_1.5.0                   
[125] splines_4.0.2               statmod_1.4.35             
[127] sn_1.6-2                    multtest_2.46.0            
[129] plotly_4.9.3                xtable_1.8-4               
[131] jsonlite_1.7.2              R6_2.5.0                   
[133] TFisher_0.2.0               KEGGdzPathwaysGEO_1.28.0   
[135] pillar_1.6.1                htmltools_0.5.1.1          
[137] hgu133plus2.db_3.2.3        glue_1.4.2                 
[139] fastmap_1.1.0               DT_0.17                    
[141] BiocParallel_1.24.1         codetools_0.2-18           
[143] mvtnorm_1.1-1               utf8_1.2.1                 
[145] lattice_0.20-41             numDeriv_2016.8-1.1        
[147] hgu133a.db_3.2.3            curl_4.3                   
[149] gtools_3.8.2                openssl_1.4.3              
[151] survival_3.2-7              rmarkdown_2.6              
[153] org.Rn.eg.db_3.12.0         munsell_0.5.0              
[155] GenomeInfoDbData_1.2.4      iterators_1.0.13           
[157] haven_2.3.1                 reshape2_1.4.4             
[159] gtable_0.3.0                rbibutils_2.0              </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2        EGSEA_1.18.1             
 [3] pathview_1.30.1           topGO_2.42.0             
 [5] SparseM_1.78              GO.db_3.12.1             
 [7] graph_1.68.0              gage_2.40.1              
 [9] patchwork_1.1.1           NMF_0.23.0               
[11] cluster_2.1.0             rngtools_1.5             
[13] pkgmaker_0.32.2           registry_0.5-1           
[15] edgeR_3.32.1              limma_3.46.0             
[17] EnsDb.Hsapiens.v86_2.99.0 ensembldb_2.14.0         
[19] AnnotationFilter_1.14.0   GenomicFeatures_1.42.1   
[21] AnnotationDbi_1.52.0      Biobase_2.50.0           
[23] GenomicRanges_1.42.0      GenomeInfoDb_1.26.7      
[25] IRanges_2.24.1            S4Vectors_0.28.1         
[27] BiocGenerics_0.36.1       forcats_0.5.1            
[29] stringr_1.4.0             dplyr_1.0.4              
[31] purrr_0.3.4               readr_1.4.0              
[33] tidyr_1.1.2               tibble_3.1.2             
[35] ggplot2_3.3.5             tidyverse_1.3.0          
[37] here_1.0.1                workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3              rtracklayer_1.50.0         
  [3] Glimma_2.0.0                bit64_4.0.5                
  [5] knitr_1.31                  multcomp_1.4-16            
  [7] DelayedArray_0.16.3         data.table_1.13.6          
  [9] hwriter_1.3.2               KEGGREST_1.30.1            
 [11] RCurl_1.98-1.3              doParallel_1.0.16          
 [13] generics_0.1.0              metap_1.4                  
 [15] org.Mm.eg.db_3.12.0         TH.data_1.0-10             
 [17] RSQLite_2.2.5               bit_4.0.4                  
 [19] mutoss_0.1-12               xml2_1.3.2                 
 [21] lubridate_1.7.9.2           httpuv_1.5.5               
 [23] SummarizedExperiment_1.20.0 assertthat_0.2.1           
 [25] xfun_0.23                   hms_1.0.0                  
 [27] evaluate_0.14               promises_1.2.0.1           
 [29] fansi_0.5.0                 progress_1.2.2             
 [31] caTools_1.18.1              dbplyr_2.1.0               
 [33] readxl_1.3.1                Rgraphviz_2.34.0           
 [35] DBI_1.1.1                   geneplotter_1.68.0         
 [37] tmvnsim_1.0-2               htmlwidgets_1.5.3          
 [39] ellipsis_0.3.2              backports_1.2.1            
 [41] annotate_1.68.0             PADOG_1.32.0               
 [43] gbRd_0.4-11                 gridBase_0.4-7             
 [45] biomaRt_2.46.3              MatrixGenerics_1.2.1       
 [47] HTMLUtils_0.1.7             vctrs_0.3.8                
 [49] cachem_1.0.4                withr_2.4.2                
 [51] globaltest_5.44.0           GenomicAlignments_1.26.0   
 [53] prettyunits_1.1.1           mnormt_2.0.2               
 [55] lazyeval_0.2.2              crayon_1.4.1               
 [57] genefilter_1.72.1           labeling_0.4.2             
 [59] pkgconfig_2.0.3             nlme_3.1-152               
 [61] ProtGenerics_1.22.0         GSA_1.03.1                 
 [63] rlang_0.4.11                lifecycle_1.0.0            
 [65] sandwich_3.0-0              BiocFileCache_1.14.0       
 [67] mathjaxr_1.2-0              modelr_0.1.8               
 [69] cellranger_1.1.0            rprojroot_2.0.2            
 [71] GSVA_1.38.2                 matrixStats_0.59.0         
 [73] Matrix_1.3-2                zoo_1.8-9                  
 [75] reprex_1.0.0                whisker_0.4                
 [77] png_0.1-7                   viridisLite_0.4.0          
 [79] bitops_1.0-7                KernSmooth_2.23-18         
 [81] Biostrings_2.58.0           blob_1.2.1                 
 [83] R2HTML_2.3.2                doRNG_1.8.2                
 [85] scales_1.1.1                memoise_2.0.0.9000         
 [87] GSEABase_1.52.1             magrittr_2.0.1             
 [89] plyr_1.8.6                  safe_3.30.0                
 [91] gplots_3.1.1                zlibbioc_1.36.0            
 [93] compiler_4.0.2              plotrix_3.8-1              
 [95] KEGGgraph_1.50.0            DESeq2_1.30.1              
 [97] Rsamtools_2.6.0             cli_3.0.0                  
 [99] XVector_0.30.0              EGSEAdata_1.18.0           
[101] MASS_7.3-53.1               tidyselect_1.1.0           
[103] stringi_1.5.3               highr_0.8                  
[105] yaml_2.2.1                  askpass_1.1                
[107] locfit_1.5-9.4              grid_4.0.2                 
[109] tools_4.0.2                 rstudioapi_0.13            
[111] foreach_1.5.1               git2r_0.28.0               
[113] farver_2.1.0                digest_0.6.27              
[115] Rcpp_1.0.6                  broom_0.7.4                
[117] later_1.1.0.1               org.Hs.eg.db_3.12.0        
[119] httr_1.4.2                  Rdpack_2.1                 
[121] colorspace_2.0-2            rvest_0.3.6                
[123] XML_3.99-0.5                fs_1.5.0                   
[125] splines_4.0.2               statmod_1.4.35             
[127] sn_1.6-2                    multtest_2.46.0            
[129] plotly_4.9.3                xtable_1.8-4               
[131] jsonlite_1.7.2              R6_2.5.0                   
[133] TFisher_0.2.0               KEGGdzPathwaysGEO_1.28.0   
[135] pillar_1.6.1                htmltools_0.5.1.1          
[137] hgu133plus2.db_3.2.3        glue_1.4.2                 
[139] fastmap_1.1.0               DT_0.17                    
[141] BiocParallel_1.24.1         codetools_0.2-18           
[143] mvtnorm_1.1-1               utf8_1.2.1                 
[145] lattice_0.20-41             numDeriv_2016.8-1.1        
[147] hgu133a.db_3.2.3            curl_4.3                   
[149] gtools_3.8.2                openssl_1.4.3              
[151] survival_3.2-7              rmarkdown_2.6              
[153] org.Rn.eg.db_3.12.0         munsell_0.5.0              
[155] GenomeInfoDbData_1.2.4      iterators_1.0.13           
[157] haven_2.3.1                 reshape2_1.4.4             
[159] gtable_0.3.0                rbibutils_2.0              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
