---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

Transcriptomic analysis of cell-free RNA from the amniotic fluid of women with suspected cytomegalovirus infection
==================================================================================================================  

This site contains the analysis of RNAseq data generated from amniotic fluid 
cell-free RNA (cfRNA). 

The data is from Lisa Hui in the Embryonic Epidemiology group at the Murdoch 
Children's Research Institute (MCRI). They have collected amniotic fluid from 
pregnancies with suspected (and subsequently confirmed by PCR) cytomegalovirus 
(CMV) infection. They also have amniotic fluid from *normal* pregnancies i.e. 
pregnancies that were confirmed by PCR to be negative for CMV infection. They 
are interested in differences in the cell-free RNA (cfRNA) between the CMV 
negative and positive pregnancies. Nineteen samples (10 CMV negative and 9 CMV 
positive) were originally sequenced (150bp paired-end) at the MCRI TGU. A 
further batch of 9 samples was subsequently sequenced. 

Follow the links below to explore the different parts of the analysis. 


Pseudoalignment-based differential gene expression (DGE) analysis
-----------------------------------------------------------------

* Pseudoalignment and quality control.
    * [MultiQC report](multiqc_report.html): Pseudoalignment and quality statistics.
    
* DGE analysis (all samples)
    * [DGE results](salmon-limma-voom.html): This is the initial 
    analysis comparing CMV positive and CMV negative samples. Following quality 
    control and normalisation, we preformed a differential expression analysis 
    between CMV positive and CMV negative samples using
    [limma](https://bioconductor.org/packages/release/bioc/html/limma.html).

* DGE analysis (samples with ultrasound abnormalities excluded)
    * [DGE results](salmon-limma-voom-exclude-CMV11.html): 
    This is the initial analysis comparing CMV positive and CMV negative samples. 
    Following quality control and normalisation, we preformed a differential 
    expression analysis between CMV positive and CMV negative samples using
    [limma](https://bioconductor.org/packages/release/bioc/html/limma.html), but 
    excluding samples that had an ultrasound abnormality.

* DGE analysis (all samples)
    * [DGE results](salmon-RUV-all.html): Following quality control and 
    normalisation, we preformed a differential expression analysis between CMV 
    positive and CMV negative samples, using
    [RUVseq](https://bioconductor.org/packages/release/bioc/html/RUVSeq.html) to 
    remove unwanted variation, followed by [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html). 

    
Mapping-based differential expression analysis
----------------------------------------------

* Read mapping, counting and quality control.
    * [MultiQC report](multiqc_report.html): Mapping and quality statistics.

* Differential expression analysis (all samples)
    * [DGE results](STAR-FC-RUV-all.html): DGE analysis comparing 
    CMV positive and CMV negative samples of the full cohort, using
    [RUVseq](https://bioconductor.org/packages/release/bioc/html/RUVSeq.html) to 
    remove unwanted variation, followed by [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html).

* Differential expression analysis (samples with ultrasound abnormalities excluded)
    * [DGE results](STAR-FC-RUV-exclude-US-ab.html): 
    DGE analysis comparing CMV positive and CMV negative samples, excluding 
    samples with ultrasound abnormalities, using 
    [RUVseq](https://bioconductor.org/packages/release/bioc/html/RUVSeq.html) to 
    remove unwanted variation, followed by [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html).


Differential exon usage (DEU) analysis
--------------------------------------------

* Differential exon usage analysis
    * [DEU results (all samples)](STAR-DEXSeq.html): DEU analysis comparing 
    CMV positive and CMV negative samples of the full cohort using
    [DEXSeq](https://bioconductor.org/packages/3.13/bioc/html/DEXSeq.html).
    * [DEU results (samples with ultrasound abnormalities excluded)](STAR-DEXSeq-exclude-US-ab.html):
    DEU analysis comparing CMV positive and CMV negative samples, excluding 
    samples with ultrasound abnormalities, using [DEXSeq](https://bioconductor.org/packages/3.13/bioc/html/DEXSeq.html).


Licenses
--------

The code in this analysis is covered by the [MIT license][mit] and the written
content on this website is covered by a [Creative Commons CC-BY][cc] license.


Citations
---------

[mit]: https://choosealicense.com/licenses/mit/ "MIT License"
[cc]: https://creativecommons.org/licenses/by/4.0/ "CC-BY License"
