---
title: "Differential Transcript Usage (DTU) Analysis: DRIMseq"
output:
  html_document:
    df_print: paged
---

```{r, message=FALSE, echo=FALSE}
library(here)
library(tidyverse)
library(readr)
library(limma)
library(edgeR)
library(patchwork)
library(tximport)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(DRIMSeq)
library(BANDITS)
library(stageR)
```

# Data import

```{r, message=FALSE, echo=FALSE}
files <- file.path(list.files(here( "data/gene-transcriptome-analysis/quants"), full.names = TRUE), 
                   "quant.sf")
sample_names <- strsplit2(files, "_")[,6]
names(files) <- sample_names
files <- files[grepl("CMV", names(files))]
```

Load sample information.

```{r, echo=FALSE, message=FALSE}
samps1 <- read_csv(here("data/CMV-AF-database-corrected-oct-2020.csv"))
samps2 <- read_csv(here("data/samples.csv"))

samps1 %>% full_join(samps2, by = c("sequencing_ID" = "SampleId")) %>%
  mutate(pair = ifelse(!is.na(matched_pair), matched_pair,
                        ifelse(!is.na(MatchedPair), MatchedPair, NA)),
         CMV_status = ifelse(!is.na(CMV_status), CMV_status,
                         ifelse(!is.na(TestResult), TestResult, NA)),
         Sex = toupper(Sex),
         Indication = tolower(Indication)) %>%
  dplyr::rename(sex = Sex, 
         id = sequencing_ID, 
         indication = Indication,
         GA_at_amnio = `GA_at_amnio-completed_weeks`) -> samps
       
read_csv(file = here("data/metadata.csv")) %>%
  inner_join(read_csv(file = here("data/joindata.csv")), 
                      by = c("Record.ID" = "UR")) %>%
  right_join(samps, by = c("ID post-extraction" = "id")) %>%
  na_if("NA") %>%
  mutate(sex = ifelse(!is.na(sex), sex,
                        ifelse(!is.na(Sex), toupper(Sex), NA)),
         GA_at_amnio = ifelse(!is.na(GA_at_amnio), GA_at_amnio,
                         ifelse(!is.na(GA.at.amnio), GA.at.amnio, NA))) %>%
  dplyr::rename(id = `ID post-extraction`) %>%
  dplyr::select(id, 
                CMV_status, 
                pair, 
                sex, 
                GA_at_amnio, 
                indication) %>%
  dplyr::filter(id %in% names(files)) %>%
  drop_na() -> targets
  
targets %>% knitr::kable()
```

Only retain paired samples with clinical information for donstream analysis. 

```{r}
int <- intersect(names(files), targets$id)
targets <- targets[match(int, targets$id),]
files <- files[match(int, names(files))]

files
```


Read in transcript count data and remove transcripts with no counts.

```{r}
txi <- tximport(files, type = "salmon", txOut = TRUE)
cts <- txi$counts
cts <- cts[rowSums(cts) > 0,]

dim(cts)
```

Plot effective library size per sample.

```{r}
colSums(cts) %>%
  data.frame %>%
  rownames_to_column(var = "sample") %>%
  dplyr::rename("libsize" = ".") %>%
ggplot(aes(x = sample, y = libsize)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

Associate transcripts with gene IDs for gene-level summarization. Filter out genes without Entrez IDs, genes with only one transcript isoform and transcripts without any counts across all samples.

```{r}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
txdf <- select(txdb, keys(txdb, keytype = "TXNAME"), 
               "GENEID", "TXNAME", "SYMBOL") %>%
  drop_na(GENEID)

txdf %>% dplyr::inner_join(select(org.Hs.eg.db, 
                                  keys = unique(txdf$GENEID), 
                           columns = c("SYMBOL","ENTREZID"),
                           keytype="ENTREZID"), 
                           by = c("GENEID" = "ENTREZID")) %>%
  dplyr::distinct() %>%
  dplyr::filter((duplicated(GENEID) | 
                  duplicated(GENEID, fromLast = TRUE)) &
                  TXNAME %in% rownames(cts)) -> txdf
 
head(txdf, n = 10) %>% knitr::kable()
```

Match transcript count data to transcript annotation table.

```{r}
cts <- cts[match(txdf$TXNAME, rownames(cts)),]

head(cts) %>% knitr::kable()
```

# DRIMseq analysis

Filter transcripts using *BANDITS* strategy.

```{r}
transcripts_to_keep <- filter_transcripts(gene_to_transcript = txdf[, c(2,1)],
                                          transcript_counts = cts, 
                                          min_transcript_proportion = 0.01,
                                          min_transcript_counts = min(table(targets$CMV_status)), 
                                          min_gene_counts = ncol(cts))

length(transcripts_to_keep)
```

Keep only transcripts that pass filtering criteria.

```{r}
keep <- match(transcripts_to_keep, rownames(cts))

cts <- cts[keep,]
txdf <- txdf[keep,]

cts %>% data.frame %>%
  rownames_to_column(var = "feature_id") %>% 
  mutate(gene_id = txdf$GENEID) %>%
  relocate(gene_id, .before = feature_id) -> cts

d <- dmDSdata(counts = cts, 
              samples = targets %>% dplyr::select(id, CMV_status) %>%
                dplyr::rename(sample_id = id,
                              condition = CMV_status) %>% data.frame)
d
```

Setup the design matrix.

```{r}
design_full <- model.matrix(~ condition, 
                            data = DRIMSeq::samples(d))
colnames(design_full)
```

Perform DTU analysis.

```{r}
out <- here("data/rds/dmResults.rds")

if(!file.exists(out)){  
set.seed(1)
  d <- dmPrecision(d, design = design_full)
  d <- dmFit(d, design = design_full)
  d <- dmTest(d, coef = "conditionneg")
  
  saveRDS(d, file = out)
  
} else{
  d <- readRDS(out)
  
}
```

Extract the results by gene.

```{r}
res <- DRIMSeq::results(d)
head(res) %>% knitr::kable()
```

Extract the results by transcript.

```{r}
res.txp <- DRIMSeq::results(d, level="feature")
head(res.txp) %>% knitr::kable()
```

Convert any NAs in the pvalue column into 1’s to avoid problems in downstream analysis. 

```{r}
no.na <- function(x) ifelse(is.na(x), 1, x)
res$pvalue <- no.na(res$pvalue)
res.txp$pvalue <- no.na(res.txp$pvalue)
```

Identify genes with statistically significant DTU.

```{r}
idx <- which(res$adj_pvalue < 0.05)
res[idx,] %>% knitr::kable()
```

```{r, fig.asp=1}
plotProportions(d, res$gene_id[idx][1], "condition")
```

# Adjust p-values with stageR

A typical analysis of differential transcript usage would involve asking first: “which genes contain any evidence of DTU?”, and secondly, “which transcripts in the genes that contain some evidence may be participating in the DTU?” A gene may pass the first stage without exhibiting enough evidence to identify one or more transcripts that are participating in the DTU. The stageR package is designed to allow for such two-stage testing procedures, where the first stage is called a screening stage and the second stage a confirmation stage. 

We first construct a vector of p-values for the screening stage.

```{r}
pScreen <- setNames(res$pvalue, res$gene_id)
```

We construct a one column matrix of the confirmation p-values:

```{r}
pConfirmation <- matrix(res.txp$pvalue, ncol = 1)
strp <- function(x) substr(x, 1, 15)
rownames(pConfirmation) <- strp(res.txp$feature_id)
```

We arrange a two column data.frame with the transcript and gene identifiers.

```{r}
tx2gene <- res.txp[,c("feature_id", "gene_id")]
tx2gene[,1] <- strp(tx2gene[,1])
```

The following functions then perform the stageR analysis. We must specify an alpha, which will be the overall false discovery rate target for the analysis, defined below. Unlike typical adjusted p-values or q-values, we cannot choose an arbitrary threshold later: after specifying alpha=0.05, we need to use 5% as the target in downstream steps. 

```{r}
stageRObj <- stageRTx(pScreen = pScreen, 
                      pConfirmation = pConfirmation,
                      pScreenAdjusted = FALSE, 
                      tx2gene = tx2gene)
stageRObj <- stageWiseAdjustment(stageRObj, method = "dtu", alpha = 0.05) 
suppressWarnings({
  drim.padj <- getAdjustedPValues(stageRObj, 
                                  order=FALSE,
                                  onlySignificantGenes=TRUE)
})

head(drim.padj) %>% knitr::kable()
```

```{r}
select(org.Hs.eg.db, 
       keys = drim.padj$geneID, 
       columns = c("SYMBOL","ENTREZID"),
       keytype = "ENTREZID") %>%
  dplyr::distinct() %>% 
  dplyr::inner_join(drim.padj, by = c("ENTREZID" = "geneID")) %>%
  dplyr::arrange(gene, transcript) -> top_gene

top_gene %>% knitr::kable()
```

```{r, fig.asp=20}
dtu_genes <- unique(top_gene$ENTREZID) 
p <- vector("list", length(dtu_genes))

for(i in 1:length(p)){
  p[[i]] <- plotProportions(d, unique(top_gene$ENTREZID)[1], "condition") +
    ggtitle(top_gene$SYMBOL[top_gene == dtu_genes[i]]) +
    theme(axis.text = element_text(size = 8))
}

wrap_plots(p, ncol = 1)
```

# Gene set testing

```{r}
gst <- goana(dtu_genes, 
             universe = d@results_gene$gene_id)
topGO(gst) %>% 
  mutate(FDR = p.adjust(P.DE)) %>% 
  knitr::kable()
```

```{r}
gst <- kegga(dtu_genes, 
             universe = d@results_gene$gene_id)
topKEGG(gst) %>% 
  mutate(FDR = p.adjust(P.DE)) %>% 
  knitr::kable()
```



