---
title: "Differential Exon Usage Analysis: DEXSeq"
output:
  html_document:
    df_print: paged
---

```{r, message=FALSE, echo=FALSE}
library(here)
library(tidyverse)
library(readr)
library(NMF)
library(patchwork)
library(ggplot2)
library(DEXSeq)
library(limma)
```

# Data import

```{r}
files <- list.files(here("data/star-genome-analysis/count-exons-pe"), 
                        pattern="PE.txt$", full.names = TRUE)
names(files) <- strsplit2(files, "_")[,6]
files <- files[grepl("CMV", names(files))]

flatGff <- list.files(here("data/star-genome-analysis"), 
                           pattern="DEXSeq.chr.gff$", full.names = TRUE)

basename(files)
```

Load sample information.

```{r, echo=FALSE, message=FALSE}
samps1 <- read_csv(here("data/CMV-AF-database-corrected-oct-2020.csv"))
samps2 <- read_csv(here("data/samples.csv"))

samps1 %>% full_join(samps2, by = c("sequencing_ID" = "SampleId")) %>%
  mutate(pair = ifelse(!is.na(matched_pair), matched_pair,
                        ifelse(!is.na(MatchedPair), MatchedPair, NA)),
         CMV_status = ifelse(!is.na(CMV_status), CMV_status,
                         ifelse(!is.na(TestResult), TestResult, NA)),
         Sex = toupper(Sex),
         Indication = tolower(Indication)) %>%
  dplyr::rename(sex = Sex, 
         id = sequencing_ID, 
         indication = Indication,
         GA_at_amnio = `GA_at_amnio-completed_weeks`) -> samps
       
read_csv(file = here("data/metadata.csv")) %>%
  inner_join(read_csv(file = here("data/joindata.csv")), 
                      by = c("Record.ID" = "UR")) %>%
  right_join(samps, by = c("ID post-extraction" = "id")) %>%
  na_if("NA") %>%
  mutate(sex = ifelse(!is.na(sex), sex,
                        ifelse(!is.na(Sex), toupper(Sex), NA)),
         GA_at_amnio = ifelse(!is.na(GA_at_amnio), GA_at_amnio,
                         ifelse(!is.na(GA.at.amnio), GA.at.amnio, NA))) %>%
  dplyr::rename(id = `ID post-extraction`) %>%
  dplyr::select(id, 
                CMV_status, 
                pair, 
                sex, 
                GA_at_amnio, 
                indication) %>%
  dplyr::filter(id %in% names(files)) %>%
  drop_na() -> targets
  
targets %>% knitr::kable()
```

Only retain paired samples with clinical information for donstream analysis. 

```{r}
int <- intersect(names(files), targets$id)
targets <- targets[match(int, targets$id),]
files <- files[match(int, names(files))]

basename(files)
```

Read in transcript count data and remove transcripts with no counts.

```{r}
sampleTable <- data.frame(row.names = targets$id,
                          condition = targets$CMV_status,
                          pair = targets$pair)
head(sampleTable) %>% knitr::kable()
```

Setup the **DEXSeq** data.

```{r}
formulaFullModel    =  ~ sample + exon + pair:exon + condition:exon
formulaReducedModel =  ~ sample + exon + pair:exon

out <- here("data/rds/DEXSeq.rds")
if(!file.exists(out)){
  dxd <- DEXSeqDataSetFromHTSeq(
    files,
    sampleData = sampleTable,
    design = ~ sample + exon + condition:exon,
    flattenedfile = flatGff )
} else {
  dxd <- readRDS(file = out)
}
```
```{r}
if(!file.exists(out)){
  dxd = estimateSizeFactors( dxd )
} 
```

```{r}
if(!file.exists(out)){
  BPPARAM = MulticoreParam(min(26, multicoreWorkers()))
  dxd = estimateDispersions( dxd, 
                             BPPARAM=BPPARAM,
                             formula = formulaFullModel )
}
plotDispEsts( dxd )
```

```{r}
if(!file.exists(out)){
  dxd = testForDEU( dxd, 
                    BPPARAM=BPPARAM,
                    reducedModel = formulaReducedModel, 
                    fullModel = formulaFullModel )
}
```

```{r}
if(!file.exists(out)){
  dxd = estimateExonFoldChanges(dxd, BPPARAM=BPPARAM )
  saveRDS(dxd, file = out)
} 
```

```{r}
dxr1 = DEXSeqResults( dxd )
table ( dxr1$padj < 0.1 )
```

```{r}
table ( tapply( dxr1$padj < 0.1, dxr1$groupID, any ) )
```

```{r}
DEXSeq::plotMA( dxr1, cex=0.8 )
```

```{r, warning=FALSE, message=FALSE, fig.asp = 1.5}
plotDEXSeq( dxr1, unique(dxr1$groupID[which(dxr1$padj < 0.1)])[1], 
            legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2,
            expression=FALSE, splicing=TRUE,
            displayTranscripts=TRUE)
```
```{r, warning=FALSE, message=FALSE, fig.asp=1.5}
plotDEXSeq( dxr1, unique(dxr1$groupID[which(dxr1$padj < 0.1)])[2], 
            legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 ,
            expression=FALSE, splicing=TRUE,
            displayTranscripts=TRUE)
```

