---
title: "Limma-voom paired analysis EXCLUDING US-ab samples"
output:
  html_document:
    df_print: paged
---

```{r, message=FALSE, echo=FALSE}
library(here)
library(tidyverse)
library(EnsDb.Hsapiens.v86)
library(readr)
library(limma)
library(edgeR)
library(NMF)
library(patchwork)
library(EGSEA)
library(gt)
source(here("code/output.R"))
```

The data showed some adapter contamination and sequence duplication issues. Adapters were removed using `Trimmomatic` and both paired and unpaired reads were retained. Only paired reads were initially mapped with `Star` in conjunction with `GRCh38` and `gencode_v34` to detect all junctions, across all samples. Paired and unpaired reads were then mapped to `GRCh38` separately using `Star`. Duplicates were removed from paired and unpaired mapped data using `Picard MarkDuplicates`. Reads were then counted across features from `gencode_v34` using `featureCounts`.        

# Data import

Set up `DGElist` object for downstream analysis. Sum paired and unpaired counts prior to downstream analysis.

```{r, message=FALSE, echo=FALSE}
rawPE <- read_delim(here("data/star-genome-analysis/counts-pe/counts.txt"), delim = "\t", skip = 1)
rawSE <- read_delim(here("data/star-genome-analysis/counts-se/counts.txt"), delim = "\t", skip = 1)
samps <- strsplit2(colnames(rawPE)[c(7:ncol(rawPE))], "_")[,5]
batch <- factor(strsplit2(colnames(rawPE)[c(7:ncol(rawPE))], 
                                     "_")[,1], labels = 1:2)
batch <- tibble(batch = batch, id = samps)
colnames(rawPE)[7:ncol(rawPE)] <- samps
colnames(rawSE)[7:ncol(rawSE)] <- samps

counts <- rawPE[, 7:ncol(rawPE)] + rawSE[, 7:ncol(rawSE)] 
dge <- DGEList(counts = counts,
                 genes = rawPE[,c(1,6)])
dge
```

Load sample information and file names. Exclude samples with ultrasound abnormalities.

```{r, echo=FALSE, message=FALSE}
samps1 <- read_csv(here("data/CMV-AF-database-corrected-oct-2020.csv"))
samps2 <- read_csv(here("data/samples.csv"))

samps1 %>% full_join(samps2, by = c("sequencing_ID" = "SampleId")) %>%
  mutate(pair = ifelse(!is.na(matched_pair), matched_pair,
                        ifelse(!is.na(MatchedPair), MatchedPair, NA)),
         CMV_status = ifelse(!is.na(CMV_status), CMV_status,
                         ifelse(!is.na(TestResult), TestResult, NA)),
         Sex = toupper(Sex),
         Indication = tolower(Indication)) %>%
  dplyr::rename(sex = Sex, 
         id = sequencing_ID, 
         indication = Indication,
         GA_at_amnio = `GA_at_amnio-completed_weeks`) -> samps
       
read_csv(file = here("data/metadata.csv")) %>%
  inner_join(read_csv(file = here("data/joindata.csv")), 
                      by = c("Record.ID" = "UR")) %>%
  right_join(samps, by = c("ID post-extraction" = "id")) %>%
  na_if("NA") %>%
  mutate(sex = ifelse(!is.na(sex), sex,
                        ifelse(!is.na(Sex), toupper(Sex), NA)),
         GA_at_amnio = ifelse(!is.na(GA_at_amnio), GA_at_amnio,
                         ifelse(!is.na(GA.at.amnio), GA.at.amnio, NA))) %>%
  dplyr::rename(id = `ID post-extraction`) %>%
  dplyr::select(id, 
                CMV_status, 
                pair, 
                sex, 
                GA_at_amnio, 
                indication) %>%
  left_join(batch) %>%
  dplyr::filter(id %in% colnames(dge)) %>%
  drop_na() -> targets

m <- match(colnames(dge), targets$id)
targets <- targets[m[!is.na(m)], ]
targets <- targets[targets$indication %in% "no_us_ab",]

targets
```


# Quality control

Genes that do not have an adequate number of reads in any sample should be filtered out prior to downstream analyses. From a biological perspective, genes that are not expressed at a biologically meaningful level in any condition are not of interest. Statistically, we get a better estimate of the mean-variance relationship in the data and reduce the number of statistical tests that are performed during differential expression analyses.

Filter out lowly expressed genes and genes without Entrez IDs and calculate TMM normalization factors.

```{r, echo=FALSE, message=FALSE}
z <- dge[, colnames(dge) %in% targets$id] # retain only relevant samples
z$genes$Ensembl <- strsplit2(z$genes$Geneid, ".", fixed = TRUE)[,1]
z$group <- targets$CMV_status

edb <- EnsDb.Hsapiens.v86 # add Gene Symbols and Entrez IDs
z$genes <- left_join(z$genes, ensembldb::genes(edb, 
                                        filter = GeneIdFilter(z$genes$Ensembl), 
                                        columns = c("gene_id", 
                                                    "symbol", 
                                                    "entrezid"), 
                                        return.type = "data.frame"), 
                       by = c("Ensembl" = "gene_id"))
z$genes$entrezid <- unlist(sapply(z$genes$entrezid, function(x) {
  if(is.null(x)) NA else x[length(x)]
}), use.names = FALSE)

keep <- !is.na(z$genes$entrezid) & !is.null(z$genes$entrezid)
x <- z[keep, ] # remove genes without Entrez IDs

keep <- filterByExpr(x, group = z$group)
x <- x[keep, ] # remove lowly expressed genes

y <- calcNormFactors(x)
y
```
Plotting the distribution log-CPM values shows that a majority of genes within each sample are either not expressed or lowly-expressed with log-CPM values that are small or negative.

```{r, echo=FALSE, message=FALSE}
L <- mean(z$samples$lib.size) * 1e-6
M <- median(z$samples$lib.size) * 1e-6

par(mfrow = c(1,2))
lcpmz <- cpm(z, log = TRUE)
lcpm.cutoff <- log2(10/M + 2/L)
nsamples <- ncol(z)
col <- scales::hue_pal()(nsamples)
plot(density(lcpmz[,1]), col = col[1], lwd = 1, ylim = c(0, 2), las = 2, 
     main = "", xlab = "")
title(main = "Unfiltered data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den <- density(lcpmz[,i])
  lines(den$x, den$y, col = col[i], lwd = 1)
}

lcpmy <- cpm(y, log=TRUE)
plot(density(lcpmy[,1]), col = col[1], lwd = 1, ylim = c(0, 0.25), las = 2, 
     main = "", xlab = "")
title(main = "Filtered data", xlab = "Log-cpm")
abline(v = lcpm.cutoff, lty = 3)
for (i in 2:nsamples){
  den <- density(lcpmy[,i])
  lines(den$x, den$y, col = col[i], lwd = 1)
}
```

Although in excess of 30 million reads were obtained per sample, we can see that after mapping, duplicate removal and quantification of gene expression the median library size is just under than 4 million reads. This suggests that we are likely to only be capturing the most abundant cfRNAs.  

It is assumed that all samples should have a similar range and distribution of expression values. The raw data looks fairly uniform between samples, although TMM normalization further improves this.

```{r, fig.height=8, fig.width=9, echo=FALSE, message=FALSE}
dat <- data.frame(lib = y$samples$lib.size,
                  status = y$group,
                  sample = colnames(y))
p1 <- ggplot(dat, aes(x = sample, y = lib, fill = status)) +
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Sample", y = "Library size",
       fill = "CMV Status") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed") +
  scale_x_discrete(limits = dat$sample)

dat <- reshape2::melt(cpm(y, normalized.lib.sizes = FALSE, log = TRUE),
                      value.name = "cpm")
dat$status <- rep(y$group, each = nrow(y))
colnames(dat)[2] <- "sample"
p2 <- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = "Sample", y = "Library size",
       fill = "CMV Status") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed")

dat <- reshape2::melt(cpm(y, normalized.lib.sizes = TRUE, log = TRUE),
                      value.name = "cpm")
dat$status <- rep(y$group, each = nrow(y))
colnames(dat)[2] <- "sample"
p3 <- ggplot(dat, aes(x = sample, y = cpm, fill = status)) +
  geom_boxplot(show.legend = FALSE, outlier.size = 0.75) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7)) +
  labs(x = "Sample", y = "Library size",
       fill = "CMV Status") +
  geom_hline(yintercept = median(dat$lib), linetype = "dashed")

p1 / (p2 + p3) + plot_layout(guides = "collect")
```

Multi-dimensional scaling (MDS) plots show the largest sources of variation in the data. They are a good way of exploring the relationships between the samples and identifying structure in the data. The following series of MDS plots examines the first four principal components. The samples are coloured by various known features of the samples such as CMV Status and foetal sex. The MDS plots do not show the samples strongly clustering by any of the known features of the data, although there does seem to be some separation between the CMV positive and negative samples in the 1st and 2nd principal components. This indicates that there are possibly some differentially expressed genes between CMV positive and negative samples.

A weak batch effect is also evident in the 3rd principal component, when we examine the plots coloured by batch.

```{r, fig.height=40, fig.width=8, echo=FALSE, message=FALSE}
dims <- list(c(1,2), c(1,3), c(2,3), c(3,4))
vars <- c("CMV_status", "pair", "sex", "GA_at_amnio", "indication", "batch")
patches <- vector("list", length(vars))

for(i in 1:length(vars)){
  p <- vector("list", length(dims))
  
  for(j in 1:length(dims)){
    mds <- plotMDS(cpm(y, log = TRUE), top = 1000, gene.selection="common", 
                   plot = FALSE, dim.plot = dims[[j]])
    dat <- tibble::tibble(x = mds$x, y = mds$y,
                          sample = targets$id,
                          variable = pull(targets, vars[i]))
    
    p[[j]] <- ggplot(dat, aes(x = x, y = y, colour = variable)) +
      geom_text(aes(label = sample), size = 2.5) +
      labs(x = glue::glue("Principal component {dims[[j]][1]}"), 
           y = glue::glue("Principal component {dims[[j]][2]}"),
           colour = vars[i])
  }
  
  patches[[i]] <- wrap_elements(wrap_plots(p, ncol = 2, guides = "collect") +
    plot_annotation(title = glue::glue("Coloured by: {vars[i]}")) &
    theme(legend.position = "bottom"))
  
}

wrap_plots(patches, ncol = 1)
```
Check XIST expression to confirm that all sex assignments are correct.

```{r, message=FALSE, echo=FALSE}
dat <- data.frame(targets, XIST = z$counts[which(z$genes$symbol == "XIST"),]) 
dat %>% gt() %>%
  tab_style(
    style = cell_fill(color = "lightgreen"),
    locations = cells_body(
      rows = (sex == "F" & XIST < 10) | (sex == "M" & XIST >= 10))
  ) %>% 
  tab_style(
    style = cell_text(weight = "bold"),
    locations = cells_column_labels(columns = 1:ncol(dat))
  ) 

```

# Differential expression analysis

Due to the variability in the data, the TMM normalised data was transformed using `voomWithQualityWeights`. This takes into account the differing library sizes and the mean variance relationship in the data as well as calculating sample-specific quality weights. Linear models were fit in `limma`, taking into account the `voom` weights. The CMV positive samples were compared to the CMV negative samples, taking into account the sample pairs. A summary of the number of differentially expressed genes is shown below.

```{r, echo=FALSE, message=FALSE}
design <- model.matrix(~0 + y$group + targets$pair, data = targets)
colnames(design) <- c(levels(factor(y$group)), unique(targets$pair)[-1])
v <- voomWithQualityWeights(y, design, plot = TRUE)
cont <- makeContrasts(contrasts = "pos - neg",
                      levels = design)
fit <- lmFit(v, design)
cfit <- contrasts.fit(fit, cont)
fit2 <- eBayes(cfit, robust = TRUE)
fitSum <- summary(decideTests(fit2, p.value = 0.05))
fitSum
```

There were `r fitSum["Down",]` down-regulated and `r fitSum["Up",]` up-regulated genes between CMV positive and CMV negative samples at FDR < 0.05.  

These are the top 10 differentially expressed genes.

```{r, echo=FALSE, message=FALSE}
top <- topTable(fit2, n = 500)
readr::write_csv(top, file = here("output/star-fc-limma-voom-no_us_ab.csv"))
head(top, n=10)
```

The following plots show the expression of the top 12 ranked differentially expressed genes for CMV positive and CMV negative samples. Although there is significant variability within the groups and the log2 fold changes are not large, there are obvious differences in expression for the top ranked genes.

```{r, fig.width=7, fig.height=8, echo=FALSE, message=FALSE}
dat <- reshape2::melt(cpm(y, log = TRUE),
                      value.name = "cpm")
dat$status <- rep(y$group, each = nrow(y))
dat$gene <- rep(y$genes$Geneid, ncol(y))

p <- vector("list", 12)

for(i in 1:length(p)){
p[[i]] <- ggplot(data = subset(dat, dat$gene == top$Geneid[i]), 
       aes(x = status, y = cpm, colour = status)) +
  geom_jitter(width = 0.25) +
  stat_summary(fun = "mean", geom = "crossbar") +
  labs(x = "Status", y = "log2 CPM", colour = "Status") +
  ggtitle(top$symbol[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = "collect", ncol = 3) & 
  theme(legend.position = "bottom")
```
```{r}
topTable(fit2, num = Inf) %>% 
  mutate(sig = ifelse(adj.P.Val <= 0.05, "<= 0.05", "> 0.05")) -> dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = sig)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val < 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = symbol), 
            size = 2, colour = "black", max.overlaps = 15) +
  labs(x = expression(~log[2]~"(Fold Change)"), 
       y = expression(~-log[10]~"(P-value)"),
       colour = "FDR") +
  scale_colour_brewer(palette = "Set1")
```

```{r, fig.asp=1, message=FALSE}
cut <- 0.05
o <- order(targets$CMV_status)
aheatmap(v$E[v$genes$Geneid %in% top$Geneid[top$adj.P.Val < cut], o], 
         annCol = list(CMV_result = targets$CMV_status[o]),
         Colv = NA,
         labRow = v$genes$symbol[v$genes$Geneid %in% 
                                   top$Geneid[top$adj.P.Val < cut]],
         main = glue::glue("DEGs at FDR < {cut}"))
```

# Gene set enrichment analysis (GSEA)

Testing for enrichment of Gene Ontology (GO) categories among statistically significant differentially expressed genes.

```{r, message=FALSE}
go <- goana(top$entrezid[top$adj.P.Val < 0.05], universe = v$genes$entrezid, 
            trend = v$genes$Length)
topGO(go, number = Inf) %>%
  mutate(FDR = p.adjust(P.DE)) %>%
  dplyr::filter(FDR < 0.05) %>%
  knitr::kable(format.args = list(scientific = -1), digits = 50)
```

GSEA helps us to interpret the results of a differential expression analysis. The `camera` function performs a competitive test to assess whether the genes in a given set are highly ranked in terms of differential expression relative to genes that are not in the set. We have tested several collections of gene sets from the Broad Institute's Molecular Signatures Database [MSigDB](http://software.broadinstitute.org/gsea/msigdb/index.jsp).  

Build gene set indexes. 

```{r, echo=FALSE, message=FALSE}
gsAnnots <- buildIdx(entrezIDs = v$genes$entrezid, species = "human",
                     msigdb.gsets = c("h", "c2", "c5"))
```

The GO gene sets consist of genes annotated by the same GO terms.

```{r, echo=FALSE, message=FALSE}
c5Cam <- camera(v, gsAnnots$c5@idx, design, contrast = cont, trend.var = TRUE)
write.csv(c5Cam[c5Cam$FDR < 0.05,], 
          file = here("output/star-fc-limma-voom-no_us_ab-gsea-c5.csv"))
head(c5Cam, n = 20)
```

The Hallmark gene sets  are coherently expressed signatures derived by aggregating many MSigDB gene sets to represent well-defined biological states or processes.

```{r, echo=FALSE, message=FALSE}
hCam <- camera(v, gsAnnots$h@idx, design, contrast = cont, trend.var = TRUE)
head(hCam, n = 20)
```

Barcode plots show the enrichment of gene sets among up or down-regulated genes. The following barcode plots show the enrichment of the top 4 hallmark gene sets among the genes differentially expressed between CMV positive and CMV negative samples.

```{r, fig.width=8, fig.height=6, echo=FALSE, message=FALSE}
par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(fit2$t[,1], gsAnnots$h@idx[[rownames(hCam)[i]]], 
              main = rownames(hCam)[i], cex.main = 0.75)
})
```

The curated gene sets are compiled from online pathway databases, publications in PubMed, and knowledge of domain experts.

```{r, echo=FALSE, message=FALSE}
c2Cam <- camera(v, gsAnnots$c2@idx, design, contrast = cont, trend.var = TRUE)
write.csv(c2Cam[c2Cam$FDR < 0.05,], 
          file = here("output/star-fc-limma-voom-no_us_ab-gsea-c2.csv"))
head(c2Cam, n = 20)
```

The following barcode plots show the enrichment of the top 4 curated gene sets among the genes differentially expressed between CMV positive and CMV negative samples.

```{r, fig.width=8, fig.height=6, echo=FALSE, message=FALSE}
par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(fit2$t[,1], gsAnnots$c2@idx[[rownames(c2Cam)[i]]], 
              main = rownames(c2Cam)[i], cex.main = 0.75)
})
```

The KEGG gene sets encompass all of the pathways defined in the [Kegg pathway database](https://www.genome.jp/kegg/pathway.html). 

```{r, echo=FALSE, message=FALSE}
keggCam <- camera(v, gsAnnots$kegg@idx, design, contrast = cont, 
                  trend.var = TRUE)
head(keggCam, n = 20)
```

```{r, fig.width=8, fig.height=7, echo=FALSE, message=FALSE}
par(mfrow=c(2,2))
sapply(1:4, function(i){
  barcodeplot(fit2$t[,1], gsAnnots$kegg@idx[[rownames(keggCam)[i]]], 
              main = rownames(keggCam)[i], cex.main = 0.75)
})
```

# Brain development genes

Test only the specialised brain development gene set for differential expression between CMV positive and CMV negative samples. This reduces the multiple testing burden and can identify DEGs from a particular set of interest.

```{r, echo=FALSE, message=FALSE}
brainSet <- readr::read_delim(file = here("data/brain-development-geneset.txt"),
                              delim = "\t", skip = 2, col_names = "BRAIN_DEV")
brainSet
```
```{r, echo=FALSE, message=FALSE}
fit <- lmFit(v[v$genes$symbol %in% brainSet$BRAIN_DEV, ], design)
cfit <- contrasts.fit(fit, cont)
fit2 <- eBayes(cfit, robust = TRUE)
fitSum <- summary(decideTests(fit2, p.value = 0.05))
fitSum
```

```{r, echo=FALSE, message=FALSE}
topBrain <- topTable(fit2, n = Inf)
topBrain
```
The following plots show the expression of the top 9 genes from the brain development set as ranked by their differential expression with regard to CMV positive and CMV negative status. 

```{r, fig.width=7, fig.height=8, echo=FALSE, message=FALSE}
b <- y[y$genes$entrezid %in% topBrain$entrezid[1:9], ]
dat <- reshape2::melt(cpm(b, log = TRUE),
                      value.name = "cpm")
dat$status <- rep(b$group, each = nrow(b))
dat$gene <- rep(b$genes$Geneid, ncol(b))

p <- vector("list", 9)

for(i in 1:length(p)){
p[[i]] <- ggplot(data = subset(dat, dat$gene == topBrain$Geneid[i]), 
       aes(x = status, y = cpm, colour = status)) +
  geom_jitter(width = 0.25) +
  stat_summary(fun = "mean", geom = "crossbar") +
  labs(x = "Status", y = "log2 CPM", colour = "Status") +
  ggtitle(topBrain$symbol[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = "collect", ncol = 3) & 
  theme(legend.position = "bottom")
```

# Summary

Although the effective library sizes were low, the data is generally of good quality. We found at total of `r nrow(top[top$adj.P.Value < 0.05,])` differentially expressed genes at FDR < 0.05. The significant genes were enriched for GO terms associated with interferon response and similar. Further gene set testing results indicate an upregulation of interferon response genes in the CMV positive samples, relative to the CMV negative samples, which is consistent with the top genes from the DE analysis.

